---
title: "CCA"
author: "Julia Golebiowska"
date: "2024-10-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
CCA  - Canonical correspondance analysis on viral data from the Elbe estuary for each fraction and and data type - gene counts and expression. This analysis revels the potential drivers of communities diversity and activity. 



```{r test}
library("beepr")
library(vegan)
library(ggplot2)
library(ggrepel)
library(dplyr)
library(tidyr)
library(ggforce)
```
## Viral metabolic potential

##1. Read data and metadata for each fraction and data type


```{r , echo=FALSE}
#LF
log_gene_abund_LF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_gene_abund_all_LF.RDS")
log_expression_LF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_expression_LF.RDS")
#HF
log_gene_abund_HF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_gene_abund_all_HF.RDS")
log_expression_HF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_expression_HF.RDS")
#vector of numeric parameters columns from metadata
env_numeric <- c("Sat_O2_TBDHereon", "Temperature_TBDHereon", "Salinity_TBDHereon", "Turbidity_TBDHereon", "pH_TBDHereon", "O2_TBDHereon", "SPM_mgperL", "DOC_mg.L", "TN_mg.L", "DIC_mg.L", "Silicate_mg.L", "Ammonium_mg.L", "Nitrite_mg.L", "Nitrate_mg.L", "Total_DIN_µM", "SRP_mgperL", "TotalDissolvedPhosphate_mg.L", "Ammonium_µM", "Phosphate_µM", "RespirationRate_O2ug.L.h", "POC_mgperL", "PTC_mgperL", "PTN_mgperL", "PTH_mgperL", "dCH4_nM", "Chla_ug/l", "dCO2_uM")
#metadata
metadata_METAG_FL = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/metadata_METAG_FL.RDS")
metadata_METAT_FL = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/metadata_METAT_FL.RDS")
metadata_METAG_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/metadata_METAG_LF.RDS")
metadata_METAT_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/metadata_METAT_LF.RDS")
metadata_METAG_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/metadata_METAG_HF.RDS")
metadata_METAT_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/metadata_METAT_HF.RDS")

#reformat metadata - sample names into rownames  
rownames(metadata_METAG_FL) <- metadata_METAG_FL$BioSample
rownames(metadata_METAT_FL) <- metadata_METAT_FL$BioSample
rownames(metadata_METAG_LF) <- metadata_METAG_LF$BioSample
rownames(metadata_METAT_LF) <- metadata_METAT_LF$BioSample
rownames(metadata_METAG_HF) <- metadata_METAG_HF$BioSample
rownames(metadata_METAT_HF) <- metadata_METAT_HF$BioSample
#take only numeric parameters
metadata_METAG_FL <- metadata_METAG_FL[,env_numeric]
metadata_METAT_FL <- metadata_METAT_FL[,env_numeric]
metadata_METAG_LF <- metadata_METAG_LF[,env_numeric]
metadata_METAT_LF <- metadata_METAT_LF[,env_numeric]
metadata_METAG_HF <- metadata_METAG_HF[,env_numeric]
metadata_METAT_HF <- metadata_METAT_HF[,env_numeric]
#prepare metadat for analysis
#replace all NA by average to proceed CCA
#define function
#it replaces na by mean 
#this is an approch to proceed with table having empty values
#it returns scaled (normalised) columns
avr_to_na_norm <- function(x){
  x[which(is.na(x))] <- mean(x, na.rm = T)
  x   
}

#for FL
metadata_METAG_FL_filled <- apply(metadata_METAG_FL, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_FL_filled <- as.data.frame(metadata_METAG_FL_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_FL <- apply(metadata_METAT_FL ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_FL_filled <- apply(metadata_METAT_FL, 2, avr_to_na_norm)
metadata_METAT_FL_filled <- as.data.frame(metadata_METAT_FL_filled) %>% select_if(~!any(is.nan(.))) 
#for LF
metadata_METAG_LF <- data.frame(apply(metadata_METAG_LF ,2,function(x) as.numeric(gsub(",", ".", as.character(x)))),
                   check.names=F, row.names = rownames(metadata_METAG_LF))
metadata_METAG_LF_filled <- apply(metadata_METAG_LF, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_LF_filled <- as.data.frame(metadata_METAG_LF_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_LF <- data.frame(apply(metadata_METAT_LF ,2,function(x) as.numeric(gsub(",", ".", as.character(x)))),
                   check.names=F, row.names = rownames(metadata_METAT_LF))
metadata_METAT_LF_filled <- apply(metadata_METAT_LF, 2, avr_to_na_norm)
metadata_METAT_LF_filled <- as.data.frame(metadata_METAT_LF_filled) %>% select_if(~!any(is.nan(.))) 
#HF
metadata_METAG_HF <- data.frame(apply(metadata_METAG_HF ,2,function(x) as.numeric(gsub(",", ".", as.character(x)))),
                   check.names=F, row.names = rownames(metadata_METAG_HF))
metadata_METAG_HF_filled <- apply(metadata_METAG_HF, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_HF_filled <- as.data.frame(metadata_METAG_HF_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_HF <- data.frame(apply(metadata_METAT_HF ,2,function(x) as.numeric(gsub(",", ".", as.character(x)))),
                   check.names=F, row.names = rownames(metadata_METAT_HF))
metadata_METAT_HF_filled <- apply(metadata_METAT_HF, 2, avr_to_na_norm)
metadata_METAT_HF_filled <- as.data.frame(metadata_METAT_HF_filled) %>% select_if(~!any(is.nan(.))) 
pathways <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/Elbe_viruses_distribution/VIBRANT_results_Combined_viruses/VIBRANT_AMG_pathways_Combined_viruses.tsv", sep='\t')
vir_genecluster_annot <- readRDS("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_genes/vir_genecluster_annot.RDS")
#keep only neccessery columns
vir_genecluster_annot <- vir_genecluster_annot[, c("genome", "KO", "DESCRIPTION", "gene_cluster")]
save.image(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/cca_viral_genes.RData")
```

##2. Free-living CCA 
###2.1 Gene abundance FL 
```{r CCA gene_abund FL}
#reformat viruses
#t_log_gene_abund_FL <- t(log_gene_abund_FL)
log_gene_abund_FL <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA/PCA_outputs/log_gene_abund_all_FL.RDS")
log_gene_abund_all_FL <- as.data.frame(log_gene_abund_all_FL)
count_gene_abund_FL <- exp(log_gene_abund_all_FL) #reverse log - we cannot have negative numbers for cca 
rownames(count_gene_abund_FL) <- gsub("GROS22.(1|2)_", "", rownames(count_gene_abund_FL))
rownames(count_gene_abund_FL) <- gsub("_METAG.genecount.profile", "", rownames(count_gene_abund_FL))

cca_gene_abund_FL <- cca(as.matrix(count_gene_abund_FL[which(rownames(count_gene_abund_FL) %in% rownames(metadata_METAG_FL_filled)),]), as.matrix(metadata_METAG_FL_filled))
# summary(cca_gene_abund_FL) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower
#remove original file from workspace to make space 
rm(log_gene_abund_FL)
rm(count_gene_abund_FL)
save.image(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/cca_viral_genes.RData")
```
```{r}
cc_gene_abund_FL <- cancor(as.matrix(count_gene_abund_FL[which(rownames(count_gene_abund_FL) %in% rownames(metadata_METAG_FL_filled)),]), as.matrix(metadata_METAG_FL_filled))
cc_gene_abund_FL$cor
#why correlation is one???
```
```{r}
plot(cca_gene_abund_FL)
```
```{r}
veg_1 = as.data.frame(cca_gene_abund_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_gene_abund_FL$CCA$v)
veg_2["genus"] = row.names(veg_2)
veg_2 <- merge(veg_2, vir_genecluster_annot, by.x ="row.names", by.y = "gene_cluster", all.x =T)
#reformat so is easlier to merge
pathways <- pathways %>% separate_rows(Present.AMG.KOs, sep = ",", convert = FALSE)
veg_2 <- merge(veg_2, pathways, by.x ="KO", by.y = "Present.AMG.KOs", all.x =T)

#to make picture clear - remove unclassified from visualisation
plot_metabolism = ggplot() +
  geom_point(data = veg_2[which(!is.na(veg_2$Metabolism)),], aes(x = CCA1, y = CCA2, colour = Metabolism)) #+  geom_point(data =veg_1, aes(x = CCA1, y = CCA2), color = "blue")
plot_metabolism
```
```{r}
veg_1 = as.data.frame(cca_gene_abund_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_gene_abund_FL$CCA$v)
veg_2["genus"] = row.names(veg_2)
veg_2 <- merge(veg_2, vir_genecluster_annot, by.x ="row.names", by.y = "gene_cluster", all.x =T)
#reformat so is easlier to merge
pathways <- pathways %>% separate_rows(Present.AMG.KOs, sep = ",", convert = FALSE)
veg_2 <- merge(veg_2, pathways, by.x ="KO", by.y = "Present.AMG.KOs", all.x =T)

#to make picture clear - remove unclassified from visualisation
plot_metabolism = ggplot() +
  geom_point(data = veg_2[which(veg_2$Metabolism = "Xenobiotics biodegradation and metabolism"),], aes(x = CCA1, y = CCA2, colour = Metabolism)) #+  geom_point(data =veg_1, aes(x = CCA1, y = CCA2), color = "blue")
 
plot_metabolism

```
We didnt obtain metabolic clusters - check pathways
```{r}
veg_2$PsbA <- c(veg_2$KO == "K02703")
plot_pathways = ggplot() +
  geom_point(data = veg_2[which(veg_2$PsbA==T), ], aes(x = CCA1, y = CCA2, colour = PsbA)) 
# + geom_point(data =veg_1, aes(x = CCA1, y = CCA2), color = "blue") 
plot_pathways 
#check AlaS
veg_2$AlaS <- c(veg_2$KO == "K00643")
plot_pathways = ggplot() +
  geom_point(data = veg_2[which(veg_2$Metabolism=="Amino acid metabolism"), ], aes(x = CCA1, y = CCA2, colour = AlaS)) 
# + geom_point(data =veg_1, aes(x = CCA1, y = CCA2), color = "blue") 
plot_pathways 
#TODO Xenobiotics biodegradation and metabolism

```
No interesting pattern for AlaS and PsbA genes
```{r}
#visualise all interesting pathways
selected_pathways <-c(#"Valine, leucine and isoleucine degradation",
  #"Lysine biosynthesis",
                      #"Histidine metabolism",
                      #"beta-Alanine metabolism",
                      "D-Alanine metabolism", 
                      #"Phosphonate and phosphinate metabolism",
                      "Lipoarabinomannan (LAM) biosynthesis", 
                      "Terpenoid backbone biosynthesis",
                      "Biosynthesis of enediyne antibiotics", 
                      #"Penicillin and cephalosporin biosynthesis", 
                      #"Benzoate degradation",
                      #"Chloroalkane and chloroalkene degradation") #adaptaion to antropological influence 
#Metabolism of xenobiotics by cytochrome P450",
"Photosynthesis"
)

plot_pathways = ggplot() +
  geom_point(data = veg_2[which(veg_2$Pathway %in% selected_pathways), ], aes(x = CCA1, y = CCA2, colour = Pathway)) + geom_mark_ellipse(data = veg_2[which(veg_2$Pathway %in% selected_pathways), ],aes( x = CCA1, y = CCA2,color=Pathway)) 
# + geom_point(data =veg_1, aes(x = CCA1, y = CCA2), color = "blue") 
 
plot_pathways

```

```{r}

veg_1_env_FL = veg_1[which(rownames(veg_1) %in% c( "Phosphate_µM", "TotalDissolvedPhosphate_mg.L", "Salinity_TBDHereon",  "Temperature_TBDHereon", "SRP_mgperL",
"Total_DIN_µM","pH_TBDHereon", "DIC_mg", "dCO2_uM", "dCH4_nM", "O2_TBDHereon", "Phosphate_µM")),]
#veg_1_env_FL = veg_1
FL_vir_gene_abund_plot <- plot_pathways +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(data = veg_1_env_FL,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.1, "cm")), size=0.1
  ) + #### Error here??????????
  geom_text_repel(
    data = veg_1_env_FL,
    aes(x = CCA1, y = CCA2, label = veg_1_env_FL$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))

tiff("CCA_gene_abund_FL.tiff",width = 550, height = 480, units = "px")
FL_vir_gene_abund_plot
dev.off()
```

Salinity and temperature looks prominent.

###2.2 Expression FL

```{r CCA expression FL}
#FL
log_expression_FL <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_expression_FL.RDS")
log_expression_FL <- as.data.frame(log_expression_FL)
counts_expression_FL <- exp(log_expression_FL) # transform to counts 0 positive values needed for cca
counts_expression_FL <- as.data.frame(counts_expression_FL)

rownames(counts_expression_FL) <- gsub("GROS22.(1|2)_", "", rownames(counts_expression_FL))
rownames(counts_expression_FL) <- gsub("_METAT.genecount.profile", "", rownames(counts_expression_FL))

cca_expression_FL <- cca(as.matrix(counts_expression_FL), as.matrix(metadata_METAT_FL_filled))
rm(log_expression_FL)
rm(counts_expression_FL)
#summary(cca_expression_FL) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower
save.image(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/cca_viral_genes.RData")
beep(sound = 2, expr = NULL)

```
```{r}
cc_expression_FL <- cancor(as.matrix(t_viral_expression_FL), as.matrix(metadata_METAT_FL_filled))
cc_expression_FL$cor
#why correlation is one???
```
```{r}
veg_expr_FL_1 = as.data.frame(cca_expression_FL$CCA$biplot)
veg_expr_FL_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2_expr_FL = as.data.frame(cca_expression_FL$CCA$v)
veg_2_expr_FL["genus"] = row.names(veg_2_expr_FL)
veg_2_expr_FL <- merge(veg_2_expr_FL, vir_genecluster_annot, by.x ="row.names", by.y = "gene_cluster", all.x =T)
#reformat so is easier to merge
pathways <- pathways %>% separate_rows(Present.AMG.KOs, sep = ",", convert = FALSE)
veg_2_expr_FL <- merge(veg_2_expr_FL, pathways, by.x ="KO", by.y = "Present.AMG.KOs", all.x =T)
```
```{r}
#C5-Branched dibasic acid metabolism
#Photosynthesis
#Sulfur metabolism
#Arachidonic acid metabolism
#Ether lipid metabolism
#Fatty acid degradation
#Fatty acid biosynthesis
#Cyanoamino acid metabolism
#Selenocompound metabolism
#Glutathione metabolism
#Retinol metabolism
#Terpenoid backbone biosynthesis
#Penicillin and cephalosporin biosynthesis
#Phenylpropanoid biosynthesis
#Monobactam biosynthesis
#Novobiocin biosynthesis
#Streptomycin biosynthesis
#Stilbenoid, diarylheptanoid and gingerol biosynthesis
#Xylene degradation
#Steroid degradation
#Chloroalkane and chloroalkene degradation
#Chlorocyclohexane and chlorobenzene degradation
#Fluorobenzoate degradation

plot = ggplot(data = veg_2_expr_FL[which(veg_2_expr_FL$Metabolism=="Folding, sorting and degradation" &  veg_2_expr_FL$Total.AMGs > 10),]) +
  geom_point( aes(x = CCA1, y = CCA2, colour = Pathway)) + geom_mark_ellipse(aes(x = CCA1, y = CCA2, color=Pathway)) # + theme(legend.position="none")
#+ geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot 
```
Explore pathways
```{r}
#
selected_pathways <-c(#"C5-Branched dibasic acid metabolism",
#"Photosynthesis",
"Ether lipid metabolism",
"Fatty acid degradation",
"Fatty acid biosynthesis",
"Cyanoamino acid metabolism",
"Selenocompound metabolism",
"Terpenoid backbone biosynthesis",
"Penicillin and cephalosporin biosynthesis",
"Phenylpropanoid biosynthesis",
"Novobiocin biosynthesis",
"Stilbenoid, diarylheptanoid and gingerol biosynthesis",
"Xylene degradation",
"Chloroalkane and chloroalkene degradation",
"Fluorobenzoate degradation")
# plot = ggplot() +
#   geom_point(data = veg_2_expr_FL[which(veg_2_expr_FL$Pathway %in% selected_pathways[1:5]),], aes(x = CCA1, y = CCA2, colour = Pathway))
FL_vir_expr_pathways_plot = ggplot(data = veg_2_expr_FL[which(veg_2_expr_FL$Pathway %in% selected_pathways),]) +
  geom_point(aes(x = CCA1, y = CCA2, colour = Pathway))# + geom_mark_ellipse(aes(x = CCA1, y = CCA2, color=Pathway)) # +
plot
beep(sound = 1, expr = NULL)
```
Not informative - focus on gene abundance in the othe fraction
```{r}
#first general expression plot 
veg_1 = veg_expr_FL_1[which(rownames(veg_expr_FL_1) %in% c( "Phosphate_µM", "Salinity_TBDHereon",  "Temperature_TBDHereon", "Total_DIN_µM","pH_TBDHereon", "DIC_mg", "dCO2_uM",  "O2_TBDHereon", "Phosphate_µM")),]

FL_vir_expr_pathways_cca_plot <- FL_vir_expr_pathways_plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=0.05
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) + ggtitle("Free-living expression")
tiff("CCA_expr_FL.tiff",width = 550, height = 480, units = "px")
FL_vir_expr_pathways_cca_plot
dev.off()
```
```{r}
FL_vir_expr_plot = ggplot() +
  geom_point(data = veg_2_expr_FL[which(!is.na(veg_2_expr_FL$Metabolism)),], aes(x = CCA1, y = CCA2, colour = Metabolism)) #+ geom_mark_ellipse(aes(data = veg_2_expr_FL, x = CCA1, y = CCA2, color=Pathway)) # +
FL_vir_expr_plot
beep(sound = 1, expr = NULL)
```
```{r}
#first general expression plot 
veg_1 = veg_expr_FL_1[which(rownames(veg_expr_FL_1) %in% c( "Sat_O2_TBDHereon", "Temperature_TBDHereon", "Salinity_TBDHereon", "pH_TBDHereon", "O2_TBDHereon", "DOC_mg.L",  "TN_mg.L", "DIC_mg.L", "Silicate_mg.L", "Ammonium_mg.L", "Nitrite_mg.L", "SRP_mgperL",  "Chla_ug/l", "dCO2_uM")),]

FL_vir_expr_cca_plot <- FL_vir_expr_plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=0.05
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
FL_vir_expr_cca_plot
```

## 3. Light fraction CCA
### 3.1 Gene abundance LF  
```{r CCA gene_abund LF}
#reformat viruses
log_gene_abund_LF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_gene_abund_all_LF.RDS")
log_gene_abund_LF <- as.data.frame(log_gene_abund_LF)
count_gene_abund_LF <- exp(log_gene_abund_LF) #reverse log - we cannot have negative numbers for cca 
rownames(count_gene_abund_LF) <- gsub("GROS22.(1|2)_", "", rownames(count_gene_abund_LF))
rownames(count_gene_abund_LF) <- gsub("_METAG.genecount.profile", "", rownames(count_gene_abund_LF))
cca_gene_abund_LF <- cca(as.matrix(count_gene_abund_LF[which(rownames(count_gene_abund_LF) %in% rownames(metadata_METAG_LF_filled)),]), as.matrix(metadata_METAG_LF_filled))
# summary(cca_gene_abund_FL) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower
#remove original file from workspace to make space 
rm(log_gene_abund_LF)
rm(count_gene_abund_LF)
save.image(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/cca_viral_genes.RData")
```

```{r}
cc_gene_abund_LF <- cancor(as.matrix(t_viral_gene_abund_LF), as.matrix(metadata_METAG_LF_filled))
cc_gene_abund_LF$cor
#why correlation is one???
```
Explore different pathways - find those with clear pattern - direction
```{r}
veg_1 = as.data.frame(cca_gene_abund_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2_gene_LF = as.data.frame(cca_gene_abund_LF$CCA$v)
veg_2_gene_LF["genus"] = row.names(veg_2_gene_LF)
veg_2_gene_LF <- merge(veg_2_gene_LF, vir_genecluster_annot, by.x ="row.names", by.y = "gene_cluster", all.x =T)
#reformat so is easlier to merge
pathways <- pathways %>% separate_rows(Present.AMG.KOs, sep = ",", convert = FALSE)
veg_2_gene_LF <- merge(veg_2_gene_LF, pathways, by.x ="KO", by.y = "Present.AMG.KOs", all.x =T)

plot = ggplot(data = veg_2_gene_LF[which(veg_2_gene_LF$Metabolism=="Folding, sorting and degradation" &  veg_2_gene_LF$Total.AMGs > 10),]) +
  geom_point( aes(x = CCA1, y = CCA2, colour = Pathway)) + geom_mark_ellipse(aes(x = CCA1, y = CCA2, color=Pathway)) # + theme(legend.position="none")
#+ geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot 
beep(sound = 1, expr = NULL)
```
```{r}
#
selected_pathways <-c(#"Steroid hormone biosynthesis",
  "Synthesis and degradation of ketone bodies",
  "Photosynthesis",
  "Oxidative phosphorylation")
plot_gene_LF = ggplot() +
  geom_point(data = veg_2_gene_LF[which(veg_2_gene_LF$Pathway %in% selected_pathways),], aes(x = CCA1, y = CCA2, colour = Pathway)) + geom_mark_ellipse(data = veg_2_gene_LF[which(veg_2_gene_LF$Pathway %in% selected_pathways),], aes(x = CCA1, y = CCA2, color=Pathway))
plot_gene_LF
#beep(sound = 1, expr = NULL)
```

```{r}
veg_LF_1 = as.data.frame(cca_gene_abund_LF$CCA$biplot)
veg_LF_1["env"] = row.names(veg_LF_1)

#extracting the data; genusv
veg_LF_2 = as.data.frame(cca_gene_abund_LF$CCA$v)
veg_LF_2["genus"] = row.names(veg_LF_2)

# #explore data to find pathways which makes clusters
# plot = ggplot() +
#   geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
#   geom_point(data =
#                veg_1, aes(x = CCA1, y = CCA2), color = "blue")
# 
# plot
```
```{r}
vec_param<- c("Sat_O2_TBDHereon", "Temperature_TBDHereon", "Salinity_TBDHereon",  "pH_TBDHereon", "DOC_mg.L", "TN_mg.L", "Total_DIN_µM", "SRP_mgperL", "TotalDissolvedPhosphate_mg.L",  "Phosphate_µM",  "PTC_mgperL", "PTN_mgperL", "dCH4_nM", "dCO2_uM")     
veg_LF_1 = veg_1[which(rownames(veg_1) %in% vec_param),]
veg_1 = veg_LF_1
LF_vir_gene_abund_plot  <- plot_gene_LF +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=0.05
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("CCA_gene_abund_LF.tiff",width = 550, height = 480, units = "px")
LF_vir_gene_abund_plot 
dev.off()

```

Salinity and temperature looks prominent.

###2.2 Expression LF

```{r CCA expression FL}
#FL
log_expression_LF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_expression_LF.RDS")
log_expression_LF <- as.data.frame(log_expression_LF)
counts_expression_LF <- exp(log_expression_LF) #add absolute value of minimum to get positive values - not sure if it is correct 

counts_expression_LF <- as.data.frame(counts_expression_LF)

rownames(counts_expression_LF) <- gsub("GROS22.(1|2)_", "", rownames(counts_expression_LF))
rownames(counts_expression_LF) <- gsub("_METAT.genecount.profile", "", rownames(counts_expression_LF))

cca_expression_LF <- cca(as.matrix(counts_expression_LF), as.matrix(metadata_METAT_LF_filled))
rm(log_expression_LF)
rm(counts_expression_LF)
#summary(cca_expression_LF) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower
save.image(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/cca_viral_genes.RData")
beep(sound = 2, expr = NULL)

```
```{r}
cc_expression_LF <- cancor(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
cc_expression_LF$cor
#why correlation is one???
```
```{r}
veg_1_expr_LF = as.data.frame(cca_expression_LF$CCA$biplot)
veg_1_expr_LF["env"] = row.names(veg_1)

#extracting the data; genus
veg_2_expr_LF = as.data.frame(cca_expression_LF$CCA$v)
veg_2_expr_LF["genus"] = row.names(veg_2_expr_LF)
veg_2_expr_LF <- merge(veg_2_expr_LF, vir_genecluster_annot, by.x ="row.names", by.y = "gene_cluster", all.x =T)
#reformat so is easlier to merge
pathways <- pathways %>% separate_rows(Present.AMG.KOs, sep = ",", convert = FALSE)
veg_2_expr_LF <- merge(veg_2_expr_LF, pathways, by.x ="KO", by.y = "Present.AMG.KOs", all.x =T)

```

```{r}
#Steroid hormone biosynthesis
#Ether lipid metabolism
#Glycerolipid metabolism
#Fatty acid degradation
#Amino acid metabolism
#Amino acid metabolism


plot = ggplot(data = veg_2_expr_LF[which(veg_2_expr_LF$Metabolism=="Xenobiotics biodegradation and metabolism" &  veg_2_expr_LF$Total.AMGs > 10),]) +
  geom_point(aes(x = CCA1, y = CCA2, colour = Pathway)) + geom_mark_ellipse(aes(x = CCA1, y = CCA2, color=Pathway)) # + theme(legend.position="none")
#+ geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot 
beep(sound = 1, expr = NULL)
```
Try to visualise interesting KO

#
```{r}
#Steroid hormone biosynthesis
#Ether lipid metabolism
#Glycerolipid metabolism
#Fatty acid degradation
#Amino acid metabolism
#Amino acid metabolism

veg_2_expr_LF_known_metabolism <- veg_2_expr_LF[which(!is.na(veg_2_expr_LF$Metabolism)),]
veg_2_expr_LF_known_metabolism$interesting_genes <- rep(c("other"), time = length(veg_2_expr_LF_known_metabolism[,1]))
#label
veg_2_expr_LF_known_metabolism[which(veg_2_expr_LF_known_metabolism$KO == "K02703"),"interesting_genes"] <-"1PsbA"
#label
veg_2_expr_LF_known_metabolism[which(veg_2_expr_LF_known_metabolism$KO == "K00643"),"interesting_genes"] <-"2AlaS"
veg_2_expr_LF_known_metabolism <- veg_2_expr_LF_known_metabolism %>% arrange(desc(interesting_genes))
#opacity <- ifelse(veg_2_expr_LF_known_metabolism$interesting_genes == "other", 0, 1)
plot = ggplot(data = veg_2_expr_LF_known_metabolism) +
  geom_point(aes(x = CCA1, y = CCA2, colour = veg_2_expr_LF_known_metabolism$interesting_genes)) + geom_mark_ellipse(aes(x = CCA1, y = CCA2, color= veg_2_expr_LF_known_metabolism$interesting_genes)) # + theme(legend.position="none")
#+ geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot 
beep(sound = 1, expr = NULL)
```

Maybe AlaS has a pattern
```{r}


vec_param <- c("Nitrate_mg.L", "RespirationRate_O2ug.L.h" , "Chla_ug/l", "PTH_mgperL", "dCO2_uM"," SPM_mgperL")
veg_1 = veg_1_expr_LF[which(rownames(veg_1_expr_LF) %in% vec_param),]

LF_vir_expr_plot <- plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=0.05
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +scale_color_manual(values=c("red", "blue","yellow")) + theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18)) + ggtitle("Light fraction expression of PsbA and AlaS") + guides(colour=guide_legend(title="Genes"))
LF_vir_expr_plot <- LF_vir_expr_plot + guides(colour=guide_legend(title="Genes"))
tiff("LF_vir_expr_plot_AlaS_PsbA.tiff")
LF_vir_expr_plot 
dev.off()
```
##4. Heavy fraction CCA
### 4.1. Gene abundance HF
```{r CCA gene_abund LF}
#reformat viruses
log_gene_abund_HF <- readRDS(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/log_gene_abund_all_HF.RDS")
log_gene_abund_HF <- as.data.frame(log_gene_abund_HF)
count_gene_abund_HF <- exp(log_gene_abund_HF) #reverse log - we cannot have negative numbers for cca 
rownames(count_gene_abund_HF) <- gsub("GROS22.(1|2)_", "", rownames(count_gene_abund_HF))
rownames(count_gene_abund_HF) <- gsub("_METAG.genecount.profile", "", rownames(count_gene_abund_HF))
cca_gene_abund_HF <- cca(as.matrix(count_gene_abund_HF[which(rownames(count_gene_abund_HF) %in% rownames(metadata_METAG_HF_filled)),]), as.matrix(metadata_METAG_HF_filled))
# summary(cca_gene_abund_LF) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower
#remove original file from workspace to make space 
rm(log_gene_abund_HF)
rm(count_gene_abund_HF)
save.image(file="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/PCA_outputs/cca_viral_genes.RData")
```

```{r}
cc_gene_abund_HF <- cancor(as.matrix(t_viral_gene_abund_HF), as.matrix(metadata_METAG_HF_filled))
cc_gene_abund_HF$cor
#why correlation is one???
```
Explore different pathways - find those with clear pattern - direction
```{r}
veg_1 = as.data.frame(cca_gene_abund_HF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2_gene_HF = as.data.frame(cca_gene_abund_HF$CCA$v)
veg_2_gene_HF["genus"] = row.names(veg_2_gene_HF)
veg_2_gene_HF <- merge(veg_2_gene_HF, vir_genecluster_annot, by.x ="row.names", by.y = "gene_cluster", all.x =T)
#reformat so is easlier to merge
pathways <- pathways %>% separate_rows(Present.AMG.KOs, sep = ",", convert = FALSE)
veg_2_gene_HF <- merge(veg_2_gene_HF, pathways, by.x ="KO", by.y = "Present.AMG.KOs", all.x =T)
```
```{r}

selected_pathways <-c("Photosynthesis", 
                       "Steroid hormone biosynthesis", 
                      "D-Arginine and D-ornithine metabolism",  "Lipoarabinomannan (LAM) biosynthesis",  "Geraniol degradation", "Carotenoid biosynthesis")


plot_gene_HF = ggplot(data = veg_2_gene_HF[which(veg_2_gene_HF$Pathway %in% selected_pathways),]) +
  geom_point(aes(x = CCA1, y = CCA2, colour = Pathway)) + geom_mark_ellipse(aes(x = CCA1, y = CCA2, color=Pathway))
plot_gene_HF
#beep(sound = 1, expr = NULL)
```

```{r}
veg_HF_1 = as.data.frame(cca_gene_abund_HF$CCA$biplot)
veg_HF_1["env"] = row.names(veg_HF_1)

#extracting the data; genusv
veg_HF_2 = as.data.frame(cca_gene_abund_HF$CCA$v)
veg_HF_2["genus"] = row.names(veg_HF_2)

# #explore data to find pathways which makes clusters
# plot = ggplot() +
#   geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
#   geom_point(data =
#                veg_1, aes(x = CCA1, y = CCA2), color = "blue")
# 
# plot
```
```{r}
vec_param<- c("Temperature_TBDHereon",  "DOC_mg.L", "TN_mg.L", "Total_DIN_µM", "SRP_mgperL",  "Phosphate_µM",  "PTC_mgperL", "Nitrate_mg.L", "dCO2_uM")     
veg_HF_1 = veg_1[which(rownames(veg_1) %in% vec_param),]
HF_vir_gene_abund_plot  <- plot_gene_HF +
  theme_bw() +
  geom_segment(
    data = veg_HF_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=0.05
  ) +
  geom_text_repel(
    data = veg_HF_1,
    aes(x = CCA1, y = CCA2, label = veg_HF_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("CCA_gene_abund_HF.tiff",width = 550, height = 480, units = "px")
HF_vir_gene_abund_plot 
dev.off()

```

Salinity and temperature looks prominent.

###2.2 Expression HF

```{r CCA expression HF}
#reformat viruses
t_viral_expression_LF <- t(viral_expression_LF)
t_viral_expression_LF <- as.data.frame(t_viral_expression_LF)

rownames(t_viral_expression_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_LF))
rownames(t_viral_expression_LF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_LF))

cca_expression_LF <- cca(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
summary(cca_expression_LF) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower

```
```{r}
cc_expression_LF <- cancor(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
cc_expression_LF$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("TotalDissolvedPhosphate_mg.L
", "Temperature_TBDHereon
0", "pH_TBDHereon", "Salinity_TBDHereon")),]
plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```