---
title: "CCA"
author: "Julia Golebiowska"
date: "2024-10-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
CCA  - Canonical correspondence analysis on viral data from the Elbe estuary for each fraction and and data type - viral abundance and expression based on individual marker genes. This analysis revels the potential drivers of communities diversity and activity. 



```{r test}
library(vegan)
library(ggplot2)
library(ggrepel)
library(dplyr)
library(tidyverse)
library(ggforce)
library(stringr)
library(RColorBrewer)

setwd("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CCA_marker_gene")
```

## Viral diversity 

##1. Read taxonomy data and unify within clusters 


```{r , echo=FALSE}
#load marker genes
geneabund_drep_marker <- readRDS("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/vibrant_annot_drep_marker_abund.RDS")
#remove  repeating rows - I dont need host taxonomy 
# next time generate special object with viral taxonomy and salinity niche classification
geneabund_drep_marker <- geneabund_drep_marker %>% select(contains("GROS")) %>% distinct()
#drep clusters 
drep_clusters <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Cdb.csv", header = TRUE, sep = ",")
#taxonomy
drep_clusters[,"genome"] = sub(".fna", "", drep_clusters[,"genome"])
proviruses_taxonomy <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/Elbe_viruses_distribution/genomad_proviruses_output/Combined_proviruses_annotate/Combined_proviruses_taxonomy.tsv", sep = "\t")
viruses_taxonomy <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/Elbe_viruses_distribution/genomad_viruses_output/Combined_viruses_taxonomy.tsv", sep = "\t")
#analyse if similar taxonomy for each cluster
drep_clusters <- left_join(drep_clusters, viruses_taxonomy, by=c("genome"="seq_name"))
drep_clusters %>% group_by(primary_cluster) %>% summarise(count_genomes= n_distinct(genome), count_lineages = n_distinct(lineage))  %>%  filter(count_lineages > 1) 
#only 46 have more then 1 lineage - even when more lineages they agree at some point 
#lets unify those 
#keep lineages which dominates by more then 50%
#if less then 50% go to upper taxa level
#continue until Viruses - as we analyse only viruses we will put -"other" label to all analysis
#for this purpose separate into columns
drep_clusters <- drep_clusters %>% 
  separate(lineage, c("level_1", "level_2", "level_3", "level_4", "level_5",  "level_6", "level_7"), ";")
drep_clusters[drep_clusters == ""] <- NA
columns_to_keep = c("primary_cluster", "genome", "level_1", "level_2", "level_3", "level_4", "level_5",  "level_6", "level_7")
#primary_cluster, genome, level_1, level_2, level_3, level_4, level_5, level_6, level_7
drep_clusters_fin_taxonomy <- drep_clusters %>%  add_count(primary_cluster, level_7,  name = "level_7_percentage", wt = !is.na(level_7))  %>% ungroup() %>%
  add_count(primary_cluster, level_6, wt = !is.na(level_6), name = "level_6_percentage") %>% ungroup() %>%
  add_count(primary_cluster, level_5, wt = !is.na(level_5), name = "level_5_percentage") %>% ungroup() %>%
  add_count(primary_cluster, level_4, wt = !is.na(level_4), name = "level_4_percentage") %>% ungroup() %>%
  add_count(primary_cluster, level_3, wt = !is.na(level_3), name = "level_3_percentage") %>% ungroup() %>% 
  add_count(primary_cluster, level_2, wt = !is.na(level_2), name = "level_2_percentage") %>% ungroup() %>%
  add_count(primary_cluster, level_1, wt = !is.na(level_1), name = "level_1_percentage") %>% mutate(level_2_percentage = level_2_percentage/level_1_percentage, level_3_percentage = level_3_percentage/level_1_percentage, level_4_percentage = level_4_percentage/level_1_percentage, level_5_percentage = level_5_percentage/level_1_percentage, level_6_percentage = level_6_percentage/level_1_percentage, level_7_percentage = level_7_percentage/level_1_percentage, level_1_percentage = level_1_percentage/level_1_percentage)  
#unify taxon according to pseudocode above
drep_clusters_fin_taxonomy_unified <- drep_clusters_fin_taxonomy %>% group_by(primary_cluster) %>%
  filter(if(any(level_7_percentage  > 0.5,  na.rm = T)) level_7_percentage > 0.5 else TRUE ) %>% ungroup() %>% group_by(primary_cluster) %>% 
  filter(if(any(level_6_percentage  > 0.5,  na.rm = T)) level_6_percentage > 0.5 else TRUE ) %>% ungroup() %>% group_by(primary_cluster) %>%
  filter(if(any(level_5_percentage  > 0.5,  na.rm = T)) level_5_percentage > 0.5 else TRUE ) %>% ungroup() %>% group_by(primary_cluster) %>%
  filter(if(any(level_4_percentage  > 0.5,  na.rm = T)) level_4_percentage > 0.5 else TRUE ) %>% ungroup() %>% group_by(primary_cluster) %>%
  filter(if(any(level_3_percentage  > 0.5,  na.rm = T)) level_3_percentage > 0.5 else TRUE ) %>% ungroup() %>% group_by(primary_cluster) %>%
  filter(if(any(level_2_percentage  > 0.5,  na.rm = T)) level_2_percentage > 0.5 else TRUE ) %>% ungroup() %>% group_by(primary_cluster) %>% 
  filter(if(any(level_1_percentage  > 0.5,  na.rm = T)) level_1_percentage > 0.5 else TRUE ) %>% ungroup() %>% select(-c("genome","secondary_cluster","threshold", "cluster_method", "comparison_algorithm", "n_genes_with_taxonomy", "agreement", "taxid")) %>% distinct()
  
drep_clusters_fin_taxonomy_unified <- drep_clusters_fin_taxonomy_unified %>% rowwise() %>% mutate(lineage=case_when(level_7_percentage > 0.5 ~ str_c(c(level_1, level_2, level_3, level_4, level_5, level_6, level_7), collapse=";"), level_6_percentage > 0.5 ~ str_c(c(level_1, level_2, level_3, level_4, level_5, level_6), collapse=";"), level_5_percentage > 0.5 ~ str_c(c(level_1, level_2, level_3, level_4, level_5), collapse=";"), level_4_percentage > 0.5 ~ str_c(c(level_1, level_2, level_3, level_4), collapse=";"),  level_3_percentage > 0.5 ~ str_c(c(level_1, level_2, level_3), collapse=";"), level_2_percentage > 0.5 ~ str_c(c(level_1, level_2), collapse=";"),  level_1_percentage > 0.5  ~ level_1))

#unite lineage and remove repeating rows
drep_clusters_fin_taxonomy_unified = drep_clusters_fin_taxonomy_unified %>% 
  separate("lineage", c("level_1", "level_2", "level_3", "level_4", "level_5",  "level_6", "level_7"), sep = ";", remove = FALSE) %>% select(c("primary_cluster", "level_1", "level_2", "level_3", "level_4", "level_5", "level_6", "level_7", "lineage")) %>% distinct()
#test if we have one line per cluster
drep_clusters_fin_taxonomy_unified %>% group_by(primary_cluster) %>% summarise(n=n()) %>% arrange(desc(n))  
#good
#add marker genes counts
geneabund_drep_marker_taxonomy <- left_join(geneabund_drep_marker, drep_clusters_fin_taxonomy_unified, by = "primary_cluster")
#make a taxonomy boxplot
#read metadata 
path = "C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/metadata/"
metadata <- read.csv(paste0(path,"/PhysicochemicalParameters_mod3.csv"), header=TRUE, sep=",", row.names=1) 
metadata$Station <- factor(metadata$Station, levels = c("BunthausSpitze", "Muhlenberger Loch","Twielenfleth", "Schwarztonnensand", "Brunsbuttel","Meedem Grund"))

#sum by lineage and add metadata - but first make it long format 
geneabund_drep_marker_taxonomy_long <- geneabund_drep_marker_taxonomy %>% pivot_longer(cols = starts_with("GROS"), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% group_by(lineage, sampleid, level_5, primary_cluster) %>% summarise(counts=sum(counts)) %>% filter(counts>0) %>% left_join(metadata)
# %>% group_by(sampleid, level_5) %>% summarise(counts=sum(counts), level_5, primary_cluster) 
# %>% ungroup() %>% group_by(lineage, sampleid, level_5, primary_cluster) %>% summarise(counts=sum(counts)) 
#add poly- meso- oligo - haline labels
geneabund_drep_marker_taxonomy_long <- geneabund_drep_marker_taxonomy_long %>% mutate(Salinity_level = case_when(Salinity_PSU <= 5  ~ 'oligohaline', Salinity_PSU > 5 & Salinity_PSU <= 18  ~ 'mesohaline', Salinity_PSU > 18  ~ 'polyhaline'))
#remove stations  - counts without metadata 
#think about finding metadata for them 
geneabund_drep_marker_taxonomy_long <- geneabund_drep_marker_taxonomy_long %>% filter(!is.na(Station))
#change NA to unknown label for plotting
geneabund_drep_marker_taxonomy_long <- geneabund_drep_marker_taxonomy_long %>% mutate(level_5 = replace_na(level_5, "unknown"))
saveRDS(geneabund_drep_marker_taxonomy_long, file ="geneabund_drep_marker_taxonomy_long.RDS")
#box plot
#simplified lineages for plotting

 tiff("Relative_abundace_taxonomy_across_different_seasons.tiff", unit="px", width = 1000, height = 1000)
ggplot(geneabund_drep_marker_taxonomy_long) + geom_bar(position ="fill", stat = "identity", aes(x=Station, y=counts, fill = level_5)) + scale_y_continuous(labels=scales::percent_format()) + facet_wrap(~data_type *Sample_date) + ggtitle("Relative abundace across different seasons") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=10), axis.title.x = element_blank(), axis.text.x = element_text(angle=90, hjust=1, size=10)) + guides(fill=guide_legend(title="Viral class")) 
dev.off()
 tiff("Relative_abundace_taxonomy_across_different_fraction.tiff", unit="px", width = 1000, height = 1000)
ggplot(geneabund_drep_marker_taxonomy_long) + geom_bar(position ="fill", stat = "identity", aes(x=Station, y=counts, fill = level_5)) + scale_y_continuous(labels=scales::percent_format()) + facet_wrap(~data_type *Sample_type) + ggtitle("Relative abundace across different fractions")+ ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), axis.title.x = element_blank(), axis.text.x = element_text(angle=90, hjust=1, size=12)) + guides(fill=guide_legend(title="Viral class")) 
dev.off()

viruses_taxonomy_salinity <- ggplot(geneabund_drep_marker_taxonomy_long) + geom_bar(position ="fill", stat = "identity", aes(x=sampleid, y=counts, fill = level_5)) + scale_y_continuous(labels=scales::percent_format()) + facet_wrap(~data_type *Salinity_level, scales= "free") + ggtitle("Relative abundace across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), axis.text.x=element_blank()) + guides(fill=guide_legend(title="Viral class")) + scale_x_discrete(labels = NULL, breaks = NULL) 

tiff("Relative_abundace_taxonomy_across_different_salinity.tiff", unit="px", width = 1000, height = 1000)
viruses_taxonomy_salinity
dev.off()
```
## 3. Characterise clusters by salinity preferences and fractions


```{r}
drep_clusters_fin_taxonomy <- drep_clusters_fin_taxonomy %>% mutate(AccessionNumber_TBDSven = str_extract(genome, "SAMEA[0-9]+"))
drep_clusters_fin_taxonomy_description <- merge(drep_clusters_fin_taxonomy, distinct(geneabund_drep_marker_taxonomy_long[, c("AccessionNumber_TBDSven", "Sample_type", "Salinity_level")]), all.x = T, all.y = F, by="AccessionNumber_TBDSven") 
find_niche <-function(x, threshold){
  #calculate rel frequencies
  tab = prop.table(table(x))
  #check id=f there are values >= 0.8
  #then take this value
  if(any(tab>=threshold)){
    names(tab[which(tab>=threshold)])
  }
  #else concatinate values which are higher the 0.2
  else {
    paste0(names(tab[which(tab>= 0.2)]), collapse = ";")
  }}
#statistics for each cluster
drep_clusters_fin_taxonomy_description <- 
  drep_clusters_fin_taxonomy_description %>% 
  add_count(primary_cluster,  name = "count_genomes", wt = !is.na(genome)) %>% ungroup() %>% 
  add_count(primary_cluster, Salinity_level,  name = "percentage_salinity", wt = !is.na(Salinity_level)) %>% ungroup() %>%
  mutate(percentage_salinity = (percentage_salinity/count_genomes)*100) %>% ungroup() %>% 
  add_count(primary_cluster, Sample_type,  name = "percentage_fraction", wt = !is.na(Sample_type)) %>% ungroup()  %>% 
  mutate(percentage_fraction = (percentage_fraction/count_genomes)*100) %>% ungroup() %>% group_by(primary_cluster) %>% mutate(Salinity_niche= find_niche(Salinity_level, 0.8), Fraction_niche= find_niche(Sample_type, 0.8)) %>% ungroup()
#add data to dataframe with primary clusters only to have all calculated taxonomy, 
drep_clusters_fin_taxonomy_unified <- merge(drep_clusters_fin_taxonomy_unified, distinct(drep_clusters_fin_taxonomy_description[, c("count_genomes", "Salinity_niche", "Fraction_niche", "primary_cluster")]), all.x = T, all.y = F, by="primary_cluster") 
#ok with prepared table lets do the basic statistics for clster which are bigger then 5 
drep_clusters_fin_taxonomy_unified$Salinity_niche <- as.character(drep_clusters_fin_taxonomy_unified$Salinity_niche)
drep_clusters_fin_taxonomy_unified$Fraction_niche <- as.character(drep_clusters_fin_taxonomy_unified$Fraction_niche)
#check correlation between salinity and fraction niche  
corr_spearman <- cor(x=rank(drep_clusters_fin_taxonomy_unified$Salinity_niche), y=rank(drep_clusters_fin_taxonomy_unified$Fraction_niche), method = 'spearman')
#0.2 weak positive corelation
#for statistic choose clusters size of more then 4
drep_clusters_fin_taxonomy_unified %>% filter(count_genomes>4) %>% summary() 
drep_clusters_fin_taxonomy_unified_stat <- drep_clusters_fin_taxonomy_unified %>% filter(count_genomes>4)
drep_clusters_fin_taxonomy_unified %>% filter(count_genomes>4) %>% summary() 
#do venn diagrams
library(VennDiagram)
 
# Generate 3 sets of 200 words
#debug
oligo <- drep_clusters_fin_taxonomy_unified_stat$primary_cluster[which(grepl("oligo", drep_clusters_fin_taxonomy_unified_stat$Salinity_niche))]
meso <- drep_clusters_fin_taxonomy_unified_stat$primary_cluster[which(grepl("meso", drep_clusters_fin_taxonomy_unified_stat$Salinity_niche))]
poly <- drep_clusters_fin_taxonomy_unified_stat$primary_cluster[which(grepl("poly", drep_clusters_fin_taxonomy_unified_stat$Salinity_niche))]

 
# Chart
venn.diagram(
  x = list(oligo, meso, poly),
  category.names = c("oligohaline" , "mesohaline" , "polyhaline"),
  filename = 'salinity_venn_diagramm.png',
  output=TRUE
)
#most viral clusters have specificity towards salinity. The least in polyhaline only but these samples has the lowest frequencies. The intersections with mesohaline for both polyhaline and oligohaline has also high values as - mesohaine water are neigbouring for both of these conditions. Interesting there is no intersection for polyhaline and oligohaline water and very few for all three categories what strongly suggest seperate viral saline niches. 
###for fraction
# Generate 3 sets of 200 words
#debug
Free_living <- drep_clusters_fin_taxonomy_unified_stat$primary_cluster[which(grepl("Free", drep_clusters_fin_taxonomy_unified_stat$Fraction_niche))]
Light_fraction <- drep_clusters_fin_taxonomy_unified_stat$primary_cluster[which(grepl("Light", drep_clusters_fin_taxonomy_unified_stat$Fraction_niche))]
Heavy_fraction <- drep_clusters_fin_taxonomy_unified_stat$primary_cluster[which(grepl("Heavy", drep_clusters_fin_taxonomy_unified_stat$Fraction_niche))]
 
# Chart
venn.diagram(
  x = list(Free_living, Light_fraction, Heavy_fraction),
  category.names = c("Free-living" , "Light fraction" , "Heavyfraction"),
  filename = 'fraction_venn_diagramm.png',
  output=TRUE
)
#In contrary for fractions there is no such specificity. Most clusters are present in all fraction or as a second category in particulate fractions

```
```{r}
#add niche description
#and do plots
geneabund_drep_marker_taxonomy_long <- geneabund_drep_marker_taxonomy_long %>% merge(drep_clusters_fin_taxonomy_description[,c("primary_cluster", "Salinity_niche", "Fraction_niche")], all.x=T, by="primary_cluster")
Salinity_niche_taxa_plot <- ggplot(na.omit(geneabund_drep_marker_taxonomy_long[, c("sampleid", "counts", "level_5", "data_type", "Salinity_niche")])) + geom_bar(position ="fill", stat = "identity", aes(x=sampleid, y=counts, fill = level_5)) + facet_wrap(~data_type *Salinity_niche, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_x_discrete(labels = NULL, breaks = NULL) + ggtitle("Relative abundace across different salinity niches") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) + guides(fill=guide_legend(title="Viral class"))
tiff("Salinity_niche_taxa_plot.tiff", unit="px", width = 1000, height = 1000)
Salinity_niche_taxa_plot
dev.off()
#facet_wrap(~data_type *Salinity_niche, scales="free")
Fraction_niche_taxa_plot <- ggplot(na.omit(geneabund_drep_marker_taxonomy_long[, c("sampleid", "counts", "level_5", "data_type", "Fraction_niche")])) + geom_bar(position ="fill", stat = "identity", aes(x=sampleid, y=counts, fill = level_5)) + facet_wrap(~data_type *Fraction_niche, scales= "free_x") + scale_x_discrete(labels = NULL, breaks = NULL) + scale_y_continuous(labels=scales::percent_format())  + ggtitle("Relative abundace across different fraction niches") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) + guides(fill=guide_legend(title="Viral class")) 
tiff("Fraction_niche_taxa_plot.tiff", unit="px", width = 1000, height = 1000)
Fraction_niche_taxa_plot
dev.off()
```
## 2.Read metadata for each fraction and data type
```{r}
#no as much differences in taxonomy as expected - taking into account PCA of genomic potential viral diversity -
#but there are things to corre -NA  from metadata and different colours for level 3/4 
#FL
viral_expression_FL = readRDS( file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/viral_expression_FL.RDS")
metadata_METAG_FL = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/metadata_METAG_FL.RDS")
metadata_METAT_FL = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/metadata_METAT_FL.RDS")

colnames_FL <- colnames(geneabund_drep_marker)[which(grepl(paste(metadata_METAG_FL$AccessionNumber_TBDSven, collapse="|"), colnames(geneabund_drep_marker)))]
#lets work with data frame as wll be easlier to 
geneabund_drep_marker <- as.data.frame(geneabund_drep_marker)
#set rownames as primary cluster to track particular viral cluster in cca plots
rownames(geneabund_drep_marker) <- geneabund_drep_marker$primary_cluster
colnames_METAG_FL <- colnames_FL[grep("METAG", colnames_FL)]
#keep rownames
viral_diversity_FL <- geneabund_drep_marker[,colnames_METAG_FL]
colnames_METAT_FL <- colnames_FL[grep("METAT", colnames_FL)]
viral_activity_FL <- geneabund_drep_marker[,colnames_METAT_FL] 
#LF

metadata_METAG_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/metadata_METAG_LF.RDS")
metadata_METAT_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/metadata_METAT_LF.RDS")
colnames_LF <- colnames(geneabund_drep_marker)[which(grepl(paste(metadata_METAG_LF$AccessionNumber_TBDSven, collapse="|"), colnames(geneabund_drep_marker)))]
colnames_METAG_LF <- colnames_LF[which(grepl("METAG", colnames_LF))]
#remove na fromm colnames_METAG_LF
viral_diversity_LF <- geneabund_drep_marker[,colnames_METAG_LF] 
colnames_METAT_LF <- colnames_LF[which(grepl("METAT", colnames_LF))]
viral_activity_LF <- geneabund_drep_marker[,colnames_METAT_LF] 
#HF
viral_expression_HF = readRDS( file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/viral_expression_HF.RDS")
metadata_METAG_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/metadata_METAG_HF.RDS")
metadata_METAT_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/mantel_test/metadata_METAT_HF.RDS")
colnames_HF <- colnames(geneabund_drep_marker)[which(grepl(paste(metadata_METAG_HF$AccessionNumber_TBDSven, collapse="|"), colnames(geneabund_drep_marker)))]
colnames_METAG_HF <- colnames_HF[which(grepl("METAG", colnames_HF))]

viral_diversity_HF <- geneabund_drep_marker[,colnames_METAG_HF] 
colnames_METAT_HF <- colnames_HF[which(grepl("METAT", colnames_HF))]
viral_activity_HF <- geneabund_drep_marker[,colnames_METAT_HF] 
#vector of numeric parameters columns from metadata
env_numeric <- colnames(metadata_METAG_LF)[9:45]
#reformat metadata - sample names into rownames  
# rownames(metadata_METAG_FL) <- metadata_METAG_FL$BioSample
# rownames(metadata_METAT_FL) <- metadata_METAT_FL$BioSample
# rownames(metadata_METAG_LF) <- metadata_METAG_LF$BioSample
# rownames(metadata_METAT_LF) <- metadata_METAT_LF$BioSample
# rownames(metadata_METAG_HF) <- metadata_METAG_HF$BioSample
# rownames(metadata_METAT_HF) <- metadata_METAT_HF$BioSample
#take only numeric parameters
metadata_METAG_FL <- metadata_METAG_FL[,env_numeric]
metadata_METAT_FL <- metadata_METAT_FL[,env_numeric]
metadata_METAG_LF <- metadata_METAG_LF[,env_numeric]
metadata_METAT_LF <- metadata_METAT_LF[,env_numeric]
metadata_METAG_HF <- metadata_METAG_HF[,env_numeric]
metadata_METAT_HF <- metadata_METAT_HF[,env_numeric]
#prepare metadat for analysis
#replace all NA by average to proceed CCA
#define function
#it replaces na by mean 
#this is an approch to proceed with table having empty values
#it returns scaled (normalised) columns
avr_to_na_norm <- function(x){
  x[which(is.na(x))] <- mean(x, na.rm = T)
  x   
}

#for FL
metadata_METAG_FL_filled <- apply(metadata_METAG_FL, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_FL_filled <- as.data.frame(metadata_METAG_FL_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_FL <- apply(metadata_METAT_FL ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_FL_filled <- apply(metadata_METAT_FL, 2, avr_to_na_norm)
metadata_METAT_FL_filled <- as.data.frame(metadata_METAT_FL_filled) %>% select_if(~!any(is.nan(.))) 
#for LF
metadata_METAG_LF <- apply(metadata_METAG_LF ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAG_LF_filled <- apply(metadata_METAG_LF, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_LF_filled <- as.data.frame(metadata_METAG_LF_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numeric
metadata_METAT_LF <- apply(metadata_METAT_LF ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_LF_filled <- apply(metadata_METAT_LF, 2, avr_to_na_norm)
metadata_METAT_LF_filled <- as.data.frame(metadata_METAT_LF_filled) %>% select_if(~!any(is.nan(.))) 
#HF
metadata_METAG_HF_filled <- apply(metadata_METAG_HF, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_HF_filled <- as.data.frame(metadata_METAG_HF_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_HF <- apply(metadata_METAT_HF ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_HF_filled <- apply(metadata_METAT_HF, 2, avr_to_na_norm)
metadata_METAT_HF_filled <- as.data.frame(metadata_METAT_HF_filled) %>% select_if(~!any(is.nan(.))) 
```

##2. Free-living CCA 
###2.1 Diversity FL 
```{r CCA diversity FL}
#reformat viruses
t_viral_diversity_FL <- t(viral_diversity_FL)
t_viral_diversity_FL <- as.data.frame(t_viral_diversity_FL)
ps <- 0.000001

t_viral_diversity_FL <- t_viral_diversity_FL 
rownames(t_viral_diversity_FL) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity_FL))
rownames(t_viral_diversity_FL) <- gsub("_METAG.genecount.profile", "", rownames(t_viral_diversity_FL))

cca_diversity_FL <- cca(as.matrix(t_viral_diversity_FL), as.matrix(metadata_METAG_FL_filled))
summary(cca_diversity_FL) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower
```
```{r}
cc_diversity_FL <- cancor(as.matrix(t_viral_diversity_FL), as.matrix(metadata_METAG_FL_filled))
cc_diversity_FL$cor
#why correlation is one???
```
```{r}
plot(cca_diversity_FL)
```
```{r}
veg_1 = as.data.frame(cca_diversity_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_diversity_FL$CCA$v)
veg_2["primary_claster"] = row.names(veg_2)
drep_clusters_fin_taxonomy_unified$Salinity_niche <- factor(drep_clusters_fin_taxonomy_unified$Salinity_niche, levels = c("oligohaline", "mesohaline;oligohaline", "mesohaline", "mesohaline;polyhaline", "polyhaline", "oligohaline;polyhaline", "mesohaline;oligohaline;polyhaline"))

veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"


```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = class), alpha=0.5) + geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class)) 

veg_1 = veg_1[which(rownames(veg_1) %in% c("Phosphate_uM", "Nitrate_uM", "pH_TBDHereon", "Salinity_PSU", "Temperature_TBDHereon", "Turbidity_NTU", "Total_DIN_uM")),]
FL_vir_diversity_plot_class <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("FL_vir_diversity_plot_class_biplot.tiff", unit="px", width = 1000, height = 1000)
FL_vir_diversity_plot_class
dev.off()
```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Salinity_niche), alpha=1)  + scale_color_manual(values=c(brewer.pal(n = 5, name = "Blues"), "yellow", "purple"))

veg_1 = veg_1[which(rownames(veg_1) %in% c("Phosphate_uM", "Nitrate_uM", "pH_TBDHereon", "Salinity_PSU", "Temperature_TBDHereon", "Turbidity_NTU", "Total_DIN_uM")),]
FL_vir_diversity_plot_salinity <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("FL_vir_diversity_salinity_biplot.tiff", unit="px", width = 1000, height = 1000)
FL_vir_diversity_plot_salinity
dev.off()
```

```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Fraction_niche), alpha=1)  

veg_1 = veg_1[which(rownames(veg_1) %in% c("Phosphate_uM", "Nitrate_uM", "pH_TBDHereon", "Salinity_PSU", "Temperature_TBDHereon", "Turbidity_NTU", "Total_DIN_uM")),]
FL_vir_diversity_plot_fraction <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("FL_vir_diversity_fraction_biplot.tiff", unit="px", width = 1000, height = 1000)
FL_vir_diversity_plot_fraction
dev.off()
```

Salinity and temperature looks prominent.

###2.2 Expression FL

```{r CCA expression FL}
#reformat viruses
t_viral_expression_FL <- t(viral_expression_FL)
t_viral_expression_FL <- as.data.frame(t_viral_expression_FL)

rownames(t_viral_expression_FL) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_FL))
rownames(t_viral_expression_FL) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_FL))

cca_expression_FL <- cca(as.matrix(t_viral_expression_FL), as.matrix(metadata_METAT_FL_filled))
summary(cca_expression_FL) #scalling of the parameters doesnt matter sqrt of activity results makes proportion explained lower

```
```{r}
cc_activity_FL <- cancor(as.matrix(t_viral_expression_FL), as.matrix(metadata_METAT_FL_filled))
cc_activity_FL$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_FL$CCA$v)
veg_2["genus"] = row.names(veg_2)

veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"

plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") + 
  geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class)) +
                      geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("Phosphate_uM", "Nitrate_uM", "pH_TBDHereon", "Salinity_PSU", "Temperature_TBDHereon", "Turbidity_NTU")),]
plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```
###2.3 Activity FL

```{r CCA activity FL}
#reformat viruses
t_viral_activity_FL <- t(viral_activity_FL)
t_viral_activity_FL <- as.data.frame(t_viral_activity_FL)

rownames(t_viral_activity_FL) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_activity_FL))
rownames(t_viral_activity_FL) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_activity_FL))

cca_activity_FL <- cca(as.matrix(t_viral_activity_FL), as.matrix(metadata_METAT_FL_filled))
summary(cca_activity_FL) #scalling of the parameters doesnt matter sqrt of activity results makes proportion explained lower

```
```{r}
plot(cca_activity_FL)
```
```{r}
veg_1 = as.data.frame(cca_activity_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_activity_FL$CCA$v)
veg_2["genus"] = row.names(veg_2)

veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"


```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, color = class)) + 
  geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class))

plot
veg_1 = veg_1[which(rownames(veg_1) %in%  c("TotalDissolvedPhosphate_mg.L", "pH_TBDHereon", "Salinity_PSU", "Temperature_TBDHereon", "Turbidity_NTU")),]
FL_vir_activity_plot <- plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))

tiff("FL_vir_activity_biplot.tiff", unit="px", width = 1000, height = 1000)
FL_vir_activity_plot
dev.off()
```

```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Salinity_niche), alpha=1)  + scale_color_manual(values=c(brewer.pal(n = 5, name = "Blues"), "yellow", "purple"))

FL_vir_activity_plot_salinity <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("FL_vir_activity_salinity_biplot.tiff", unit="px", width = 1000, height = 1000)
FL_vir_activity_plot_salinity
dev.off()
```
## 3. Light fraction CCA
### 3.1 Diversity LF  
```{r CCA diversity LF}
#reformat viruses
t_viral_diversity_LF <- t(viral_diversity_LF)
t_viral_diversity_LF <- as.data.frame(t_viral_diversity_LF)
ps <- 0.000001

t_viral_diversity_LF <- t_viral_diversity_LF 
rownames(t_viral_diversity_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity_LF))
rownames(t_viral_diversity_LF) <- gsub("_METAG.genecount.profile", "", rownames(t_viral_diversity_LF))

cca_diversity_LF <- cca(as.matrix(t_viral_diversity_LF), as.matrix(metadata_METAG_LF_filled))
summary(cca_diversity_LF) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower

```
```{r}
cc_diversity_LF <- cancor(as.matrix(t_viral_diversity_LF), as.matrix(metadata_METAG_LF_filled))
cc_diversity_LF$cor
#why correlation is one???
```
```{r}
plot(cca_diversity_LF)
```
```{r}
veg_1 = as.data.frame(cca_diversity_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_diversity_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)
veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"


```
```{r}
#veg_1 = veg_1[which(!rownames(veg_1) %in% c("Sat_O2_Perc", "O2_uM", "Nitrite_uM", "dCH4_uM", "SPM_mgperL")),]
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, color = class)) + 
  geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class)) 
#+                      geom_point(data = veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
veg_1 = veg_1[which(rownames(veg_1) %in% c("Phosphate_uM", "Nitrate_uM", "pH_TBDHereon", "Salinity_PSU", "Sat_O2_Perc", "Temperature_TBDHereon", "Turbidity_NTU")),]


LF_vir_diversity_plot <- plot +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))

tiff("LF_vir_diversity_plot_biplot.tiff", unit="px", width = 1000, height = 1000)
LF_vir_diversity_plot
dev.off()
```

Salinity and temperature looks prominent.
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Salinity_niche), alpha=1)  + scale_color_manual(values=c(brewer.pal(n = 5, name = "Blues"), "yellow", "purple"))

LF_vir_diversity_plot_salinity <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("LF_vir_diversity_salinity_biplot.tiff", unit="px", width = 1000, height = 1000)
LF_vir_diversity_plot_salinity
dev.off()
```
###2.2 Expression LF

```{r CCA expression LF}
#reformat viruses
t_viral_expression_LF <- t(viral_expression_LF)
t_viral_expression_LF <- as.data.frame(t_viral_expression_LF)

rownames(t_viral_expression_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_LF))
rownames(t_viral_expression_LF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_LF))

cca_expression_LF <- cca(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
summary(cca_expression_LF) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower

```
```{r}
cc_expression_LF <- cancor(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
cc_expression_LF$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("TotalDissolvedPhosphate_mg.L", "Salinity_PSU", "Temperature_TBDHereon", "Silicate_mg.L")),]
plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))

```
###3.3 Activity LF

```{r CCA activity FL}
#reformat viruses
t_viral_activity_LF <- t(viral_activity_LF)
t_viral_activity_LF <- as.data.frame(t_viral_activity_LF)

rownames(t_viral_activity_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_activity_LF))
rownames(t_viral_activity_LF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_activity_LF))

cca_activity_LF <- cca(as.matrix(t_viral_activity_LF), as.matrix(metadata_METAT_LF_filled))
summary(cca_activity_LF) #scalling of the parameters doesnt matter sqrt of activity results makes proportion explained lower

```
```{r}
plot(cca_activity_LF)
```
```{r}
veg_1 = as.data.frame(cca_activity_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_activity_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)

veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"


```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, color = class)) + 
  geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class))

veg_1 = veg_1[which(rownames(veg_1) %in% c("TotalDissolvedPhosphate_mg.L", "Salinity_PSU", "Temperature_TBDHereon", "Silicate_mg.L")),]
LF_vir_activity_plot <- plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))

tiff("LF_vir_activity_biplot.tiff", unit="px", width = 1000, height = 1000)
LF_vir_activity_plot
dev.off()
```

```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Salinity_niche), alpha=1)  + scale_color_manual(values=c(brewer.pal(n = 5, name = "Blues"), "yellow", "purple"))

LF_vir_activity_plot_salinity <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("LF_vir_activity_salinity_biplot.tiff", unit="px", width = 1000, height = 1000)
LF_vir_activity_plot_salinity
dev.off()
```
##4. Heavy fraction CCA
### 4.1. Diversity HF
```{r CCA diversity HF}
#reformat viruses
t_viral_diversity_HF <- t(viral_diversity_HF)
t_viral_diversity_HF <- as.data.frame(t_viral_diversity_HF)
ps <- 0.000001

t_viral_diversity_HF <- t_viral_diversity_HF 
rownames(t_viral_diversity_HF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity_HF))
rownames(t_viral_diversity_HF) <- gsub("_METAG.genecount.profile", "", rownames(t_viral_diversity_HF))

cca_diversity_HF <- cca(as.matrix(t_viral_diversity_HF), as.matrix(metadata_METAG_HF_filled))
summary(cca_diversity_HF) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower

```
```{r}
plot(cca_diversity_HF)
```

```{r}
cc_diversity_HF <- cancor(as.matrix(t_viral_diversity_HF), as.matrix(metadata_METAG_HF_filled))
cc_diversity_HF$cor
#why correlation is one???
```
```{r}
plot(cca_diversity_HF)
```
```{r}
veg_1 = as.data.frame(cca_diversity_HF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_diversity_HF$CCA$v)
veg_2["genus"] = row.names(veg_2)

veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"


```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, color = class), alpha =0.5) + 
  geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class)) 

veg_1 = veg_1[which(rownames(veg_1) %in% c("Phosphate_uM", "Nitrate_uM", "DOC_mg", "Ammonium_mg.L", "dCO2_uM", "Salinity_PSU", "Temperature_TBDHereon", "Silicate_mg.L")),]

HF_vir_diversity_plot <- plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("HF_vir_diversity_biplot.tiff", unit="px", width = 1000, height = 1000)
HF_vir_diversity_plot
dev.off()
```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Salinity_niche), alpha=1)  + scale_color_manual(values=c(brewer.pal(n = 5, name = "Blues"), "yellow", "purple"))

HF_vir_diversity_plot_salinity <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("HF_vir_diversity_salinity_biplot.tiff", unit="px", width = 1000, height = 1000)
HF_vir_diversity_plot_salinity
dev.off()
```
Salinity and temperature looks prominent.

###2.2 Expression HF

```{r CCA expression HF}
#reformat viruses
t_viral_expression_LF <- t(viral_expression_LF)
t_viral_expression_LF <- as.data.frame(t_viral_expression_LF)

rownames(t_viral_expression_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_LF))
rownames(t_viral_expression_LF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_LF))

cca_expression_LF <- cca(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
summary(cca_expression_LF) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower

```
```{r}
cc_expression_LF <- cancor(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
cc_expression_LF$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  # geom_point(data =
  #              veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("TotalDissolvedPhosphate_mg.L", "pH_TBDHereon", "Salinity_PSU", "Temperature_TBDHereon", "Turbidity_NTU")),]
plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```

###3.3 Activity HF

```{r CCA activity HF}
#reformat viruses
t_viral_activity_HF <- t(viral_activity_HF)
t_viral_activity_HF <- as.data.frame(t_viral_activity_HF)

rownames(t_viral_activity_HF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_activity_HF))
rownames(t_viral_activity_HF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_activity_HF))

cca_activity_HF <- cca(as.matrix(t_viral_activity_HF), as.matrix(metadata_METAT_HF_filled))
summary(cca_activity_HF) #scalling of the parameters doesnt matter sqrt of activity results makes proportion explained lower

```
```{r}
plot(cca_activity_HF)
```

```{r}
cc_activity_HF <- cancor(as.matrix(t_viral_activity_HF), as.matrix(metadata_METAT_HF_filled))
cc_activity_HF$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_activity_HF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_activity_HF$CCA$v)
veg_2["genus"] = row.names(veg_2)

veg_2 <- merge(veg_2, drep_clusters_fin_taxonomy_unified[, c("primary_cluster", "level_5", "Salinity_niche", "Fraction_niche")], by.x = "row.names", by.y = "primary_cluster")
#rename the column
colnames(veg_2)[which(colnames(veg_2)=="level_5")] <- "class"
veg_2$class[which(is.na(veg_2$class))] <- "unknown"

```

```{r}

plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, color = class), alpha=0.5) + 
  geom_mark_ellipse(data = veg_2, aes(x = CCA1, y = CCA2, color=class)) 

veg_1 = veg_1[which(rownames(veg_1) %in%  c("TotalDissolvedPhosphate_mg.L", "Silicate_mg.L", "Salinity_PSU", "Temperature_TBDHereon")),]
HF_vir_activity_plot <- plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("HF_vir_activity_biplot.tiff", unit="px", width = 1000, height = 1000)
HF_vir_activity_plot
dev.off()
```
```{r}
plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2, colour = Salinity_niche), alpha=1)  + scale_color_manual(values=c(brewer.pal(n = 5, name = "Blues"), "yellow", "purple"))

HF_vir_activity_plot_salinity <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 3
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
tiff("HF_vir_activity_salinity_biplot.tiff", unit="px", width = 1000, height = 1000)
HF_vir_activity_plot_salinity
dev.off()
```

##5. All fractions
```{r CCA activity HF}
#reformat viruses
t_viral_diversity <- t(geneabund_drep_marker)
t_viral_diversity <- as.data.frame(t_viral_diversity)

rownames(t_viral_diversity) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity))
rownames(t_viral_diversity) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_diversity))
metadata$Sample_type <- as.factor(metadata$Sample_type)
metadata_all <- metadata[, c("Temperature_TBDHereon", "Salinity_PSU")]
cca_diversity <- cca(as.matrix(t_viral_diversity), as.matrix(metadata_all))
summary(cca_diversity) #scalling of the parameters doesnt matter sqrt of activity results makes proportion explained lower

```
```{r}
plot(cca_diversity)
```

##6. Build model

We will try to use viral abundance data based on individual marker genes 
```{r}
library(glmnet)
#divide dataset by test and train
#try METAG first
#we have to transform data so we have counts of each primary cluster in separate column and each row is a different samples with a parameter
geneabund_drep_marker_taxonomy_long <- readRDS(file="geneabund_drep_marker_taxonomy_long.RDS")
#remove taxonomy distribution so we dont we multiple separe vectors for the same sample but one which represents community
geneabund_drep_marker_taxonomy_wide <-  geneabund_drep_marker_taxonomy_long %>% ungroup() %>% select(-c("lineage", "level_5"))  %>%  pivot_wider(names_from=primary_cluster, values_from=counts, values_fill = 0, names_prefix="cluster_") %>% distinct()
geneabund_drep_marker_taxonomy_wide_METAG <- geneabund_drep_marker_taxonomy_wide %>% filter(data_type == "METAG") 
                                                                                            
#keep rows where we have collected param
param ="Phosphate_uM"
#param= "DOC_mg.L"
geneabund_drep_marker_taxonomy_wide_METAG <- geneabund_drep_marker_taxonomy_wide_METAG %>% filter(!is.na(Salinity_PSU))

geneabund_drep_marker_taxonomy_wide_METAT <- geneabund_drep_marker_taxonomy_wide %>% filter(data_type == "METAT")
#draw randomly train and test set 80:20
set.seed(123)
p <- nrow(geneabund_drep_marker_taxonomy_wide_METAG)
train_rows <- sample(size=p*0.8, 1:p)
geneabund_drep_marker_taxonomy_wide_METAG_train <- geneabund_drep_marker_taxonomy_wide_METAG[train_rows,]
geneabund_drep_marker_taxonomy_wide_METAG_train_x <- geneabund_drep_marker_taxonomy_wide_METAG_train %>% ungroup() %>% select(starts_with("cluster"))
geneabund_drep_marker_taxonomy_wide_METAG_test <- geneabund_drep_marker_taxonomy_wide_METAG[-train_rows,]
geneabund_drep_marker_taxonomy_wide_METAG_test_x <- geneabund_drep_marker_taxonomy_wide_METAG_test %>% ungroup() %>% select(starts_with("cluster"))
y_train <- geneabund_drep_marker_taxonomy_wide_METAG_train$Salinity_PSU 
y_test <- geneabund_drep_marker_taxonomy_wide_METAG_test$Salinity_PSU 
# #try the same for phosphate 
# y_train <- as.vector(geneabund_drep_marker_taxonomy_wide_METAG_train$Phosphate_uM) 
# y_test <- as.vector(geneabund_drep_marker_taxonomy_wide_METAG_test$Phosphate_uM)

models <- list()
#train for different alpha values
for (i in 0:20) {
  name <- paste0("alpha", i/20)
  

  models[[name]] <- cv.glmnet(as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_train_x), y_train, alpha=i/20)
  
}


#test models
results <- data.frame()
for (i in 0:20) {
  name <- paste0("alpha", i/20)
  
  ## Use each model to predict 'y' given the Testing dataset
  predicted <- predict(models[[name]], 
                       s=models[[name]]$lambda.1se, newx=as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_test_x))
  
  ## Calculate the Mean Squared Error...
  mse <- mean((y_test - predicted)^2)
  mae <- mean((y_test - predicted))
  #add pseudo counts so no inf values
  #y_test= y_test + 0.001
  mape <- mean(abs((y_test - predicted)/(y_test)))*100
  ## Store the results
  rsq <- 1-models[[name]]$cvm/var(y_test)
  temp <- data.frame(alpha=i/20, mse = mse,  mae = mae, mape = mape, rsq=rsq)
  results <- rbind(results, temp)
}
```
```{r}

predicted_y <- predict(models[["alpha0.25"]], s=models[["alpha0.25"]]$lambda.1se, newx=as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_test_x))
plot(y_test, predicted_y, ylim=c(0,20))
linear_model<-lm(y_test ~ predicted_y)
R2<-summary(linear_model)$r.squared
abline(a=0, b=1, col ="black")
#looks like two groups - 
#try to find wich samples have the biggest mae ~ 7 - check if they have sth in common
```
We will try to use viral abundance data based on individual marker genes 
```{r}

geneabund_drep_marker_taxonomy_wide_METAG <- geneabund_drep_marker_taxonomy_wide_METAG %>% filter(!is.na(Phosphate_uM))

geneabund_drep_marker_taxonomy_wide_METAT <- geneabund_drep_marker_taxonomy_wide %>% filter(data_type == "METAT")
#draw randomly train and test set 80:20
set.seed(123)
p <- nrow(geneabund_drep_marker_taxonomy_wide_METAG)
train_rows <- sample(size=p*0.8, 1:p)
geneabund_drep_marker_taxonomy_wide_METAG_train <- geneabund_drep_marker_taxonomy_wide_METAG[train_rows,]
geneabund_drep_marker_taxonomy_wide_METAG_train_x <- geneabund_drep_marker_taxonomy_wide_METAG_train %>% ungroup() %>% select(starts_with("cluster"))
geneabund_drep_marker_taxonomy_wide_METAG_test <- geneabund_drep_marker_taxonomy_wide_METAG[-train_rows,]
geneabund_drep_marker_taxonomy_wide_METAG_test_x <- geneabund_drep_marker_taxonomy_wide_METAG_test %>% ungroup() %>% select(starts_with("cluster"))
y_train <- geneabund_drep_marker_taxonomy_wide_METAG_train$Salinity_PSU 
y_test <- geneabund_drep_marker_taxonomy_wide_METAG_test$Salinity_PSU 
# #try the same for phosphate 
y_train <- as.vector(geneabund_drep_marker_taxonomy_wide_METAG_train$Phosphate_uM)
y_test <- as.vector(geneabund_drep_marker_taxonomy_wide_METAG_test$Phosphate_uM)

models <- list()
#train for different alpha values
for (i in 0:20) {
  name <- paste0("alpha", i/20)
  

  models[[name]] <- cv.glmnet(as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_train_x), y_train, alpha=i/20)
}


#test models
results <- data.frame()
for (i in 0:20) {
  name <- paste0("alpha", i/20)
  
  ## Use each model to predict 'y' given the Testing dataset
  predicted <- predict(models[[name]], 
                       s=models[[name]]$lambda.1se, newx=as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_test_x))
  
  ## Calculate the Mean Squared Error...
  mse <- mean((y_test - predicted)^2)
  mae <- mean((y_test - predicted))
  #add pseudo counts so no inf values
  #y_test= y_test + 0.001
  mape <- mean(abs((y_test - predicted)/(y_test)))*100
  ## Store the results
  rsq <- 1-cvfit$cvm/var(y)
  temp <- data.frame(alpha=i/20, mse = mse, name = name, mae = mae, mape = mape, rsq=rsq)
  results <- rbind(results, temp)
}
```
```{r}

predicted_y <- predict(models[["alpha0.15"]], s=models[["alpha0.15"]]$lambda.1se, newx=as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_test_x))
plot(y_test, predicted_y, ylim=c(0,4))
#abline(lm(y_test ~ predicted_y), b=1, col=c("red")
abline(a=0, b=1, col ="black")
#looks like two groups - 
#try to find wich samples have the biggest mae ~ 7 - check if they have sth in common
```
```{r}

geneabund_drep_marker_taxonomy_wide_METAG <- geneabund_drep_marker_taxonomy_wide_METAG %>% filter(!is.na(DOC_mg.L))

geneabund_drep_marker_taxonomy_wide_METAT <- geneabund_drep_marker_taxonomy_wide %>% filter(data_type == "METAT")
#draw randomly train and test set 80:20
set.seed(123)
p <- nrow(geneabund_drep_marker_taxonomy_wide_METAG)
train_rows <- sample(size=p*0.8, 1:p)
geneabund_drep_marker_taxonomy_wide_METAG_train <- geneabund_drep_marker_taxonomy_wide_METAG[train_rows,]
geneabund_drep_marker_taxonomy_wide_METAG_train_x <- geneabund_drep_marker_taxonomy_wide_METAG_train %>% ungroup() %>% select(starts_with("cluster"))
geneabund_drep_marker_taxonomy_wide_METAG_test <- geneabund_drep_marker_taxonomy_wide_METAG[-train_rows,]
geneabund_drep_marker_taxonomy_wide_METAG_test_x <- geneabund_drep_marker_taxonomy_wide_METAG_test %>% ungroup() %>% select(starts_with("cluster"))

# #try the same for phosphate 
y_train <- as.vector(geneabund_drep_marker_taxonomy_wide_METAG_train$DOC_mg.L)
y_test <- as.vector(geneabund_drep_marker_taxonomy_wide_METAG_test$DOC_mg.L)

models <- list()
#train for different alpha values
for (i in 0:20) {
  name <- paste0("alpha", i/20)
  

  models[[name]] <- cv.glmnet(as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_train_x), y_train, type.measure="mae", alpha=i/20)
}


#test models
results <- data.frame()
for (i in 0:20) {
  name <- paste0("alpha", i/20)
  
  ## Use each model to predict 'y' given the Testing dataset
  predicted <- predict(models[[name]], 
                       s=models[[name]]$lambda.1se, newx=as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_test_x))
  
  ## Calculate the Mean Squared Error...
  mse <- mean((y_test - predicted)^2)
  mae <- mean((y_test - predicted))
  #add pseudo counts so no inf values
  #y_test= y_test + 0.001
  mape <- mean(abs((y_test - predicted)/(y_test)))*100
  ## Store the results
  temp <- data.frame(alpha=i/20, mse = mse, name = name, mae = mae, mape = mape)
  results <- rbind(results, temp)
}
```
```{r}

predicted_y <- predict(models[["alpha0.5"]], s=models[["alpha0.5"]]$lambda.1se, newx=as.matrix(geneabund_drep_marker_taxonomy_wide_METAG_test_x))
plot(y_test, predicted_y, ylim=c(0,8))
#abline(lm(y_test ~ predicted_y), b=1, col=c("red")
abline(a=0, b=1, col ="black")
#looks like two groups - 
#try to find wich samples have the biggest mae ~ 7 - check if they have sth in common
```

## 7. Viral diversity indexes - shanon simpson, richness based on occurence and niches
