---
title: "CCA"
author: "Julia Golebiowska"
date: "2024-10-13"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Introduction
CCA  - Canonical correspondence analysis on viral data from the Elbe estuary for each fraction and and data type - vial abundance and activity based on individual marker genes. This analysis revels the potential drivers of communities diversity and activity. 



```{r test}
library(vegan)
library(ggplot2)
library(ggrepel)
library(dplyr)
```
## Viral diversity 

##1. Read data and metadata for each fraction and data type


```{r , echo=FALSE}
#load marker genes
geneabund_drep_marker_taxa <- readRDS("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/geneabund_drep_marker_taxa.RDS")
#remove  repeating rows - I dont need host taxonomy 
# next time generate special object with viral taxonomy and salinity niche classification
geneabund_drep_marker_taxa <- geneabund_drep_marker_taxa %>% select(contains("GROS")) %>% distinct()
#drep clusters 
drep_clusters <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Cdb.csv", header = TRUE, sep = ",")
drep_clusters[,"genome"] = sub(".fna", "", drep_clusters[,"genome"])
#FL

metadata_METAG_FL = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/metadata_METAG_FL.RDS")
metadata_METAT_FL = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/metadata_METAT_FL.RDS")
marker_gene_METAG_FL <- geneabund_drep_marker_taxa %>% select(contains(metadata_METAG_FL$AccessionNumber_TBDSven)) %>% select(contains("METAG"))
marker_gene_METAT_FL <- geneabund_drep_marker_taxa %>% select(contains(metadata_METAT_FL$AccessionNumber_TBDSven)) %>% select(contains("METAG"))
#LF
viral_diversity_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/viral_diversity_LF.RDS")
viral_expression_LF = readRDS( file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/viral_expression_LF.RDS")
metadata_METAG_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/metadata_METAG_LF.RDS")
metadata_METAT_LF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/metadata_METAT_LF.RDS")
#HF
viral_diversity_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/viral_diversity_HF.RDS")
viral_expression_HF = readRDS( file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/viral_expression_HF.RDS")
metadata_METAG_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/metadata_METAG_HF.RDS")
metadata_METAT_HF = readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/metadata_METAT_HF.RDS")
#vector of numeric parameters columns from metadata
env_numeric <- c("Sat_O2_TBDHereon", "Temperature_TBDHereon", "Salinity_TBDHereon", "Turbidity_TBDHereon", "pH_TBDHereon", "O2_TBDHereon", "SPM_mgperL", "DOC_mg.L", "TN_mg.L", "DIC_mg.L", "Silicate_mg.L", "Ammonium_mg.L", "Nitrite_mg.L", "Nitrate_mg.L", "Total_DIN_µM", "SRP_mgperL", "TotalDissolvedPhosphate_mg.L", "Ammonium_µM", "Phosphate_µM", "RespirationRate_O2ug.L.h", "POC_mgperL", "PTC_mgperL", "PTN_mgperL", "PTH_mgperL", "dCH4_nM", "Chla_ug/l", "dCO2_uM")
#reformat metadata - sample names into rownames  
rownames(metadata_METAG_FL) <- metadata_METAG_FL$BioSample
rownames(metadata_METAT_FL) <- metadata_METAT_FL$BioSample
rownames(metadata_METAG_LF) <- metadata_METAG_LF$BioSample
rownames(metadata_METAT_LF) <- metadata_METAT_LF$BioSample
rownames(metadata_METAG_HF) <- metadata_METAG_HF$BioSample
rownames(metadata_METAT_HF) <- metadata_METAT_HF$BioSample
#take only numeric parameters
metadata_METAG_FL <- metadata_METAG_FL[,env_numeric]
metadata_METAT_FL <- metadata_METAT_FL[,env_numeric]
metadata_METAG_LF <- metadata_METAG_LF[,env_numeric]
metadata_METAT_LF <- metadata_METAT_LF[,env_numeric]
metadata_METAG_HF <- metadata_METAG_HF[,env_numeric]
metadata_METAT_HF <- metadata_METAT_HF[,env_numeric]
#prepare metadat for analysis
#replace all NA by average to proceed CCA
#define function
#it replaces na by mean 
#this is an approch to proceed with table having empty values
#it returns scaled (normalised) columns
avr_to_na_norm <- function(x){
  x[which(is.na(x))] <- mean(x, na.rm = T)
  x   
}

#for FL
metadata_METAG_FL_filled <- apply(metadata_METAG_FL, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_FL_filled <- as.data.frame(metadata_METAG_FL_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_FL <- apply(metadata_METAT_FL ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_FL_filled <- apply(metadata_METAT_FL, 2, avr_to_na_norm)
metadata_METAT_FL_filled <- as.data.frame(metadata_METAT_FL_filled) %>% select_if(~!any(is.nan(.))) 
#for LF
metadata_METAG_LF <- apply(metadata_METAG_LF ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAG_LF_filled <- apply(metadata_METAG_LF, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_LF_filled <- as.data.frame(metadata_METAG_LF_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numeric
metadata_METAT_LF <- apply(metadata_METAT_LF ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_LF_filled <- apply(metadata_METAT_LF, 2, avr_to_na_norm)
metadata_METAT_LF_filled <- as.data.frame(metadata_METAT_LF_filled) %>% select_if(~!any(is.nan(.))) 
#HF
metadata_METAG_HF_filled <- apply(metadata_METAG_HF, 2, avr_to_na_norm)
#same columns can be empty - get rid of them 
metadata_METAG_HF_filled <- as.data.frame(metadata_METAG_HF_filled) %>% select_if(~!any(is.nan(.))) 
#the same procedure for METAT
#first replace all comas by points and change to numric
metadata_METAT_HF <- apply(metadata_METAT_HF ,2,function(x) as.numeric(gsub(",", ".", as.character(x))))
metadata_METAT_HF_filled <- apply(metadata_METAT_HF, 2, avr_to_na_norm)
metadata_METAT_HF_filled <- as.data.frame(metadata_METAT_HF_filled) %>% select_if(~!any(is.nan(.))) 
```

##2. Free-living CCA 
###2.1 Diversity FL 
```{r CCA diversity FL}
#reformat viruses
t_viral_diversity_FL <- t(marker_gene_METAG_FL)
t_viral_diversity_FL <- as.data.frame(t_viral_diversity_FL)
ps <- 0.000001

t_viral_diversity_FL <- t_viral_diversity_FL 
rownames(t_viral_diversity_FL) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity_FL))
rownames(t_viral_diversity_FL) <- gsub("_METAG.genecount.profile", "", rownames(t_viral_diversity_FL))

cca_diversity_FL <- cca(as.matrix(t_viral_diversity_FL), as.matrix(metadata_METAG_FL_filled))
summary(cca_diversity_FL) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower

```
```{r}
cc_diversity_FL <- cancor(as.matrix(t_viral_diversity_FL), as.matrix(metadata_METAG_FL_filled))
cc_diversity_FL$cor
#why correlation is one???
```
```{r}
plot(cca_diversity_FL)
```
```{r}
veg_1 = as.data.frame(cca_diversity_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_diversity_FL$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```
```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("O2_TBDHereon", "Phosphate_µM", "Silicate_mg.L", "Temperature_TBDHereon", "Total_DIN_µM")),]
FL_vir_diversity_plot <- plot +
  #geom_text_repel(data = veg_2,
  #                aes(x = CCA1, y = CCA2, label =),
  #                nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
FL_vir_diversity_plot
```

Salinity and temperature looks prominent.

###2.2 Expression FL

```{r CCA expression FL}
#reformat viruses
t_viral_expression_FL <- t(viral_expression_FL)
t_viral_expression_FL <- as.data.frame(t_viral_expression_FL)

rownames(t_viral_expression_FL) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_FL))
rownames(t_viral_expression_FL) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_FL))

cca_expression_FL <- cca(as.matrix(t_viral_expression_FL), as.matrix(metadata_METAT_FL_filled))
summary(cca_expression_FL) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower

```
```{r}
cc_expression_FL <- cancor(as.matrix(t_viral_expression_FL), as.matrix(metadata_METAT_FL_filled))
cc_expression_FL$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_FL$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_FL$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("Sat_O2_TBDHereon", "TotalDissolvedPhosphate_mg.L
","Silicate_mg.L","Salinity_TBDHereon","O2_TBDHereon", "pH_TBDHereon", "Temperature_TBDHereon
0", "pH_TBDHereon", "Silicate_mg.L", "dCO2_uM")),]
plot +
  # geom_text_repel(data = veg_2,
    #              aes(x = CCA1, y = CCA2, label = F),
    #              nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```
## 3. Light fraction CCA
### 3.1 Diversity LF  
```{r CCA diversity LF}
#reformat viruses
t_viral_diversity_LF <- t(viral_diversity_LF)
t_viral_diversity_LF <- as.data.frame(t_viral_diversity_LF)
ps <- 0.000001

t_viral_diversity_LF <- t_viral_diversity_LF 
rownames(t_viral_diversity_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity_LF))
rownames(t_viral_diversity_LF) <- gsub("_METAG.genecount.profile", "", rownames(t_viral_diversity_LF))

cca_diversity_LF <- cca(as.matrix(t_viral_diversity_LF), as.matrix(metadata_METAG_LF_filled))
summary(cca_diversity_LF) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower

```
```{r}
cc_diversity_LF <- cancor(as.matrix(t_viral_diversity_LF), as.matrix(metadata_METAG_LF_filled))
cc_diversity_LF$cor
#why correlation is one???
```
```{r}
plot(cca_diversity_LF)
```
```{r}
veg_1 = as.data.frame(cca_diversity_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_diversity_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```
```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("O2_TBDHereon", "pH_TBDHereon", "Phosphate_µM", "Silicate_mg.L", "Temperature_TBDHereon", "Total_DIN_µM")),]
plot +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```

Salinity and temperature looks prominent.

###2.2 Expression LF

```{r CCA expression LF}
#reformat viruses
t_viral_expression_LF <- t(viral_expression_LF)
t_viral_expression_LF <- as.data.frame(t_viral_expression_LF)

rownames(t_viral_expression_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_LF))
rownames(t_viral_expression_LF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_LF))

cca_expression_LF <- cca(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
summary(cca_expression_LF) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower

```
```{r}
cc_expression_LF <- cancor(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
cc_expression_LF$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("Ammonium_mg.L", "Temperature_TBDHereon", "Salinity_TBDHereon", "TotalDissolvedPhosphate_mg.L")),]
plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```
##4. Heavy fraction CCA
### 4.1. Diversity HF
```{r CCA diversity HF}
#reformat viruses
t_viral_diversity_LF <- t(viral_diversity_LF)
t_viral_diversity_LF <- as.data.frame(t_viral_diversity_LF)
ps <- 0.000001

t_viral_diversity_LF <- t_viral_diversity_LF 
rownames(t_viral_diversity_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_diversity_LF))
rownames(t_viral_diversity_LF) <- gsub("_METAG.genecount.profile", "", rownames(t_viral_diversity_LF))

cca_diversity_LF <- cca(as.matrix(t_viral_diversity_LF), as.matrix(metadata_METAG_LF_filled))
summary(cca_diversity_LF) #scalling of the parameters doesnt matter sqrt of diversity results makes proportion explained lower

```
```{r}
cc_diversity_LF <- cancor(as.matrix(t_viral_diversity_LF), as.matrix(metadata_METAG_LF_filled))
cc_diversity_LF$cor
#why correlation is one???
```
```{r}
plot(cca_diversity_LF)
```
```{r}
veg_1 = as.data.frame(cca_diversity_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genusv
veg_2 = as.data.frame(cca_diversity_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```
```{r}

veg_1 = veg_1 = veg_1[which(rownames(veg_1) %in% c("Total_DIN_µM", "O2_TBDHereon","Phosphate_µM", "pH_TBDHereon", "Temperature_TBDHereon")),]
plot +
  geom_text_repel(data = veg_2,
                  aes(x = CCA1, y = CCA2, label = veg_2$genus),
                  nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```

Salinity and temperature looks prominent.

###2.2 Expression HF

```{r CCA expression HF}
#reformat viruses
t_viral_expression_LF <- t(viral_expression_LF)
t_viral_expression_LF <- as.data.frame(t_viral_expression_LF)

rownames(t_viral_expression_LF) <- gsub("GROS22.(1|2)_", "", rownames(t_viral_expression_LF))
rownames(t_viral_expression_LF) <- gsub("_METAT.genecount.profile", "", rownames(t_viral_expression_LF))

cca_expression_LF <- cca(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
summary(cca_expression_LF) #scalling of the parameters doesnt matter sqrt of expression results makes proportion explained lower

```
```{r}
cc_expression_LF <- cancor(as.matrix(t_viral_expression_LF), as.matrix(metadata_METAT_LF_filled))
cc_expression_LF$cor
#why correlation is one???
```
```{r}
veg_1 = as.data.frame(cca_expression_LF$CCA$biplot)
veg_1["env"] = row.names(veg_1)

#extracting the data; genus
veg_2 = as.data.frame(cca_expression_LF$CCA$v)
veg_2["genus"] = row.names(veg_2)


plot = ggplot() +
  geom_point(data = veg_2, aes(x = CCA1, y = CCA2), color = "red") +
  geom_point(data =
               veg_1, aes(x = CCA1, y = CCA2), color = "blue")

plot
```

```{r}

veg_1 = veg_1[which(rownames(veg_1) %in% c("TotalDissolvedPhosphate_mg.L
", "Temperature_TBDHereon
0", "pH_TBDHereon", "Salinity_TBDHereon")),]
plot +
  # geom_text_repel(data = veg_2,
  #                 aes(x = CCA1, y = CCA2, label = veg_2$genus),
  #                 nudge_y = -0.05) +
  theme_bw() +
  geom_segment(
    data = veg_1,
    aes(
      x = 0,
      y = 0,
      xend = CCA1,
      yend = CCA2
    ),
    arrow = arrow(length = unit(0.25, "cm")), size=1
  ) +
  geom_text_repel(
    data = veg_1,
    aes(x = CCA1, y = CCA2, label = veg_1$env),
    nudge_y = -0.05,
    color = "blue",
    size = 5
  ) +
  theme(axis.text = element_text(size = 16),
        axis.title = element_text(size = 18))
```