---
title: "mentel_test_Elbe"
author: "Julia Golebiowska"
date: "2024-04-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyverse)
library(ggplot2)
library(vegan)
setwd("~/IGB_phd/virus/viral_abundances")
```
```{r load metadata}
##Load Metadata for later
path = "C:/Users/jgolebiowska/Documents/IGB_phd/workshop_gene_catalog/"

#combine metadata 
metadata_sim <- read.csv(paste0(path,"/SAMEAID_SampleID_simplified.csv"), header=TRUE, sep=";") 
#select the needed columns to avoid doubling computations
metadata_sim <-metadata_sim[,c("BioSample", "ProjectID")]
#read simplified metadata, as it has a ProjectID 
metadata <- read.csv(paste0(path,"/SAMEAID_SampleID.csv"), header=TRUE, sep=",") %>% 
  left_join(metadata_sim, by = "BioSample")  

#add chlorophyll (and gas values)  to metadata from another file
metadata_more <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_genes/Stations_Daten_RTG_250923.csv", sep=",", skip=2, header = F)
metadata_more[1,] <- paste(metadata_more[1,], metadata_more[2,], sep = "_")
#as header
names(metadata_more) <- metadata_more[1,]
#remove the first line - header  manually
metadata_more <- metadata_more[-1,]
#update enumeration
rownames(metadata_more) <- 1:nrow(metadata_more)
metadata_more <-metadata_more[metadata_more!="" & !is.na(metadata_more[,1]),]
metadata_more$Cruise_[metadata_more$Cruise_=="LP210308"]<-"Mar 21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP210503"]<-"Mai 21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP210725"]<-"Jul 21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220228"]<-"Feb 22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220522"]<-"Mai 22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220613"]<-"Jun 22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP221107"]<-"Nov 22"
#put the same name as in metadata
colnames(metadata_more)[1] <- "Sample_date"
colnames(metadata_more)[5] <- "Station"
#unify station names
metadata_more$Station[metadata_more$Station=="Medem Grund"] <- "Medemgrund"
#unify date names (remove '-')
metadata <-metadata %>% mutate(Sample_date = str_replace(Sample_date, "-"," "))
#unify station names
metadata$Station[metadata$Station=="Meedem Grund"] <- "Medemgrund"
metadata_more$Station[metadata_more$Station=="Medem Grund"] <- "Medemgrund"

#remove another meassurement from the same station and date
metadata_more <- metadata_more[-5,]
#update enumeration
rownames(metadata_more) <- 1:nrow(metadata_more)
#remove 'micro' sign
colnames(metadata_more)[which(colnames(metadata_more)=="Chla_µg/l")] <- "Chla_ug/l"
colnames(metadata_more)[which(colnames(metadata_more)=="dCO2_µM")] <- "dCO2_uM"
#left_join with the CO2 CH4 Chlorophyll values from  metadata_more by Sample date
metadata <- metadata %>% merge(metadata_more[, c("Sample_date", "Station", "dCH4_nM", "Chla_ug/l","dCO2_uM")], by = c("Sample_date", "Station"), all.x = T)

```

```{r load motus}
path = "C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/MOTUs/"
motus_final_taxa_metadata <- readRDS(file=paste0(path,"motus_final_taxa_metadata.RDS"))
```

```{r edit motus}
motus_final_taxa_metadata <- motus_final_taxa_metadata %>% 
  filter(!grepl("Bunthaus|Kollmar|Seemans", station), !grepl("Nov 21", date))

motus_df_METAG <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()
# only metatranscriptomes
motus_df_METAT <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()
#normalised mOTU
#check rows 
print(sum(motus_df_METAT$mOTU == motus_df_METAG$mOTU) == length(motus_df_METAG$mOTU) && (length(motus_df_METAG$mOTU) == length( motus_df_METAT$mOTU))) #ok true
#check columns 
print(sum(colnames(motus_df_METAT) == colnames(motus_df_METAG)) && (length(colnames(motus_df_METAG)) == length(colnames(motus_df_METAT)))) #false
#lets adjust columns belonging to both datasets - there are no respective metaT for each metaG and opposite  
paired_METAG <- sort(gsub("METAT", "METAG", colnames(motus_df_METAT))) 
#remove for now non numeric column mOTU 
paired_METAG <- paired_METAG[-c(1)]
#take subset which exist 
paired_METAG <- colnames(motus_df_METAG)[which(colnames(motus_df_METAG) %in% paired_METAG)]
#translate this subset into transcript samples ids
paired_METAT <- sort(gsub("METAG", "METAT", paired_METAG))
#save paired ids
paired_sampleid <- gsub("_METAG", "", paired_METAG) 
#normalise - log(transcriptions/gene counts)
#first add pseudocounts
pseudocount <- 0.000001 
motus_df_METAG_paired <- motus_df_METAG[, paired_METAG] 
motus_df_METAT_paired <- motus_df_METAT[, paired_METAT]
motus_expression <-  log(motus_df_METAT_paired + pseudocount) - log(motus_df_METAG_paired + pseudocount)

#METAT FL
motus_df_METAT_FL <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
   filter(sample_type == "Free_living") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()

#METAG FL
motus_df_METAG_FL <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
   filter(sample_type == "Free_living") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()

#METAT LF
motus_df_METAT_LF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
   filter(sample_type == "Light_fraction") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()

#METAG LF
motus_df_METAG_LF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
   filter(sample_type == "Light_fraction") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()

#METAT HF
motus_df_METAT_HF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
   filter(sample_type == "Heavy_fraction") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()

#METAG HF
motus_df_METAG_HF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
   filter(sample_type == "Heavy_fraction") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()

#expression FL
motus_expression_FL <- motus_expression[, colnames(motus_expression) %in% colnames(motus_df_METAT_FL)]  
#expression LF
motus_expression_LF <- motus_expression[, colnames(motus_expression) %in% colnames(motus_df_METAT_LF)]  
#expression HF
motus_expression_HF <- motus_expression[, colnames(motus_expression) %in% colnames(motus_df_METAT_HF)]  

## for later keep names of samples which were used for metagenomes and metatranscriptomes and finaly use for th analysis  
sampleid_METAG <- unique(motus_final_taxa_metadata[which(motus_final_taxa_metadata$data_type=="METAG"), "BioSample"])
sampleid_METAT <- unique(motus_final_taxa_metadata[which(motus_final_taxa_metadata$data_type=="METAT"), "BioSample"])
```

```{r hosts motus }
# select motus which are assigned hosts
#load hosts assignment
host_prediction_genome = read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/Host_prediction_to_genome_m90.csv")
motu_stats = readRDS(file=paste0(path,"motu_stats.RDS"))
#join Motus by MAGs which were assigned as hosts with iphop
motus_MAGS_METAG <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
  left_join(motu_stats, by = "mOTU")
host_motus_METAG <- host_prediction_genome %>% left_join(motus_MAGS_METAG, by =c("Host.genome" = "user_genome"))
#FL for METAG
motus_MAGS_METAG_FL <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
  filter(sample_type == "Free_living") %>%
  left_join(motu_stats, by = "mOTU")
#LF for METAG
motus_MAGS_METAG_LF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
  filter(sample_type == "Light_fraction") %>%
  left_join(motu_stats, by = "mOTU")
#HF for METAG
motus_MAGS_METAG_HF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
  filter(sample_type == "Heavy_fraction") %>%
  left_join(motu_stats, by = "mOTU")
#find hosts from MOTUs
host_motus_METAG <- host_prediction_genome %>% left_join(motus_MAGS_METAG, by =c("Host.genome" = "user_genome"))
host_motus_METAG_FL <- host_prediction_genome %>% left_join(motus_MAGS_METAG_FL, by =c("Host.genome" = "user_genome"))
host_motus_METAG_LF <- host_prediction_genome %>% left_join(motus_MAGS_METAG_LF, by =c("Host.genome" = "user_genome"))
host_motus_METAG_HF <- host_prediction_genome %>% left_join(motus_MAGS_METAG_HF, by =c("Host.genome" = "user_genome"))
#reformat so distances can be calculated
host_motus_METAG_wide <-  host_motus_METAG %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 
#FL
host_motus_METAG_FL_wide <-  host_motus_METAG_FL %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0)
#LF
host_motus_METAG_LF_wide <-  host_motus_METAG_LF %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0)
#HF
host_motus_METAG_HF_wide <-  host_motus_METAG_HF %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0)

#reformat so distances can be calculated
host_motus_METAG_wide <-  host_motus_METAG %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 


##### for METAT
#join Motus by MAGs which were assigned as hosts with iphop
motus_MAGS_METAT <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
  left_join(motu_stats, by = "mOTU")
#check out free-living fraction
motus_MAGS_METAT_FL <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
  filter(sample_type == "Free_living") %>%
  left_join(motu_stats, by = "mOTU")

#check out light fraction
motus_MAGS_METAT_LF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
  filter(sample_type == "Light_fraction") %>%
  left_join(motu_stats, by = "mOTU")
#check out heavy fraction
motus_MAGS_METAT_HF <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
  filter(sample_type == "Heavy_fraction") %>%
  left_join(motu_stats, by = "mOTU")


host_motus_METAT <- host_prediction_genome %>% left_join(motus_MAGS_METAT, by =c("Host.genome" = "user_genome"))
# host_motus_METAT_FL <- host_prediction_genome %>% left_join(motus_MAGS_METAT_FL, by =c("Host.genome" = "user_genome"))
# host_motus_METAT_LF <- host_prediction_genome %>% left_join(motus_MAGS_METAT_LF, by =c("Host.genome" = "user_genome"))
# host_motus_METAT_HF <- host_prediction_genome %>% left_join(motus_MAGS_METAT_HF, by =c("Host.genome" = "user_genome"))
#the same for expression - host expression
#is it expression?????
motus_expression_long <- cbind(motus_df_METAT$mOTU, motus_expression) %>%
   pivot_longer(cols = starts_with("SAMEA"), names_to = "sampleid", values_to = "counts")
#keep only sampleids
motus_expression_long$sampleid <-gsub("_METAT", "", motus_expression_long$sampleid)
host_motus_METAT$sampleid <-gsub("_METAT", "", host_motus_METAT$sampleid)
#split into fractions
#FL
motus_expression_long_FL <- motus_expression_long %>% filter(sampleid %in% FL_accnum)
#LF
host_motus_expression_long_LF <- host_motus_expression_long %>% filter(sample_type == "Light_fraction") 
#HF
host_motus_expression_long_HF <- host_motus_expression_long %>% filter(sample_type == "Heavy_fraction") 
#change mOTU column name for compatibility
colnames(motus_expression_long)[1] <- "mOTU"
#change format for long one
# add metadata from analogous dataset for METAT without original counts 
host_motus_expression_long <- host_motus_METAT[, colnames(host_motus_METAT)[which(colnames(host_motus_METAT)!="counts")]] %>% inner_join(motus_expression_long, by=c("mOTU", "sampleid"))
#FL
host_motus_expression_long_FL <- host_motus_expression_long %>% filter(sample_type == "Free_living")
#LF
host_motus_expression_long_LF <- host_motus_expression_long %>% filter(sample_type == "Light_fraction") 
#HF
host_motus_expression_long_HF <- host_motus_expression_long %>% filter(sample_type == "Heavy_fraction") 
#after selecting hosts reformat to wide
#FL
host_motus_expression_FL_wide <-  host_motus_expression_long_FL %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 
ncol(host_motus_expression_FL_wide) #19 18 samples and mOTU - good!

#FL
motus_expression_FL_wide <-  motus_expression_long_FL %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) ### 
#sanity check
ncol(motus_expression_FL_wide) #19 18 sampls and mOTU - good!
#LF
motus_expression_LF_wide <-  motus_expression_long_LF %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 
#HF
motus_expression_HF_wide <-  motus_expression_long_HF %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 
#reformat so distances can be calculated
host_motus_METAT_wide <-  host_motus_METAT %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 
host_motus_METAT_FL_wide <-  host_motus_METAT_FL %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0)
host_motus_METAT_LF_wide <-  host_motus_METAT_LF %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0)
host_motus_METAT_HF_wide <-  host_motus_METAT_HF %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0)

```


```{r load viral diversity and activity with iphop results}
#read counts according to gene markers 
geneabund_drep_marker_taxa <- readRDS("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/geneabund_drep_marker_taxa.RDS")
viral_diversity <- geneabund_drep_marker_taxa %>%
  select(contains("METAG")) %>% select(contains(sampleid_METAG$BioSample)) #some METAG excluded from analysis - further selection by vector with metagenomes samples from otus tables (where samples were preselected)
viral_activity <- geneabund_drep_marker_taxa %>%
  select(contains("METAT"))
viral_diversity_paired <- viral_diversity %>% select(contains(paired_METAG))
viral_activity_paired <- viral_activity %>% select(contains(paired_METAT))
#normalisation with pseudocounts
pseudocount <- 0.000001
viral_expression <- log(viral_activity_paired + pseudocount) - log(viral_diversity_paired + pseudocount)
FL_accnum <- metadata$AccessionNumber_TBDSven[which(metadata$Sample_type=="Free_living")] 
LF_accnum <- metadata$AccessionNumber_TBDSven[which(metadata$Sample_type=="Light_fraction")] 
HF_accnum <- metadata$AccessionNumber_TBDSven[which(metadata$Sample_type=="Heavy_fraction")] 
# viral_diversity_FL <- geneabund_drep_marker_taxa %>%
#   select(contains("METAG")) %>% select(contains(sampleid_METAG$BioSample)) %>% select(contains(FL_accnum))
# viral_diversity_LF <- geneabund_drep_marker_taxa %>%
#   select(contains("METAG")) %>% select(contains(sampleid_METAG$BioSample)) %>% select(contains(LF_accnum))
# viral_diversity_HF <- geneabund_drep_marker_taxa %>%
#   select(contains("METAG")) %>% select(contains(sampleid_METAG$BioSample)) %>% select(contains(HF_accnum))
# #some METAG excluded from analysis - further selection by vector with metagenomes samples from otus tables (where samples were preselected) and by sampleid od the FL fraction 
# #why only 55 - ? would it be ok later? when accnum vector has length of 65 - what are mission columns
# viral_activity_FL <- geneabund_drep_marker_taxa %>%
#   select(contains("METAT")) %>% select(contains(FL_accnum)) #select by the accession number of the fl samples
#FL
viral_expression_FL <- viral_expression %>% select(contains(FL_accnum)) #select by the accession number of the fl samples
#LF
viral_expression_LF <- viral_expression %>% select(contains(LF_accnum)) #select by the accession number of the lf samples
#HF
viral_expression_HF <- viral_expression %>% select(contains(HF_accnum)) #select by the accession number of the hf samples
#viral expression has less samples the respective mostus and hosts 
#check which are not present 
colnames(motus_df_METAT_FL)
colnames(host_motus_METAT_FL_wide)
viral_expression_FL_samplesid <-gsub("GROS22.2_", "", colnames(viral_expression_FL))
viral_expression_FL_samplesid <-gsub(".genecount.profile", "", viral_expression_FL_samplesid)
#are motus and host motus the same
colnames(motus_df_METAT_FL) == colnames(host_motus_METAT_FL_wide) #yes
viral_expression_FL_samplesid %in% colnames(motus_df_METAT_FL) #all from viruses are present in motus
#which samples are not present in motus
absent <- colnames(motus_df_METAT_FL)[which(!(colnames(motus_df_METAT_FL) %in% viral_expression_FL_samplesid))] absent %in% paired_METAT #motus and host probably not expression but transcription we have to fix that!!!
```

```{r distances}
#calculate distance matrixes for all samples
#param distances
#store in a list of distances 

list.param.df.dist.METAG <- list()
list.param.df.dist.METAT <- list()


metadata_METAG <- metadata %>%
  filter(AccessionNumber_TBDSven %in% sampleid_METAG$BioSample)
metadata_METAT <- metadata %>%
  filter(AccessionNumber_TBDSven %in% sampleid_METAT$BioSample) 
j=0
z=0
for (i in c(10:39, 43:45)) { #columns with parameters
  metadata_METAG[,i] <- as.numeric(sub(",", ".", metadata_METAG[,i])) #to double

    if (sum(!is.na(metadata_METAG[,i]))>0){ #if most is NA do not calculate distance matrixes #we can later tray with most NA (sum(!is.na(metadata_METAG[,i]))/length(metadata_METAG[,i]))>0.5 
  j=j+1
  dist_METAG <- dist(metadata_METAG[,i], method = "euclidean")
  list.param.df.dist.METAG[[j]] <- dist_METAG  #append to list of df
  names(list.param.df.dist.METAG)[j] <- colnames(metadata_METAG)[i]
   }

  metadata_METAT[,i] <- as.numeric(sub(",", ".", metadata_METAT[,i])) #to double

  if (sum(!is.na(metadata_METAT[,i]))>0){#if only NA do not calculate distance matrixes
  z=z+1
  dist_METAT <- dist(metadata_METAT[,i], method = "euclidean")
  list.param.df.dist.METAT[[z]] <-  dist_METAT #append to list of df
  names(list.param.df.dist.METAT)[z] <- colnames(metadata_METAT)[i]
  
  }
} 
#we have to transpose matrixes as vegdist calculates distances between columns 
motus_df_METAG.dist <- vegdist(t(motus_df_METAG[,-1]), "euclidean")
motus_df_METAT.dist <- vegdist(t(motus_df_METAT[,-1]), "euclidean")
motus_df_METAT_FL.dist <- vegdist(t(motus_df_METAT_FL[,-1]), "euclidean")
motus_df_METAT_LF.dist <- vegdist(t(motus_df_METAT_LF[,-1]), "euclidean")
motus_df_METAT_HF.dist <- vegdist(t(motus_df_METAT_HF[,-1]), "euclidean")


host_motus_METAG_wide.dist <- vegdist(t(host_motus_METAG_wide[,-1]), "euclidean")
host_motus_METAT_wide.dist <- vegdist(t(host_motus_METAT_wide[,-1]), "euclidean")
host_motus_METAT_FL_wide.dist <- vegdist(t(host_motus_METAT_FL_wide[,-1]), "euclidean")
host_motus_METAT_LF_wide.dist <- vegdist(t(host_motus_METAT_LF_wide[,-1]), "euclidean")
host_motus_METAT_HF_wide.dist <- vegdist(t(host_motus_METAT_HF_wide[,-1]), "euclidean")


viral_diversity.dist <- vegdist(t(viral_diversity), "euclidean")
viral_activity.dist <- vegdist(t(viral_activity), "euclidean")
#put all distance matrixes into two lists for METAG and METAT respectively
list.METAG.dists <- list()
list.METAG.dists <- append(list.METAG.dists, list.param.df.dist.METAG)
list.METAG.dists[[length(list.METAG.dists)+1]] <- motus_df_METAG.dist
names(list.METAG.dists)[length(list.METAG.dists)] <- "MOTUs"
list.METAG.dists[[length(list.METAG.dists)+1]] <- viral_diversity.dist
names(list.METAG.dists)[length(list.METAG.dists)] <- "viral_diversity"
list.METAG.dists[[length(list.METAG.dists)+1]] <- host_motus_METAG_wide.dist
names(list.METAG.dists)[length(list.METAG.dists)] <- "hosts_MOTUs"


#the same for METAT
list.METAT.dists <- list()
list.METAT.dists <- append(list.METAT.dists,list.param.df.dist.METAT)
list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT.dist
names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs"


list.METAT.dists[[length(list.METAT.dists)+1]] <- viral_activity.dist
names(list.METAT.dists)[length(list.METAT.dists)] <- "viral_activity"
list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_wide.dist
names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_MOTUs"
#different number of samples - needs to be added to another sistance list - individual for each fraction -start from FL
# list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT_FL.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs_FL"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT_LF.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs_LF"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT_HF.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs_HF"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_FL_wide.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_FL_MOTUs"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_LF_wide.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_LF_MOTUs"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_HF_wide.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_HF_MOTUs"


#save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```
```{r distances FL}
#calculate distance matrixes for all samples
#param distances
#store in a list of distances 

list.param.df.dist.METAG_FL <- list()
list.param.df.dist.METAT_FL <- list()


metadata_METAG_FL <- metadata %>%
  filter(AccessionNumber_TBDSven %in% sampleid_METAG$BioSample & AccessionNumber_TBDSven %in% FL_accnum)
metadata_METAT_FL <- metadata %>%
  filter(AccessionNumber_TBDSven %in% paired_sampleid & AccessionNumber_TBDSven %in% FL_accnum) 
j=0
z=0
vec_param <- which(colnames(metadata_METAG_FL) %in%  c("Sat_O2_TBDHereon", "Temperature_TBDHereon", "Salinity_TBDHereon", "Turbidity_TBDHereon", "pH_TBDHereon", "O2_TBDHereon", "SPM_mgperL", "DOC_mg.L", "TN_mg.L", "DIC_mg.L", "Silicate_mg.L", "Ammonium_mg.L", "Nitrite_mg.L", "Nitrate_mg.L", "Total_DIN_µM", "SRP_mgperL", "TotalDissolvedPhosphate_mg.L", "Ammonium_µM", "Phosphate_µM", "RespirationRate_O2ug.L.h", "POC_mgperL", "PTC_mgperL", "PTN_mgperL", "PTH_mgperL", "dCH4_nM",           "Chla_ug/l", "dCO2_uM"))

for (i in vec_param) { #columns with parameters
  metadata_METAG_FL[,i] <- as.numeric(sub(",", ".", metadata_METAG_FL[,i])) #to double

    if (sum(!is.na(metadata_METAG_FL[,i]))>0){ #if most is NA do not calculate distance matrixes #we can later tray with most NA (sum(!is.na(metadata_METAG[,i]))/length(metadata_METAG[,i]))>0.5 
  j=j+1
  dist_METAG_FL <- dist(metadata_METAG_FL[,i], method = "euclidean")
  list.param.df.dist.METAG_FL[[j]] <- dist_METAG_FL  #append to list of df
  names(list.param.df.dist.METAG_FL)[j] <- colnames(metadata_METAG_FL)[i]
   }

  metadata_METAT_FL[,i] <- as.numeric(sub(",", ".", metadata_METAT_FL[,i])) #to double

  if (sum(!is.na(metadata_METAT_FL[,i]))>0){#if only NA do not calculate distance matrixes
  z=z+1
  dist_METAT_FL <- dist(metadata_METAT_FL[,i], method = "euclidean")
  list.param.df.dist.METAT_FL[[z]] <-  dist_METAT_FL #append to list of df
  names(list.param.df.dist.METAT_FL)[z] <- colnames(metadata_METAT_FL)[i]
  
  }
} 
#we have to transpose matrixes as vegdist calculates distances between columns 
#motus_df_METAG.dist <- vegdist(t(motus_df_METAG_FL[,-1]), "euclidean")
#motus_df_METAT.dist <- vegdist(t(motus_df_METAT_FL[,-1]), "euclidean")
motus_df_METAG_FL.dist <- vegdist(t(motus_df_METAG_FL[,-1]), "euclidean")
motus_df_METAT_FL.dist <- vegdist(t(motus_df_METAT_FL[,-1]), "euclidean")
motus_df_METAT_LF.dist <- vegdist(t(motus_df_METAT_LF[,-1]), "euclidean")
motus_df_METAT_HF.dist <- vegdist(t(motus_df_METAT_HF[,-1]), "euclidean")
motus_expression_FL_wide.dist <- vegdist(t(motus_expression_FL_wide[,-1]), "euclidean")
#host_motus_METAG_wide.dist <- vegdist(t(host_motus_METAG_wide[,-1]), "euclidean")
#host_motus_METAT_wide.dist <- vegdist(t(host_motus_METAT_wide[,-1]), "euclidean")
host_motus_METAT_FL_wide.dist <- vegdist(t(host_motus_METAT_FL_wide[,-1]), "euclidean")
host_motus_METAG_FL_wide.dist <- vegdist(t(host_motus_METAG_FL_wide[,-1]), "euclidean")
host_motus_METAT_LF_wide.dist <- vegdist(t(host_motus_METAT_LF_wide[,-1]), "euclidean")
host_motus_METAT_HF_wide.dist <- vegdist(t(host_motus_METAT_HF_wide[,-1]), "euclidean")
host_motus_expression_FL_wide.dist <- vegdist(t(host_motus_expression_FL_wide[,-1]), na.rm = T, "euclidean")

#viral_diversity.dist <- vegdist(t(viral_diversity), "euclidean")
#viral_activity.dist <- vegdist(t(viral_activity), "euclidean")
viral_diversity_FL.dist <- vegdist(t(viral_diversity_FL), "euclidean")
viral_expression_FL.dist <- vegdist(t(viral_expression_FL), "euclidean")
#viral_activity_FL.dist <- vegdist(t(viral_activity_FL), "euclidean")
#put all distance matrixes into two lists for METAG and METAT respectively
list.METAG_FL.dists <- list()
list.METAG_FL.dists <- append(list.METAG_FL.dists, list.param.df.dist.METAG_FL)
list.METAG_FL.dists[[length(list.METAG_FL.dists)+1]] <- motus_df_METAG_FL.dist
names(list.METAG_FL.dists)[length(list.METAG_FL.dists)] <- "MOTUs"
list.METAG_FL.dists[[length(list.METAG_FL.dists)+1]] <- viral_diversity_FL.dist
names(list.METAG_FL.dists)[length(list.METAG_FL.dists)] <- "viral_diversity"
list.METAG_FL.dists[[length(list.METAG_FL.dists)+1]] <- host_motus_METAG_FL_wide.dist
names(list.METAG_FL.dists)[length(list.METAG_FL.dists)] <- "hosts_MOTUs"


#the same for METAT
list.METAT_FL.dists <- list()
list.METAT_FL.dists <- append(list.METAT_FL.dists,list.param.df.dist.METAT_FL)
#list.METAT_FL.dists[[length(list.METAT_FL.dists)+1]] <- motus_df_METAT_FL.dist
#names(list.METAT_FL.dists)[length(list.METAT_FL.dists)] <- "MOTUs"
# list.METAT_FL.dists[[length(list.METAT_FL.dists)+1]] <- viral_activity_FL.dist
# names(list.METAT_FL.dists)[length(list.METAT_FL.dists)] <- "viral_activity"
# list.METAT_FL.dists[[length(list.METAT_FL.dists)+1]] <- host_motus_METAT_FL_wide.dist
# names(list.METAT_FL.dists)[length(list.METAT_FL.dists)] <- "hosts_MOTUs"
list.METAT_FL.dists[[length(list.METAT_FL.dists)+1]] <- motus_expression_FL.dist
names(list.METAT_FL.dists)[length(list.METAT_FL.dists)] <- "MOTUs"
list.METAT_FL.dists[[length(list.METAT_FL.dists)+1]] <- viral_expression_FL.dist
names(list.METAT_FL.dists)[length(list.METAT_FL.dists)] <- "viral_expression"
list.METAT_FL.dists[[length(list.METAT_FL.dists)+1]] <- host_motus_expression_FL_wide.dist
names(list.METAT_FL.dists)[length(list.METAT_FL.dists)] <- "hosts_MOTUs"
#different number of samples - needs to be added to another sistance list - individual for each fraction -start from FL
# list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT_FL.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs_FL"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT_LF.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs_LF"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT_HF.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs_HF"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_FL_wide.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_FL_MOTUs"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_LF_wide.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_LF_MOTUs"
# list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_HF_wide.dist
# names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_HF_MOTUs"


#save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```

```{r run mentel test}
#check how many METAG and METAT samples
#find a file with sampleid and type (METAG or METAT)
#check it this agrees with number of columns in distance matrixes
list.METAG.dists.length <- length(list.METAG.dists)
list.METAT.dists.length <- length(list.METAT.dists)

list.dist.length <- length(list.METAG.dists)
matrix_res_stat_METAG <- matrix(nrow = list.METAG.dists.length, ncol = list.METAG.dists.length)
colnames(matrix_res_stat_METAG) <- c(names(list.METAG.dists))
matrix_res_stat_METAT <- matrix(nrow = list.METAT.dists.length, ncol = list.METAT.dists.length)
colnames(matrix_res_stat_METAT) <- c(names(list.METAT.dists))
matrix_res_sig_METAG <- matrix(nrow = list.METAG.dists.length, ncol = list.METAG.dists.length)
colnames(matrix_res_sig_METAG) <- c(names(list.METAG.dists))
matrix_res_sig_METAT <- matrix(nrow = list.METAT.dists.length, ncol = list.METAT.dists.length)
colnames(matrix_res_sig_METAT) <- c(names(list.METAT.dists))

for (i in c(1:list.dist.length)) {
  for (j in c(1:list.dist.length)) {
    #we dont want to calculate mentel between parameters or the same matrixes 
      if ((i <= length(list.param.df.dist.METAG) & j <= length(list.param.df.dist.METAG)) | (i>=j)){
       next
    }
    dist.matrix <- list.METAG.dists[[i]]
    dist.matrix2 <- list.METAG.dists[[j]]
    res <- mantel(dist.matrix, dist.matrix2, method = "spearman", permutations = 9999, na.rm = TRUE)
    matrix_res_sig_METAG[i,j] <- res$signif
    matrix_res_stat_METAG[i,j] <- res$statistic
  }
}

for (i in c(1:list.dist.length)) {
  for (j in c(1:list.dist.length)) {

    #we dont want to calculate mentel between parameters or the same matrixes 

  if ((i <= length(list.param.df.dist.METAT) & j <= length(list.param.df.dist.METAT)) | (i>=j) | j>list.METAT.dists.length){
       next
  }
    dist.matrix <- list.METAT.dists[[i]]
    dist.matrix2 <- list.METAT.dists[[j]]
    res <- mantel(dist.matrix, dist.matrix2, method = "spearman", permutations = 9999, na.rm = TRUE)
    matrix_res_sig_METAT[i,j] <- res$signif
    matrix_res_stat_METAT[i,j] <- res$statistic
  }
}

#save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```

```{r run mentel test FL}
#check how many METAG and METAT samples
#find a file with sampleid and type (METAG or METAT)
#check it this agrees with number of columns in distance matrixes
list.METAG_FL.dists.length <- length(list.METAG_FL.dists)
list.METAT_FL.dists.length <- length(list.METAT_FL.dists)

list.dist.length <- length(list.METAG_FL.dists)
matrix_res_stat_METAG_FL <- matrix(nrow = list.METAG_FL.dists.length, ncol = list.METAG_FL.dists.length)
colnames(matrix_res_stat_METAG_FL) <- c(names(list.METAG_FL.dists))
matrix_res_stat_METAT_FL <- matrix(nrow = list.METAT_FL.dists.length, ncol = list.METAT_FL.dists.length)
colnames(matrix_res_stat_METAT_FL) <- c(names(list.METAT_FL.dists))
matrix_res_sig_METAG_FL <- matrix(nrow = list.METAG_FL.dists.length, ncol = list.METAG_FL.dists.length)
colnames(matrix_res_sig_METAG_FL) <- c(names(list.METAG_FL.dists))
matrix_res_sig_METAT_FL <- matrix(nrow = list.METAT_FL.dists.length, ncol = list.METAT_FL.dists.length)
colnames(matrix_res_sig_METAT_FL) <- c(names(list.METAT_FL.dists))

for (i in c(1:list.dist.length)) {
  for (j in c(1:list.dist.length)) {
    #we dont want to calculate mentel between parameters or the same matrixes 
      if ((i <= length(list.param.df.dist.METAG_FL) & j <= length(list.param.df.dist.METAG_FL)) | (i>=j)){
       next
    }
    dist.matrix <- list.METAG_FL.dists[[i]]
    dist.matrix2 <- list.METAG_FL.dists[[j]]
    res <- mantel(dist.matrix, dist.matrix2, method = "spearman", permutations = 9999, na.rm = TRUE)
    matrix_res_sig_METAG_FL[i,j] <- res$signif
    matrix_res_stat_METAG_FL[i,j] <- res$statistic
  }
}
#METAT
for (i in c(1:list.dist.length)) {
  for (j in c(1:list.dist.length)) {

    #we dont want to calculate mentel between parameters or the same matrixes 

  if ((i <= length(list.param.df.dist.METAT_FL) & j <= length(list.param.df.dist.METAT_FL)) | (i>=j) | j>list.METAT_FL.dists.length){
       next
  }
    dist.matrix <- list.METAT_FL.dists[[i]]
    dist.matrix2 <- list.METAT_FL.dists[[j]]
    res <- mantel(dist.matrix, dist.matrix2, method = "spearman", permutations = 9999, na.rm = TRUE)
    matrix_res_sig_METAT_FL[i,j] <- res$signif
    matrix_res_stat_METAT_FL[i,j] <- res$statistic
  }
}


save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```


```{r}
#load("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")

#Reduce matrixes to rows where at least one R value is greater then 0.2 0 tiles color by significance 
#add colnames to rows to read them easier 
rownames(matrix_res_stat_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_sig_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_stat_METAG) <- colnames(matrix_res_stat_METAG)
rownames(matrix_res_sig_METAG) <- colnames(matrix_res_sig_METAG)
#we have reformat to long version
#but first lets filter all with R statistic higher then 0.2
exist_greater_02 <- function(vec) {
 sum(vec>0.2, na.rm=TRUE)>0
}
#adjust p-values
rows_for_plot_METAT <- apply(matrix_res_stat_METAT,1,exist_greater_02)
#and take last 3 columns
matrix_res_stat_METAT_plot <- matrix_res_stat_METAT[rows_for_plot_METAT,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
matrix_res_sig_METAT_plot <- matrix_res_sig_METAT[rows_for_plot_METAT,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
library(reshape2)

matrix_res_stat_METAT_plot <- melt(matrix_res_stat_METAT_plot, value.name = "R_statistic" )
matrix_res_sig_METAT_plot <- melt(matrix_res_sig_METAT_plot, value.name = "p_value" )

matrix_res_stat_METAT_plot <- left_join(matrix_res_stat_METAT_plot, matrix_res_sig_METAT_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAT_plot$data <- rep(c("Metatranscriptome"), time= length(matrix_res_stat_METAT_plot[,1]))

#prepare the same for METAG 
rows_for_plot_METAG <- apply(matrix_res_stat_METAG,1,exist_greater_02)
#and take last 3 columns
matrix_res_stat_METAG_plot <- matrix_res_stat_METAG[rows_for_plot_METAG,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]
matrix_res_sig_METAG_plot <- matrix_res_sig_METAG[rows_for_plot_METAG,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]

matrix_res_stat_METAG_plot <- melt(matrix_res_stat_METAG_plot, value.name = "R_statistic")
matrix_res_sig_METAG_plot <- melt(matrix_res_sig_METAG_plot, value.name = "p_value")

matrix_res_stat_METAG_plot <- left_join(matrix_res_stat_METAG_plot, matrix_res_sig_METAG_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAG_plot$data <- rep(c("Metagenome"), time=length(matrix_res_stat_METAG_plot[,1]))


matrix_res_stat_mentel_plot <- rbind(matrix_res_stat_METAG_plot, matrix_res_stat_METAT_plot) 
#save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
saveRDS(matrix_res_stat_mental_plot,"C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/matrix_res_stat_mental_plot.RDS")
```

```{r}

#Adjust_p_values
#add colnames to rows to read them easier 
rownames(matrix_res_stat_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_sig_METAT) <- colnames(matrix_res_sig_METAT)
rownames(matrix_res_stat_METAG) <- colnames(matrix_res_stat_METAG)
rownames(matrix_res_sig_METAG) <- colnames(matrix_res_sig_METAG)
#we have reformat to long version
#but first lets filter all with R statistic higher then 0.2

#adjust p-values
#and take last 3 columns
matrix_res_stat_METAT_plot <- matrix_res_stat_METAT[,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
matrix_res_sig_METAT_plot <- matrix_res_sig_METAT[,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]

library(reshape2)

matrix_res_stat_METAT_plot <- melt(matrix_res_stat_METAT_plot, value.name = "R_statistic" )
matrix_res_sig_METAT_plot <- melt(matrix_res_sig_METAT_plot, value.name = "p_value" )
#filter out counfouded values - all which have Mol unit
matrix_res_stat_METAT_plot <- matrix_res_stat_METAT_plot %>%  filter(!grepl("µM|uM", Var1)) 
matrix_res_stat_METAT_plot <- matrix_res_stat_METAT_plot %>%  filter(!grepl("µM|uM", Var1)) 
#now we can do p-value adjust
matrix_res_sig_METAT_plot$p_value.adj <- p.adjust(matrix_res_sig_METAT_plot$p_value, method="hochberg")

#p_value.adj_FL <- matrix_res_sig_METAT_plot %>% filter(Var2 =="MOTUs_FL") #p.adjust(matrix_res_sig_METAT_plot$p_value, method="hochberg")

matrix_res_stat_METAT_plot <- left_join(matrix_res_stat_METAT_plot, matrix_res_sig_METAT_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAT_plot$data <- rep(c("Metatranscriptome"), time= length(matrix_res_stat_METAT_plot[,1]))

#prepare the same for METAG 
#adjust p-values
#and take last 3 columns
matrix_res_stat_METAG_plot <- matrix_res_stat_METAG[,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]
matrix_res_sig_METAG_plot <- matrix_res_sig_METAG[,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]

matrix_res_stat_METAG_plot <- melt(matrix_res_stat_METAG_plot, value.name = "R_statistic")
matrix_res_sig_METAG_plot <- melt(matrix_res_sig_METAG_plot, value.name = "p_value")
#filter out counfouded values - all which have Mol unit
matrix_res_stat_METAG_plot <- matrix_res_stat_METAG_plot %>%  filter(!grepl("µM|uM", Var1)) 
matrix_res_stat_METAG_plot <- matrix_res_stat_METAG_plot %>%  filter(!grepl("µM|uM", Var1)) 
#now we can do p-value adjust
matrix_res_sig_METAG_plot$p_value.adj <- p.adjust(matrix_res_sig_METAG_plot$p_value, method="hochberg")

matrix_res_stat_METAG_plot <- left_join(matrix_res_stat_METAG_plot, matrix_res_sig_METAG_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAG_plot$data <- rep(c("Metagenome"), time=length(matrix_res_stat_METAG_plot[,1]))

matrix_res_stat_mentel_plot <- rbind(matrix_res_stat_METAG_plot, matrix_res_stat_METAT_plot) 
#save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```

```{r adjust p-values FL}

#Adjust_p_values
#add colnames to rows to read them easier 
rownames(matrix_res_stat_METAT_FL) <- colnames(matrix_res_stat_METAT_FL)
rownames(matrix_res_sig_METAT_FL) <- colnames(matrix_res_stat_METAT_FL)
rownames(matrix_res_stat_METAG_FL) <- colnames(matrix_res_stat_METAG_FL)
rownames(matrix_res_sig_METAG_FL) <- colnames(matrix_res_sig_METAG_FL)
#we have reformat to long version
#but first lets filter all with R statistic higher then 0.2

#adjust p-values
#and take last 3 columns
matrix_res_stat_METAT_FL_plot <- matrix_res_stat_METAT_FL[,c((length(matrix_res_stat_METAT_FL[1,])-2):length(matrix_res_stat_METAT_FL[1,]))]
matrix_res_sig_METAT_FL_plot <- matrix_res_sig_METAT_FL[,c((length(matrix_res_stat_METAT_FL[1,])-2):length(matrix_res_stat_METAT_FL[1,]))]
library(reshape2)

matrix_res_stat_METAT_FL_plot <- melt(matrix_res_stat_METAT_FL_plot, value.name = "R_statistic" )
matrix_res_sig_METAT_FL_plot <- melt(matrix_res_sig_METAT_FL_plot, value.name = "p_value" )
#lets remove the confonded values
matrix_res_sig_METAT_FL_plot$p_value.adj <- p.adjust(matrix_res_sig_METAT_FL_plot$p_value, method="hochberg")

#p_value.adj_FL <- matrix_res_sig_METAT_FL_plot %>% filter(Var2 =="MOTUs_FL") p.adjust(matrix_res_sig_METAT_FL_plot$p_value, method="hochberg")

matrix_res_stat_METAT_FL_plot <- left_join(matrix_res_stat_METAT_FL_plot, matrix_res_sig_METAT_FL_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAT_FL_plot$data <- rep(c("Metatranscriptome"), time= length(matrix_res_stat_METAT_FL_plot[,1]))

#prepare the same for METAG 
#adjust p-values
#and take last 3 columns
matrix_res_stat_METAG_FL_plot <- matrix_res_stat_METAG_FL[,c((length(matrix_res_stat_METAG_FL[1,])-2):length(matrix_res_stat_METAG_FL[1,]))]
matrix_res_sig_METAG_FL_plot <- matrix_res_sig_METAG_FL[,c((length(matrix_res_stat_METAG_FL[1,])-2):length(matrix_res_stat_METAG_FL[1,]))]

matrix_res_stat_METAG_FL_plot <- melt(matrix_res_stat_METAG_FL_plot, value.name = "R_statistic")
matrix_res_sig_METAG_FL_plot <- melt(matrix_res_sig_METAG_FL_plot, value.name = "p_value")
matrix_res_sig_METAG_FL_plot$p_value.adj <- p.adjust(matrix_res_sig_METAG_FL_plot$p_value, method="hochberg")

matrix_res_stat_METAG_FL_plot <- left_join(matrix_res_stat_METAG_FL_plot, matrix_res_sig_METAG_FL_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAG_FL_plot$data <- rep(c("Metagenome"), time=length(matrix_res_stat_METAG_FL_plot[,1]))

matrix_res_stat_FL_mentel_plot <- rbind(matrix_res_stat_METAG_FL_plot, matrix_res_stat_METAT_FL_plot) 
#save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```


```{r}
#common plot for both - quite hard to read -  when adjusted p-values
  mentel_test_plot <- matrix_res_stat_mentel_plot %>%
  filter(R_statistic != "NA") %>% 
  ggplot(aes(x=Var1, y=Var2)) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  facet_grid( scales = "free", rows = vars(data)) +
 # coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=1.1, vjust=1.1), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value.adj) <= .05)+1], c("","*")[(abs(p_value.adj) <= .001)+1], c("","*")[(abs(p_value.adj) <= .0001)+1])))
#show
mentel_test_plot
```

```{r}
# plot for METAG - use normal p-values as significant 
  mentel_test_METAG_plot <- matrix_res_stat_mentel_plot %>%
  filter(R_statistic != "NA") %>% 
  filter(data == "Metagenome") %>% 
  filter(Var1 != "MOTUs") %>% 
  filter(!grepl("µM|uM", Var1)) %>% #this should be done earlier - confounding adjusting p-values 
  mutate(Var1=str_remove(Var1, "_TBDHereon|_mh.L|_mgperL|_mg.L|_O2ug.L.h|_nM|_ug/l")) %>%
  ggplot(aes(x=factor(Var1), y=Var2)) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=0.95,vjust=0.2), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value.adj) <= .05)+1], c("","*")[(abs(p_value.adj) <= .001)+1], c("","*")[(abs(p_value.adj) <= .0001)+1])))
#show
mentel_test_METAG_plot
```
```{r METAG plot FL}
# plot for METAG - use normal p-values as significant 
  mentel_test_METAG_FL_plot <- matrix_res_stat_FL_mentel_plot %>%
  filter(R_statistic != "NA") %>% 
  filter(data == "Metagenome") %>% 
  filter(Var1 != "MOTUs") %>% 
 # filter(!grepl("µM|uM", Var1)) %>% 
  mutate(Var1=str_remove(Var1, "_TBDHereon|_mh.L|_mgperL|_mg.L|_O2ug.L.h|_nM|_ug/l")) %>%
  ggplot(aes(x=factor(Var1), y=Var2)) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=0.95,vjust=0.2), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value.adj) <= .05)+1], c("","*")[(abs(p_value.adj) <= .001)+1], c("","*")[(abs(p_value.adj) <= .0001)+1])))
#show
mentel_test_METAG_FL_plot
```

```{r}
# plot for METAT - use normal p -values as significant 
  mentel_test_METAT_plot <- matrix_res_stat_mentel_plot %>%
  filter(R_statistic != "NA") %>% 
  filter(data == "Metatranscriptome") %>% 
  filter(Var1 != "MOTUs") %>% 
  filter(!grepl("µM|uM", Var1)) %>% 
  mutate(Var1=str_remove(Var1, "_TBDHereon|_mh.L|_mgperL|_mg.L|_O2ug.L.h|_nM|_ug/l")) %>%
  ggplot(aes(x=Var1, y=factor(Var2, levels=c("MOTUs", "viral_activity", "hosts_MOTUs")))) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=0.95, vjust=0.2), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value.adj) <= .05)+1], c("","*")[(abs(p_value.adj) <= .001)+1], c("","*")[(abs(p_value.adj) <= .001)+1])))

```

```{r METAT plot FL}
# plot for METAT - use normal p -values as significant 
  mentel_test_METAT_FL_plot <- matrix_res_stat_FL_mentel_plot %>%
  filter(R_statistic != "NA") %>% 
  filter(data == "Metatranscriptome") %>% 
  filter(Var1 != "MOTUs") %>% 
  filter(!grepl("µM|uM", Var1)) %>% 
  mutate(Var1=str_remove(Var1, "_TBDHereon|_mh.L|_mgperL|_mg.L|_O2ug.L.h|_nM|_ug/l")) %>%
  ggplot(aes(x=Var1, y=factor(Var2, levels=c("MOTUs", "viral_activity", "hosts_MOTUs")))) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=0.95, vjust=0.2), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value.adj) <= .05)+1], c("","*")[(abs(p_value.adj) <= .001)+1], c("","*")[(abs(p_value.adj) <= .001)+1])))
mentel_test_METAT_FL_plot
#better - more significant but strange plot NA and not all comparisons included
```