---
title: "mental_test_Elbe"
author: "Julia Golebiowska"
date: "2024-04-29"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library(dplyr)
library(tidyverse)
library(ggplot2)
library(vegan)
setwd("~/IGB_phd/virus/viral_abundances")
```
```{r load metaadata}
##Load Metadata for later
path = "C:/Users/jgolebiowska/Documents/IGB_phd/workshop_gene_catalog/"

#combine metadata 
metadata_sim <- read.csv(paste0(path,"/SAMEAID_SampleID_simplified.csv"), header=TRUE, sep=";") 
#select the needed columns to avoid doubling computations
metadata_sim <-metadata_sim[,c("BioSample", "ProjectID")]
#read simplified metadata, as it has a ProjectID 
metadata <- read.csv(paste0(path,"/SAMEAID_SampleID.csv"), header=TRUE, sep=";") %>% 
  left_join(metadata_sim, by = "BioSample")  

```

```{r load motus}
path = "C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/MOTUs/"
motus_final_taxa_metadata <- readRDS(file=paste0(path,"motus_final_taxa_metadata.RDS"))
```

```{r edit motus}
motus_final_taxa_metadata <- motus_final_taxa_metadata %>% 
  filter(!grepl("Bunthaus|Kollmar|Seemans", station), !grepl("Nov 21", date))

motus_df_METAG <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()
# only metatranscriptomes
motus_df_METAT <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
#once again distinct to make it works
  select(mOTU, sampleid, counts) %>%
  distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) %>% 
  as.data.frame()
## for later keep names of samples which were used for metagenomes and metatranscriptomes and finaly use for th analysis  
sampleid_METAG <- unique(motus_final_taxa_metadata[which(motus_final_taxa_metadata$data_type=="METAG"), "BioSample"])
sampleid_METAT <- unique(motus_final_taxa_metadata[which(motus_final_taxa_metadata$data_type=="METAT"), "BioSample"])
```

```{r hosts motus }
# select motus which are assigned hosts
#load hosts assignment
host_prediction_genome = read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/Host_prediction_to_genome_m90.csv")
motu_stats = readRDS(file=paste0(path,"motu_stats.RDS"))
#join Motus by MAGs which were assigned as hosts with iphop
motus_MAGS_METAG <- motus_final_taxa_metadata %>%
  filter(data_type == "METAG") %>%
  left_join(motu_stats, by = "mOTU")
host_motus_METAG <- host_prediction_genome %>% left_join(motus_MAGS_METAG, by =c("Host.genome" = "user_genome"))
#reformat so distances can be calculated
host_motus_METAG_wide <-  host_motus_METAG %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 
##### for METAT
#join Motus by MAGs which were assigned as hosts with iphop
motus_MAGS_METAT <- motus_final_taxa_metadata %>%
  filter(data_type == "METAT") %>%
  left_join(motu_stats, by = "mOTU")
host_motus_METAT <- host_prediction_genome %>% left_join(motus_MAGS_METAT, by =c("Host.genome" = "user_genome"))
#reformat so distances can be calculated
host_motus_METAT_wide <-  host_motus_METAT %>%
  filter(!is.na(mOTU))  %>%
  select(mOTU, sampleid, counts) %>%
   distinct() %>% 
  pivot_wider(names_from="sampleid", values_from="counts", values_fill=0) 

```


```{r load viral diversity and activity with iphop results}
#read counts according to gene markers 
geneabund_drep_marker_taxa <- readRDS("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/geneabund_drep_marker_taxa.RDS")

viral_diversity <- geneabund_drep_marker_taxa %>%
  select(contains("METAG")) %>% select(contains(sampleid_METAG$BioSample)) #some METAG excluded from analysis - further selection by vector with metagenomes samples from otus tables (where samples were preselected)
viral_activity <- geneabund_drep_marker_taxa %>%
  select(contains("METAT"))

```

```{r distances}
#calculate distance matrixes for all samples
#param distances
#store in a list of distances 

list.param.df.dist.METAG <- list()
list.param.df.dist.METAT <- list()


metadata_METAG <- metadata %>%
  filter(AccessionNumber_TBDSven %in% sampleid_METAG$BioSample)
metadata_METAT <- metadata %>%
  filter(AccessionNumber_TBDSven %in% sampleid_METAT$BioSample) 
j=0
z=0
for (i in c(10:39)) { 
  metadata_METAG[,i] <- as.numeric(sub(",", ".", metadata_METAG[,i])) #to double

    if (sum(!is.na(metadata_METAG[,i]))>0){ #if only NA do not calculate distance matrixes
  j=j+1
  dist_METAG <- dist(metadata_METAG[,i], method = "euclidean")
  list.param.df.dist.METAG[[j]] <- dist_METAG  #append to list of df
  names(list.param.df.dist.METAG)[j] <- colnames(metadata_METAG)[i]
   }

  metadata_METAT[,i] <- as.numeric(sub(",", ".", metadata_METAT[,i])) #to double

  if (sum(!is.na(metadata_METAT[,i]))>0){#if only NA do not calculate distance matrixes
  z=z+1
  dist_METAT <- dist(metadata_METAT[,i], method = "euclidean")
  list.param.df.dist.METAT[[z]] <-  dist_METAT #append to list of df
  names(list.param.df.dist.METAT)[z] <- colnames(metadata_METAT)[i]
  
  }
 
} 
#we have to transpose matrixes as vegdist calculates distances between columns 
motus_df_METAG.dist <- vegdist(t(motus_df_METAG[,-1]), "euclidean")
motus_df_METAT.dist <- vegdist(t(motus_df_METAT[,-1]), "euclidean")
host_motus_METAG_wide.dist <- vegdist(t(host_motus_METAG_wide[,-1]), "euclidean")
host_motus_METAT_wide.dist <- vegdist(t(host_motus_METAT_wide[,-1]), "euclidean")
viral_diversity.dist <- vegdist(t(viral_diversity), "euclidean")
viral_activity.dist <- vegdist(t(viral_activity), "euclidean")
#put all distance matrixes into two lists for METAG and METAT respectively
list.METAG.dists <- list()
list.METAG.dists <- append(list.METAG.dists, list.param.df.dist.METAG)
list.METAG.dists[[length(list.METAG.dists)+1]] <- motus_df_METAG.dist
names(list.METAG.dists)[length(list.METAG.dists)] <- "MOTUs"
list.METAG.dists[[length(list.METAG.dists)+1]] <- viral_diversity.dist
names(list.METAG.dists)[length(list.METAG.dists)] <- "viral_diversity"
list.METAG.dists[[length(list.METAG.dists)+1]] <- host_motus_METAG_wide.dist
names(list.METAG.dists)[length(list.METAG.dists)] <- "hosts_MOTUs"

#the same for METAT
list.METAT.dists <- list()
list.METAT.dists <- append(list.METAT.dists,list.param.df.dist.METAT)
list.METAT.dists[[length(list.METAT.dists)+1]] <- motus_df_METAT.dist
names(list.METAT.dists)[length(list.METAT.dists)] <- "MOTUs"
list.METAT.dists[[length(list.METAT.dists)+1]] <- viral_activity.dist
names(list.METAT.dists)[length(list.METAT.dists)] <- "viral_activity"
list.METAT.dists[[length(list.METAT.dists)+1]] <- host_motus_METAT_wide.dist
names(list.METAT.dists)[length(list.METAT.dists)] <- "hosts_MOTUs"


save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```

```{r run mental test}
#check how many METAG and METAT samples
#find a file with sampleid and type (METAG or METAT)
#check it this agrees with number of columns in distance matrixes
list.METAG.dists.length <- length(list.METAG.dists)
list.METAT.dists.length <- length(list.METAT.dists)

list.dist.length <- length(list.METAG.dists)
matrix_res_stat_METAG <- matrix(nrow = list.METAG.dists.length, ncol = list.METAG.dists.length)
colnames(matrix_res_stat_METAG) <- c(names(list.METAG.dists))
matrix_res_stat_METAT <- matrix(nrow = list.METAT.dists.length, ncol = list.METAT.dists.length)
colnames(matrix_res_stat_METAT) <- c(names(list.METAT.dists))
matrix_res_sig_METAG <- matrix(nrow = list.METAG.dists.length, ncol = list.METAG.dists.length)
colnames(matrix_res_sig_METAG) <- c(names(list.METAG.dists))
matrix_res_sig_METAT <- matrix(nrow = list.METAT.dists.length, ncol = list.METAT.dists.length)
colnames(matrix_res_sig_METAT) <- c(names(list.METAT.dists))

for (i in c(1:list.dist.length)) {
  for (j in c(1:list.dist.length)) {
    #we dont want to calculate mental between parameters or the same matrixes 

      if ((i <= length(list.param.df.dist.METAG) & j <= length(list.param.df.dist.METAG)) | (i>=j)){
       next
    }
  
    dist.matrix <- list.METAG.dists[[i]]
    dist.matrix2 <- list.METAG.dists[[j]]
    res <- mantel(dist.matrix, dist.matrix2, method = "spearman", permutations = 9999, na.rm = TRUE)
    matrix_res_sig_METAG[i,j] <- res$signif
    matrix_res_stat_METAG[i,j] <- res$statistic
  }
}

for (i in c(1:list.dist.length)) {
  for (j in c(1:list.dist.length)) {

    #we dont want to calculate mental between parameters or the same matrixes 

  if ((i <= length(list.param.df.dist.METAT) & j <= length(list.param.df.dist.METAT)) | (i>=j) | j>list.METAT.dists.length){
       next
    }
    dist.matrix <- list.METAT.dists[[i]]
    dist.matrix2 <- list.METAT.dists[[j]]
    res <- mantel(dist.matrix, dist.matrix2, method = "spearman", permutations = 9999, na.rm = TRUE)
    matrix_res_sig_METAT[i,j] <- res$signif
    matrix_res_stat_METAT[i,j] <- res$statistic
  }
}


save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")

```
```{r}
load("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")

#Reduce matrixes to rows where at least one R value is greater then 0.2 0 tiles color by significance 
#add colnames to rows to read them easier 
rownames(matrix_res_stat_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_sig_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_stat_METAG) <- colnames(matrix_res_stat_METAG)
rownames(matrix_res_sig_METAG) <- colnames(matrix_res_sig_METAG)
#we have reformat to long version
#but first lets filter all with R statistic higher then 0.2
exist_greater_02 <- function(vec) {
 sum(vec>0.2, na.rm=TRUE)>0
}
#adjust p-values
rows_for_plot_METAT <- apply(matrix_res_stat_METAT,1,exist_greater_02)
#and take last 3 columns
matrix_res_stat_METAT_plot <- matrix_res_stat_METAT[rows_for_plot_METAT,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
matrix_res_sig_METAT_plot <- matrix_res_sig_METAT[rows_for_plot_METAT,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
library(reshape2)

matrix_res_stat_METAT_plot <- melt(matrix_res_stat_METAT_plot, value.name = "R_statistic" )
matrix_res_sig_METAT_plot <- melt(matrix_res_sig_METAT_plot, value.name = "p_value" )

matrix_res_stat_METAT_plot <- left_join(matrix_res_stat_METAT_plot, matrix_res_sig_METAT_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAT_plot$data <- rep(c("Metatranscriptome"), time= length(matrix_res_stat_METAT_plot[,1]))

#prepare the same for METAG 
rows_for_plot_METAG <- apply(matrix_res_stat_METAG,1,exist_greater_02)
#and take last 3 columns
matrix_res_stat_METAG_plot <- matrix_res_stat_METAG[rows_for_plot_METAG,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]
matrix_res_sig_METAG_plot <- matrix_res_sig_METAG[rows_for_plot_METAG,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]

matrix_res_stat_METAG_plot <- melt(matrix_res_stat_METAG_plot, value.name = "R_statistic")
matrix_res_sig_METAG_plot <- melt(matrix_res_sig_METAG_plot, value.name = "p_value")

matrix_res_stat_METAG_plot <- left_join(matrix_res_stat_METAG_plot, matrix_res_sig_METAG_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAG_plot$data <- rep(c("Metagenome"), time=length(matrix_res_stat_METAG_plot[,1]))


matrix_res_stat_mental_plot <- rbind(matrix_res_stat_METAG_plot, matrix_res_stat_METAT_plot) 
save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")

```
```{r}

#Adjust_p_values
#add colnames to rows to read them easier 
rownames(matrix_res_stat_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_sig_METAT) <- colnames(matrix_res_stat_METAT)
rownames(matrix_res_stat_METAG) <- colnames(matrix_res_stat_METAG)
rownames(matrix_res_sig_METAG) <- colnames(matrix_res_sig_METAG)
#we have reformat to long version
#but first lets filter all with R statistic higher then 0.2
exist_greater_02 <- function(vec) {
 sum(vec>0.2, na.rm=TRUE)>0
}
#adjust p-values
#and take last 3 columns
matrix_res_stat_METAT_plot <- matrix_res_stat_METAT[,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
matrix_res_sig_METAT_plot <- matrix_res_sig_METAT[,c((length(matrix_res_stat_METAT[1,])-2):length(matrix_res_stat_METAT[1,]))]
library(reshape2)

matrix_res_stat_METAT_plot <- melt(matrix_res_stat_METAT_plot, value.name = "R_statistic" )
matrix_res_sig_METAT_plot <- melt(matrix_res_sig_METAT_plot, value.name = "p_value" )
matrix_res_sig_METAT_plot$p_value.adj <- p.adjust(matrix_res_sig_METAT_plot$p_value, method="hochberg")

matrix_res_stat_METAT_plot <- left_join(matrix_res_stat_METAT_plot, matrix_res_sig_METAT_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAT_plot$data <- rep(c("Metatranscriptome"), time= length(matrix_res_stat_METAT_plot[,1]))

#prepare the same for METAG 
#adjust p-values
#and take last 3 columns
matrix_res_stat_METAG_plot <- matrix_res_stat_METAG[,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]
matrix_res_sig_METAG_plot <- matrix_res_sig_METAG[,c((length(matrix_res_stat_METAG[1,])-2):length(matrix_res_stat_METAG[1,]))]

matrix_res_stat_METAG_plot <- melt(matrix_res_stat_METAG_plot, value.name = "R_statistic")
matrix_res_sig_METAG_plot <- melt(matrix_res_sig_METAG_plot, value.name = "p_value")
matrix_res_sig_METAG_plot$p_value.adj <- p.adjust(matrix_res_sig_METAG_plot$p_value, method="hochberg")

matrix_res_stat_METAG_plot <- left_join(matrix_res_stat_METAG_plot, matrix_res_sig_METAG_plot, by = c("Var1" = "Var1", "Var2" = "Var2")) 
matrix_res_stat_METAG_plot$data <- rep(c("Metagenome"), time=length(matrix_res_stat_METAG_plot[,1]))

matrix_res_stat_mental_plot <- rbind(matrix_res_stat_METAG_plot, matrix_res_stat_METAT_plot) 
save.image("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/mental_test_elbe.RData")
```

```{r}
#common plot for both - quite hard to read -  when adjusted p-values
  mental_test_plot <- matrix_res_stat_mental_plot %>%
  filter(R_statistic != "NA") %>% 
  ggplot(aes(x=Var1, y=Var2)) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  facet_grid( scales = "free", rows = vars(data)) +
 # coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=1.1, vjust=1.1), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value.adj) <= .05)+1], c("","*")[(abs(p_value.adj) <= .001)+1], c("","*")[(abs(p_value.adj) <= .0001)+1])))
```

```{r}
# plot for METAG - use normal p-values as significant 
  mental_test_METAG_plot <- matrix_res_stat_mental_plot %>%
  filter(R_statistic != "NA") %>% 
  filter(data == "Metagenome") %>% 
  filter(Var1 != "MOTUs") %>% 
  ggplot(aes(x=Var1, y=Var2)) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=1.1, vjust=1.1), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value) <= .05)+1], c("","*")[(abs(p_value) <= .001)+1], c("","*")[(abs(p_value) <= .0001)+1])))

```

```{r}
# plot for METAT - use normal p -values as significant 
  mental_test_METAT_plot <- matrix_res_stat_mental_plot %>%
  filter(R_statistic != "NA") %>% 
  filter(data == "Metatranscriptome") %>% 
  filter(Var1 != "MOTUs") %>% 
  ggplot(aes(x=Var1, y=Var2)) +
  geom_point(aes(size=R_statistic, colour=R_statistic, ), shape=15) +
  geom_point(aes(), shape=0, size = 8.5) + 
  scale_colour_gradient(low = "white", high = "darkgreen", ) + 
  guides(colour = guide_colorbar(title = "Mantel's r", frame.colour = "black", barwidth = 1.5)) + #title.theme = element_text(angle=90, vjust=1))) +
  scale_size_continuous(guide="none") +
  scale_x_discrete()+ theme_bw() +
  scale_y_discrete(position = "right") +
  coord_fixed(ratio = 1) +
  theme(axis.title = element_blank(), axis.ticks = element_blank(), line =element_blank(),
        strip.background = element_blank()) +
  theme(rect = element_blank(),
        axis.text.x = element_text(angle = 90, hjust=1.1, vjust=1.1), 
        legend.position = "right") +
  geom_text(vjust=0.76,hjust=0.49, aes(label = paste0(c("","*")[(abs(p_value) <= .05)+1], c("","*")[(abs(p_value) <= .001)+1], c("","*")[(abs(p_value) <= .001)+1])))

```