---
title: "Phototrophy analysis"
author: "Julia Golebiowska"
date: "2024-03-14"
output: pdf_document
---
 
```{r include=FALSE}
library(dplyr)
library(tidyverse)
library(ggplot2)
setwd("~/IGB_phd/phototrophy")
path = "C:/Users/jgolebiowska/Documents/IGB_phd/workshop_gene_catalog"
```

```{r read , eval=FALSE, include=FALSE}
##Loading files
#gene_cluster_table

# clstr <- read.csv(paste0(path,"/GROS22.clstr"), sep="\t", skip=0, header = F)
# colnames(clstr) <- c("gene_cluster", "mem", "gene", "length", "pos")
# #annotation
# annot <- read.csv(paste0(path,"/GROS22.kegg-annotations.tsv"), sep="\t", skip=0, header = TRUE, quote="")
# #abundance
# geneabund <- read.csv(paste0(path,"/BICEST_cellabund.tsv"), sep="\t", skip=1, header = TRUE, row.names = 1)
# #prepare geneabund for merging  
# geneabund <- geneabund %>% 
#   rownames_to_column("gene_cluster")
# #prepare metadata from merging
# metadata_sim <- read.csv(paste0(path,"/SAMEAID_SampleID_simplified.csv"), header=TRUE, sep=";") 
# #select the needed columns to avoid doubling computations
# metadata_sim <-metadata_sim[,c("BioSample", "ProjectID")]
# #read simplified metadata, as it has a ProjectID 
# metadata <- read.csv(paste0(path,"/SAMEAID_SampleID.csv"), header=TRUE, sep=";") %>% 
#   left_join(metadata_sim, by = "BioSample")  %>% 
#   mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAT.genecount.profile")) %>%
#   mutate(data_type="METAT")
# 
# metadata2 <- read.csv(paste0(path, "/SAMEAID_SampleID.csv"), header=TRUE, sep=";") %>% 
#   left_join(metadata_sim, by = "BioSample")  %>% 
#   mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAG.genecount.profile")) %>%
#   mutate(data_type="METAG")
# 
# metadata <- rbind(metadata, metadata2)
# save.image(file="phototrophy.RData")

path = "C:/Users/jgolebiowska/Documents/IGB_phd/virus/metadata/"
metadata <- read.csv(paste0(path,"/PhysicochemicalParameters_mod3.csv"), header=TRUE, sep=",", row.names=1) 
#remove rows with NA in Acession number - no molecular data
metadata <- metadata[which(!is.na(metadata$AccessionNumber_TBDSven)),]

#add chlorophyll  to metadata from another file
metadata_more <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_genes/Stations_Daten_RTG_250923.csv", sep=",", skip=2, header = F)
metadata_more[1,] <- paste(metadata_more[1,], metadata_more[2,], sep = "_")
#as header
names(metadata_more) <- metadata_more[1,]
#remove the first line - header  manually
metadata_more <- metadata_more[-1,]
#update enumeration
rownames(metadata_more) <- 1:nrow(metadata_more)
metadata_more <-metadata_more[metadata_more!="" & !is.na(metadata_more[,1]),]
metadata_more$Cruise_[metadata_more$Cruise_=="LP210308"]<-"Mar-21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP210503"]<-"Mai-21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP210725"]<-"Jul-21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220228"]<-"Feb-22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220522"]<-"Mai-22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220613"]<-"Jun-22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP221107"]<-"Nov-22"
#put the same name as in metadata
colnames(metadata_more)[1] <- "Sample_date"
colnames(metadata_more)[5] <- "Station"
#unify station names
metadata_more$Station[metadata_more$Station=="Bunthaus"] <- "BunthausSpitze"
metadata_more$Station[metadata_more$Station=="Mühlenberger Loch"] <- "Muhlenberger Loch"
metadata_more$Station[metadata_more$Station=="Brunsbüttel"] <- "Brunsbuttel"
metadata_more$Station[metadata_more$Station=="Medem Grund"] <- "Meedem grund" 

colnames(metadata_more)[which(colnames(metadata_more)=="Chla_µg/l")] <- "Chla_ug/l"
# colnames(metadata_more)[which(colnames(metadata_more)=="dCO2_µM")] <- "dCO2_uM"
#left_join with the CO2 CH4 Chlorophyll values from  metadata_more by Sample date
metadata <- metadata %>% merge(metadata_more[, c("Sample_date", "Station", "Chlorophyll_Fluoresence", "Chla_ug/l")], by = c("Sample_date", "Station"), all.x = T, all.y = F)

#remove some uncertain and redundant data 
metadata <- metadata[,!(colnames(metadata) %in% c("RespirationRate_O2ug.L.h", "DIC_uM.L", "DOC_uM.L", "Total_DIN_uM", "Ammonium_uM", "Nitrite_uM", "Nitrate_uM", "dCH4_nM", "dCO2_nM", "Silicate_uM", "SRP_uM"))]
#ids for expression normalisation - transcription/gene counts
```

```{r load}
#or just load prepared initial data
# load(file="phototrophy.RData")
```
# 1. Rhodopsins analysis

## 1.1. Find rhodopsins genes
```{r, echo=FALSE}
#let's find rhodopsins using KO
#here the ids from KEGG database
#K00909 GRK1_7; rhodopsin kinase [EC:2.7.11.14]
#K04641 bop; bacteriorhodopsin
#K04642 hop; halorhodopsin
#K04643 sop; sensory rhodopsin - this present only 
rhd_clust <- annot %>%  
  filter(KO %in% c("K00909","K04641","K04642","K04643")) %>% 
  left_join(geneabund,  join_by(QUERY == gene_cluster))
#not enough to do meaningful analysis 
#interestigly all rhodopsins were found in viral scaffolds ?
#only sensory rhodopsins - so rather not realy responsible for phototrophy
unique(rhd_clust[,"KO"])
```
## 1.2. Explore rhodopsins spatial and temporal distribution
```{r}
#prepare the table with matadata for plot
meta_rhd_clust <- rhd_clust %>% 
  pivot_longer(!c("QUERY", "KO", "BRITE", "CAZY", "COG", "DISEASE", "DRUG", "ENZYME", "GO", "MODULE", "NETWORK", "PATHWAY", "PUBMED", "RCLASS", "REACTION", "TC", "VP", "DESCRIPTION", "length"), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>%
  left_join(metadata, by="sampleid" )
meta_rhd_clust$Stromkilometer <- as.double(meta_rhd_clust$Stromkilometer)


paired_id_METAT <- colnames(rhd_clust)[grep("METAT", colnames(rhd_clust))]
paired_id_METAG <- sub("METAT", "METAG", paired_id_METAT)
#subset those which exists
paired_id_METAG <- paired_id_METAG[which(paired_id_METAG %in%  colnames(rhd_clust))]
#go back to metat to get only thoses with paired metag
paired_id_METAT <- sub("METAG", "METAT", paired_id_METAG)
 
#meta_rhd_clust[which(meta_rhd_clust$counts == 0), "counts"]
rhd_clust_METAT <- rhd_clust %>% select(paired_id_METAT)
rownames(rhd_clust_METAT) <- rhd_clust$QUERY
rhd_clust_METAG <- rhd_clust %>% select(paired_id_METAG)
rownames(rhd_clust_METAG) <- rhd_clust$QUERY

#normalise 
pseudocounts = 0.000001
rhd_clust_expr <- log(rhd_clust_METAT + pseudocounts) - log(rhd_clust_METAG + pseudocounts)
rhd_clust_expr$gene_clust <- rownames(rhd_clust_expr)
meta_rhd_clust_expr <- rhd_clust_expr %>% 
  pivot_longer(col=-c("gene_clust"), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>%
  left_join(metadata, by="sampleid" )

meta_rhd_clust_expr$Stromkilometer <- as.double(meta_rhd_clust_expr$Stromkilometer)

#plot it!
meta_rhd_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scales= "free_y") + ggtitle("Rhd counts across different seasons, river sites and data types") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)
#usually the highest variance in the last station

########

```
There is visible trend with more rhodopsin close to sea with exception in Feb22. Lets check salinity which should be the source of this trend.
```{r}
meta_rhd_clust$Salinity_PSU <- as.double(meta_rhd_clust$Salinity_PSU)
#plot it!
meta_rhd_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid,Sample_date,Salinity_PSU, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scales= "free_y") + ggtitle("Rhd counts across different seasons, salinity and data types") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)

```
Less clear pattern - November - salinity NA

## 1.3. Check different fractions
```{r}
#check different fractions
#plot it!
meta_rhd_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid,Stromkilometer, Sample_date, Sample_type, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~ data_type  * Sample_type, scales="free_y")  + geom_smooth(method = 'loess', formula= y ~ x)
#higher variance in free living

# and salinity
meta_rhd_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid, Salinity_PSU, Sample_date, Sample_type, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~ data_type  * Sample_type, scales="free_y")  + geom_smooth(method = 'loess', formula= y ~ x)
#and expression
meta_rhd_clust_expr  %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid, Stromkilometer, Sample_date) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~Sample_date, scales= "free_y") + ggtitle("Rhd expression across different seasons and river sites") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)
#usually the highest variance in the last station

meta_rhd_clust_expr  %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid, Salinity_PSU, Sample_date) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~Sample_date, scales= "free_y") + ggtitle("Rhd expression across different seasons and river sites") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)
#usually the highest variance in the last station
```
Similar trend in all fractions. Rhodopsins gene abundance counts shows similar parabolic trend  with the lowests values in the middle of the estuary and with the highest within closer proximity to Nordsea, where is higher salinity. The increase with salinity is raported in literature. For transcriptomic data counts differences are more pronouced logistic trend (sigmoid) - clear slope starting from around 660 km with a maximum at he end of the estuary. When we focus only on salinity factor we have linear trend for metag where counts increase with salinity, but for transcriptomic data we have parabolic trend with maximum around 10 PSU and decrese after for higher salinity values. For expression - not enough data to interpret. 
## 1.4. Check correlations
```{r}
#explore correlations for METAT counts

numeric_col <- colnames(meta_rhd_clust)[c(26:28, 30:51,54, 55, 58, 59)]
#metat
meta_rhd_clust_df <-  meta_rhd_clust  %>% filter(data_type == "METAT") %>%
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  select(c("sampleid", "counts", numeric_col)) %>% 
  group_by(sampleid) %>% summarise(counts=sum(counts), across()) %>% distinct() %>%
  select(c("counts", numeric_col)) #sample id not needed anymore
   
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
meta_rhd_clust_df[ meta_rhd_clust_df == ""] <- NA 

meta_rhd_clust_matrix <- matrix(apply(meta_rhd_clust_df,2, as.numeric), ncol=ncol(meta_rhd_clust_df), dimnames=list(c(),colnames(meta_rhd_clust_df)))

meta_rhd_clust_matrix[is.na(meta_rhd_clust_matrix)] <- 0 
res_metat <- cor(meta_rhd_clust_matrix, use='pairwise.complete.obs')
res_metat[is.na(res_metat)] <- 0 
heatmap(round(res, 2), cexRow = 0.5, cexCol = 0.5, col = my_palette, scale = c("none"))
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Chlorophyll_Fluoresence, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$`Chla_ug/l`, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Turbidity_NTU, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Salinity_PSU, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$dCO2_uM, use='pairwise.complete.obs') #negatively correlated with photosyntesis
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$SPM_mgperL, use='pairwise.complete.obs')  #turbidity proxy
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Silicate_mg.L, use='pairwise.complete.obs') 
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Nitrate_mg.L, use='pairwise.complete.obs')
#metag
meta_rhd_clust_df <-  meta_rhd_clust  %>% filter(data_type == "METAG") %>%
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  select(c("sampleid", "counts", numeric_col)) %>% 
  group_by(sampleid) %>% summarise(counts=sum(counts), across()) %>% distinct() %>%
  select(c("counts", numeric_col))
 
meta_rhd_clust_df[ meta_rhd_clust_df == ""] <- NA 

#to numeric 
meta_rhd_clust_matrix <- matrix(apply(meta_rhd_clust_df,2, as.numeric), ncol=ncol(meta_rhd_clust_df), dimnames=list(c(),colnames(meta_rhd_clust_df)))

meta_rhd_clust_matrix[is.na(meta_rhd_clust_matrix)] <- 0 
res_metag <- cor(meta_rhd_clust_matrix, use='pairwise.complete.obs')
res_metag[ is.na(res_metag)] <- 0 
heatmap(round(res_metag, 2), cexRow = 0.5, cexCol = 0.5, col = my_palette, scale = c("none"))
#correlations tests for the hidhest and lowest scores
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Chlorophyll_Fluoresence, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$`Chla_ug/l`, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Turbidity_NTU, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Salinity_PSU, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$PTC_mgperL, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Sat_O2_Perc, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$dCH4_uM, use='pairwise.complete.obs')
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Ammonium_mg.L, use='pairwise.complete.obs')
#positive correlation for DIC and negative when POC. When particulate carbon sources - accessible for microbes are accessible 
#- no starvation no need for less effective rhodopsin system to use. 
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$DIC_mg.L, use='pairwise.complete.obs') 
cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$POC_mgperL, use='pairwise.complete.obs') #and turbidity proxy

```
Very strong correlation between "DOC_mg.L", "TN_mg.L". Quite strong between  "TN_mg.L", "DIC_mg.L" Between Stromkilometer and salinity lower then expected
No correlation with counts. 
## 1.5. Check the trends of rhodopsins counts across probable determinants
```{r}

#lets plot two parameters agains rhodopsins counts
#salinity ensure is double

meta_rhd_clust %>% 
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Salinity_PSU) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() + geom_smooth(method = 'loess', formula= y ~ x) + 
 facet_wrap(~ data_type, scales="free")

#pH
  meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), pH ) %>%
ggplot( aes(x=pH, y=counts)) + geom_point()  + geom_smooth(method = 'loess', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free") #omit rows with zero - artifact from NA <= 0- artifact from NA <= 0

#tubidity - negative correlation   when too much turbidity rhodopsin system cannot function
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Turbidity_NTU) %>%
ggplot(aes(x=Turbidity_NTU, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 
#two chlorophyll proxies
meta_rhd_clust$`Chla_ug/l` <- as.double(meta_rhd_clust$`Chla_ug/l`)
meta_rhd_clust$Chlorophyll_Fluoresence<- as.double(meta_rhd_clust$Chlorophyll_Fluoresence)
#chlorophyll
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), `Chla_ug/l`) %>%
ggplot(aes(x=`Chla_ug/l`, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 

#Chlorophyll_Fluoresence
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Chlorophyll_Fluoresence) %>%
ggplot(aes(x=Chlorophyll_Fluoresence, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 
#not that high correlation for both proxies
cor(meta_rhd_clust$`Chla_ug/l` ,meta_rhd_clust$Chlorophyll_Fluoresence, use="pairwise.complete.obs")
#Nitrate - nutritiousness of water
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Nitrate_mg.L) %>%
ggplot(aes(x=Nitrate_mg.L, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 
#silicate - no idea ... diatoms as proxi for water nutritiousness 
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Silicate_mg.L) %>%
ggplot(aes(x=Silicate_mg.L, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 

```
All checked parameters shows correlation between them and rhodopsins counts - pH and salinity positive and turbidity negative. Unfortunately no transcriptomic data - expected  positive correlation with turbidity. Negative corrrelation for Chla ug/l. For fluorescence the results are very vaguely or not correlated. 

## 1.6. Explore variance
```{r}
#explore the variance as it looks like is different across stations
var_meta_rhd_clust_station <- meta_rhd_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  group_by(Station, data_type) %>% 
  summarise(var = var(counts))

#variance across stations ( the variance is lower in brackish water)
rhd_var_station <- var_meta_rhd_clust_station %>% arrange(factor(Station, levels=c("Mühlenberger Loch","Twielenfleth", "Schwarztonnensand", "Brunsbuettel","Meedem Grund")))
rhd_var_station %>% filter(data_type == "METAG")
rhd_var_station %>% filter(data_type == "METAT")
```

# 2. PsbA analysis 
## 2.1 Find PsbA gene and check its spatial and temporal distribution

```{r}
#find psbA genes 
psbA_clust <- annot %>%  
  filter(KO %in% c("K02703")) %>% left_join(geneabund,  join_by(QUERY == gene_cluster))

#prepare the table with matadata for plot
meta_psbA_clust <- psbA_clust %>% 
  pivot_longer(!c("QUERY", "KO", "BRITE", "CAZY", "COG", "DISEASE", "DRUG", "ENZYME", "GO", "MODULE", "NETWORK", "PATHWAY", "PUBMED", "RCLASS", "REACTION", "TC", "VP", "DESCRIPTION", "length"), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% left_join(metadata, by="sampleid")

meta_psbA_clust$Stromkilometer <- as.double(meta_psbA_clust$Stromkilometer)

#plot it!
meta_psbA_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + ggtitle("psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 
#usually the highest variance in the beginning and at the end of the estuary 
#but very similar across  the samples
```
No common trend 
## 2.2 Check different fractions 
```{r}
meta_psbA_clust$Stromkilometer <- as.double(meta_psbA_clust$Stromkilometer)
#check different fractions
#plot it!
meta_psbA_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, Sample_type, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~ data_type * Sample_type , scale = "free_y") + geom_smooth()

#higher variance in particle associated - interesting opposite to the rhodopsins 
#turbidity
meta_psbA_clust$Turbidity_NTU <- as.double(meta_psbA_clust$Turbidity_NTU)
meta_psbA_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(Stromkilometer, data_type, Sample_type, Sample_date) %>% 
  summarise(counts=sum(counts), Turbidity_NTU) %>%
ggplot(aes(x=Turbidity_NTU, y=counts)) + geom_point() + facet_wrap(~ data_type, scale = "free_y") + geom_smooth(method='lm') #omit rows with zero 

#Salinity
meta_psbA_clust$Salinity_PSU <- as.double(meta_psbA_clust$Salinity_PSU)
meta_psbA_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(Stromkilometer, data_type, Sample_type, Sample_date) %>% 
  summarise(counts=sum(counts), Salinity_PSU) %>%
ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() + facet_wrap(~ data_type, scale = "free_y") + geom_smooth(method='lm') #omit rows with zero 
```
Cannot really compare fractions - PsbA genes have more transcripts in fractions then in free-living - more transcriptomic activity at the end of the estuary on particles, when no spatial difference in free living fraction. Highest photosynthetic potential on particles - (once again similar trends between two types), then free living. Quite suprising - as phytoplankton usually is considered to be > 5um. Hypothesis: Particles harbour great photosynthetic potential in the Elbe estuary. High photosynthetic potential at the beginning of the estuary is not utilized - urban influence? Negative correlation with turbidity - no surprise. Vague positive trend with salinty. 

No really a trend in METAG

## 2.3 Check correlations
```{r}
#explore correlations for METAT counts

numeric_col <- colnames(meta_psbA_clust)[c(21,28, 30:55, 58, 59)]
#metat
meta_psbA_clust_df <-  meta_psbA_clust  %>% filter(data_type == "METAT") %>%
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  select(c("sampleid", "counts", numeric_col)) %>% 
  group_by(sampleid) %>% summarise(counts=sum(counts), across()) %>% distinct()  %>% 
  select(c("counts", numeric_col)) #sample id not needed anymore
   
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
meta_psbA_clust_df[ meta_psbA_clust_df == ""] <- NA 

meta_psbA_clust_matrix <- matrix(apply(meta_psbA_clust_df,2, as.numeric), ncol=ncol(meta_psbA_clust_df), dimnames=list(c(),colnames(meta_psbA_clust_df)))

meta_psbA_clust_matrix[is.na(meta_psbA_clust_matrix)] <- 0 
res_metat <- cor(meta_psbA_clust_matrix, use='pairwise.complete.obs')
res_metat[is.na(res_metat)] <- 0 
heatmap(round(res_metat, 2), cexRow = 0.5, cexCol = 0.5, col = my_palette, scale = c("none"))

#metag
meta_psbA_clust_df <-  meta_psbA_clust  %>% filter(data_type == "METAG") %>%
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  select(c("sampleid", "counts", numeric_col)) %>% 
  group_by(sampleid) %>% summarise(counts=sum(counts), across()) %>% distinct()  %>% 
  select(c("counts", numeric_col)) #sample id not needed anymore
 
meta_psbA_clust_df[ meta_psbA_clust_df == ""] <- NA 

#to numeric 
meta_psbA_clust_matrix <- matrix(apply(meta_psbA_clust_df,2, as.numeric), ncol=ncol(meta_psbA_clust_df), dimnames=list(c(),colnames(meta_psbA_clust_df)))

meta_psbA_clust_matrix[is.na(meta_psbA_clust_matrix)] <- 0 
res_metag <- cor(meta_psbA_clust_matrix, use='pairwise.complete.obs')
res_metag[ is.na(res_metag)] <- 0 
heatmap(round(res_metag, 2), cexRow = 0.5, cexCol = 0.5, col = my_palette, scale = c("none"))

#and turbidity proxy

```
## 2.3 Variance analysis 
```{r}
#variance analysis 
var_meta_psbA_clust <- meta_psbA_clust  %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  group_by(Station, Sample_date) %>% 
  summarise(var = var(counts))

var_meta_psbA_clust_station <- meta_psbA_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  group_by(Station) %>% 
  summarise(var = var(counts))

#variance across stations (similarly the variance is lower n the brakish water)
var_meta_psbA_clust_station  %>% arrange(factor(Station, levels = c("Mühlenberger Loch","Twielenfleth", "Schwarztonnensand", "Brunsbüttel","Meedem Grund"))) #change the order 
```

The variance is increasing with a station order 

#3. Comperative analysis PsbA and rhd
#3.1 Temporal and spatial dynamic PsbA and rhd
Lets compare counts of rhodopsins and PsbA genes
```{r}
colnames(meta_psbA_clust)[which(names(meta_psbA_clust) == "counts")] <- "PsbA"
colnames(meta_rhd_clust)[which(names(meta_rhd_clust) == "counts")] <- "rhd"
#merge together by counts
meta_all_counts <- meta_psbA_clust %>% left_join(meta_rhd_clust[, c("rhd", "sampleid")], by = "sampleid")
#check temporal data
meta_all_counts %>% 
#  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
    group_by(sampleid,Station, Sample_date, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%
  ggplot(aes(x=sum_rhd, y=sum_PsbA)) + ggtitle(" Phototrophic genes and transcript counts  different  seasons") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free") + geom_smooth(method = "lm") 
#fractions
meta_all_counts %>% 
#  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
    group_by(sampleid,Stromkilometer, Sample_type, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%
  ggplot(aes(x=sum_rhd, y=sum_PsbA)) + ggtitle(" Phototrophic genes and transcript counts  different fractions") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_type, scale = "free") + geom_smooth(method = "lm") 
#spatial data   
meta_all_counts %>% 
#  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
    group_by(sampleid,Station, Sample_date, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%
  ggplot(aes(x=sum_rhd, y=sum_PsbA)) + ggtitle(" Phototrophic genes and transcript counts  different  river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Station, scale = "free") + geom_smooth(method = "lm") 
  

 #Lets check all - with devision on data types.
meta_all_counts %>% 
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
    group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%
  ggplot(aes(x=sum_rhd, y=sum_PsbA)) + ggtitle(" Phototrophic genes and transcript counts  different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type, scale = "free") + geom_smooth(method = "lm") 
  

```
Surprising positive correlation on particulate data 

```{r}
calcor=function(da){
data.frame(cor.test(da$sum_rhd,da$sum_PsbA)[c("estimate","p.value")])
}
meta_all_counts %>%  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%  group_by(data_type)  %>% do(calcor(.)) # metat and metag significant 
meta_all_counts %>%  group_by(sampleid, Station, Sample_date, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%  group_by(data_type, Station)  %>% do(calcor(.)) %>% filter(p.value < 0.05) #no significant
meta_all_counts %>%  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%  group_by(data_type, Sample_date)  %>% do(calcor(.)) %>% filter(p.value < 0.05)
#Mai positive significant 
meta_all_counts %>%  group_by(sampleid, Stromkilometer, Sample_type, data_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%  group_by(data_type, Sample_type)  %>% do(calcor(.)) %>% filter(p.value < 0.05)#positive significant 


```
For spatial - if correlation significant is always very negative, but for temporal few significant are quite positive. Season drives increase of phototrophic genes and positive correlation, but they are negative correlated spatialy - they dependance on microscale - coherent with literature  
#3.2 Dynamic PsbA and rhd across environmenetal parametersdrivers according to literature 

```{r}
 #Lets check all - with devision on data types.
meta_all_counts %>% 
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
    group_by(sampleid, Turbidity_NTU, Sample_date, data_type, Sample_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%
  ggplot(aes(x=Turbidity_NTU)) + ggtitle("Phototrophic genes and transcript counts  across turbidity") + geom_point(aes(y=sum_rhd), colour="red") + geom_smooth(aes(y=sum_rhd), method = "loess")  + geom_point(aes(y=sum_PsbA), colour="green") + geom_smooth(aes(y=sum_PsbA), method = "loess")  + scale_y_log10() + facet_wrap(~data_type *Sample_type, scale = "free") 
```
Both very much synchronized against turbidity - surprising ...
```{r}
 #Lets check all - with devision on data types.
meta_all_counts$`Chla_ug/l` <- as.numeric(meta_all_counts$`Chla_ug/l`)
meta_all_counts %>% 
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
    group_by(sampleid, `Chla_ug/l`, Sample_date, data_type, Sample_type) %>% 
  summarise(sum_rhd=sum(rhd), sum_PsbA=sum(PsbA)) %>%
  ggplot(aes(x=`Chla_ug/l`)) + ggtitle("Phototrophic genes and transcript counts across chlorophyl") + geom_point(aes(y=sum_rhd), colour="red") + geom_smooth(aes(y=sum_rhd), method = "loess")  + geom_point(aes(y=sum_PsbA), colour="green") + geom_smooth(aes(y=sum_PsbA), method = "loess")  + scale_y_log10() + facet_wrap(~data_type *Sample_type, scale = "free") 
```

#4. AlaS analysis

## 1.1. Find 5-aminolevulinate synthase
```{r, echo=FALSE}
#let's find AlaS
AlaS_clust <- annot %>%  
  filter(KO %in% c("K00643")) %>% 
  left_join(geneabund,  join_by(QUERY == gene_cluster))
```
## 1.2. Explore rhodopsins spatial and temporal distribution
```{r}
#prepare the table with matadata for plot
meta_AlaS_clust <- AlaS_clust %>% 
  pivot_longer(!c("QUERY", "KO", "BRITE", "CAZY", "COG", "DISEASE", "DRUG", "ENZYME", "GO", "MODULE", "NETWORK", "PATHWAY", "PUBMED", "RCLASS", "REACTION", "TC", "VP", "DESCRIPTION", "length"), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>%
  left_join(metadata, by="sampleid" )
meta_AlaS_clust$Stromkilometer <- as.double(meta_AlaS_clust$Stromkilometer)


paired_id_METAT <- colnames(AlaS_clust)[grep("METAT", colnames(AlaS_clust))]
paired_id_METAG <- sub("METAT", "METAG", paired_id_METAT)
#subset those which exists
paired_id_METAG <- paired_id_METAG[which(paired_id_METAG %in%  colnames(AlaS_clust))]
#go back to metat to get only thoses with paired metag
paired_id_METAT <- sub("METAG", "METAT", paired_id_METAG)
 
#meta_AlaS_clust[which(meta_AlaS_clust$counts == 0), "counts"]
AlaS_clust_METAT <- AlaS_clust %>% select(paired_id_METAT)
rownames(AlaS_clust_METAT) <- AlaS_clust$QUERY
AlaS_clust_METAG <- AlaS_clust %>% select(paired_id_METAG)
rownames(AlaS_clust_METAG) <- AlaS_clust$QUERY

#normalise 
pseudocounts = 0.000001
AlaS_clust_expr <- log(AlaS_clust_METAT + pseudocounts) - log(AlaS_clust_METAG + pseudocounts)
AlaS_clust_expr$gene_clust <- rownames(AlaS_clust_expr)
meta_AlaS_clust_expr <- AlaS_clust_expr %>% 
  pivot_longer(col=-c("gene_clust"), names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>%
  left_join(metadata, by="sampleid" )

meta_AlaS_clust_expr$Stromkilometer <- as.double(meta_AlaS_clust_expr$Stromkilometer)

#plot it!
meta_AlaS_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scales= "free_y") + ggtitle("AlaS counts across different seasons, river sites and data types") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)
#usually the highest variance in the last station

########

```
There is visible trend with more rhodopsin close to sea with exception in Feb22. Lets check salinity which should be the source of this trend.
```{r}
meta_AlaS_clust$Salinity_PSU <- as.double(meta_AlaS_clust$Salinity_PSU)
#plot it!
meta_AlaS_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid,Sample_date,Salinity_PSU, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scales= "free_y") + ggtitle("AlaS counts across different seasons, salinity and data types") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)

```
Less clear pattern - November - salinity NA

## 1.3. Check different fractions
```{r}
#check different fractions
#plot it!
meta_AlaS_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid,Stromkilometer, Sample_date, Sample_type, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~ data_type  * Sample_type, scales="free_y")  + geom_smooth(method = 'loess', formula= y ~ x)
#higher variance in free living

# and salinity
meta_AlaS_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid, Salinity_PSU, Sample_date, Sample_type, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~ data_type  * Sample_type, scales="free_y")  + geom_smooth(method = 'loess', formula= y ~ x)
#and expression
meta_AlaS_clust_expr  %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid, Stromkilometer, Sample_date) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~Sample_date, scales= "free_y") + ggtitle("AlaS expression across different seasons and river sites") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)
#usually the highest variance in the last station
#not enougth values
meta_AlaS_clust_expr  %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  mutate(counts=counts/1000) %>%
  group_by(sampleid, Salinity_PSU, Sample_date) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() +
  geom_tile() + facet_wrap(~Sample_date, scales= "free_y") + ggtitle("AlaS expression across different seasons and river sites") + scale_y_log10() + geom_smooth(method = 'loess', formula= y ~ x)
#usually the highest variance in the last station
```
For AlaS METAG counts and Elbe km we have exponential trend with increasing counts toward sea. For METAT counts we have sigmoid function also with increacing counts towards sea. For FL AlaS and salinity have linear trend increasing with salinity. For particulated fractions besides METAT parabolic trend with maximum around 10.  
## 1.4. Check correlations
```{r}
#explore correlations for METAT counts

numeric_col <- colnames(meta_AlaS_clust)[c(26:28, 30:51,54, 55, 58, 59)]
#metat
meta_AlaS_clust_df <-  meta_AlaS_clust  %>% filter(data_type == "METAT") %>%
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  select(c("sampleid", "counts", numeric_col)) %>% 
  group_by(sampleid) %>% summarise(counts=sum(counts), across()) %>% distinct() %>% select(c("counts", numeric_col)) #sample id not needed anymore
   
my_palette <- colorRampPalette(c("blue", "white", "red"))(n = 299)
meta_AlaS_clust_df[ meta_AlaS_clust_df == ""] <- NA 

meta_AlaS_clust_df <- apply(meta_AlaS_clust_df,2, as.numeric)
meta_AlaS_clust_matrix <- matrix(meta_AlaS_clust_df, ncol=ncol(meta_AlaS_clust_df), dimnames=list(c(),colnames(meta_AlaS_clust_df)))

meta_AlaS_clust_matrix[is.na(meta_AlaS_clust_matrix)] <- 0 
res_metat <- cor(meta_AlaS_clust_matrix, use='pairwise.complete.obs')
res_metat[is.na(res_metat)] <- 0 
heatmap(round(res, 2), cexRow = 0.5, cexCol = 0.5, col = my_palette, scale = c("none"))
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$Chlorophyll_Fluoresence, use='pairwise.complete.obs')
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$`Chla_ug/l`, use='pairwise.complete.obs')
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$Turbidity_NTU, use='pairwise.complete.obs')
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$Salinity_PSU, use='pairwise.complete.obs')
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$dCO2_uM, use='pairwise.complete.obs') #negatively correlated with photosyntesis
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$SRP_mgperL, use='pairwise.complete.obs')  #turbidity proxy
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$Sat_O2_Perc, use='pairwise.complete.obs')
cor.test(meta_AlaS_clust_df$counts, meta_AlaS_clust_df$pH, use='pairwise.complete.obs')
metag
meta_AlaS_clust_df <-  meta_AlaS_clust  %>% filter(data_type == "METAG") %>%
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  select(c("sampleid", "counts", numeric_col)) %>% 
  group_by(sampleid) %>% summarise(counts=sum(counts), across()) %>% distinct() %>%
  select(c("counts", numeric_col))
 
meta_AlaS_clust_df[ meta_AlaS_clust_df == ""] <- NA 

#to numeric 
meta_AlaS_clust_matrix <- matrix(apply(meta_AlaS_clust_df,2, as.numeric), ncol=ncol(meta_AlaS_clust_df), dimnames=list(c(),colnames(meta_AlaS_clust_df)))

meta_AlaS_clust_matrix[is.na(meta_AlaS_clust_matrix)] <- 0 
res_metag <- cor(meta_AlaS_clust_matrix, use='pairwise.complete.obs')
res_metag[ is.na(res_metag)] <- 0 
heatmap(round(res_metag, 2), cexRow = 0.5, cexCol = 0.5, col = my_palette, scale = c("none"))
#correlations tests for the hidhest and lowest scores
# cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Chlorophyll_Fluoresence, use='pairwise.complete.obs')
# cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$`Chla_ug/l`, use='pairwise.complete.obs')
# cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Turbidity_NTU, use='pairwise.complete.obs')
# cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Salinity_PSU, use='pairwise.complete.obs')
# cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$DOC_mg.L, use='pairwise.complete.obs')
# cor.test(meta_rhd_clust_df$counts, meta_rhd_clust_df$Sat_O2_Perc, use='pairwise.complete.obs')


```
Very strong correlation between "DOC_mg.L", "TN_mg.L". Quite strong between  "TN_mg.L", "DIC_mg.L" Between Stromkilometer and salinity lower then expected
No correlation with counts. 
## 1.5. Check the trends of rhodopsins counts across probable determinants
```{r}

#lets plot two parameters agains rhodopsins counts
#salinity ensure is double

meta_rhd_clust %>% 
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Salinity_PSU) %>%
  ggplot(aes(x=Salinity_PSU, y=counts)) + geom_point() + geom_smooth(method = 'loess', formula= y ~ x) + 
 facet_wrap(~ data_type, scales="free")

#pH
  meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), pH ) %>%
ggplot( aes(x=pH, y=counts)) + geom_point()  + geom_smooth(method = 'loess', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free") #omit rows with zero - artifact from NA <= 0- artifact from NA <= 0

#tubidity - negative correlation   when too much turbidity rhodopsin system cannot function
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Turbidity_NTU) %>%
ggplot(aes(x=Turbidity_NTU, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 
#two chlorophyll proxies
meta_rhd_clust$`Chla_ug/l` <- as.double(meta_rhd_clust$`Chla_ug/l`)
meta_rhd_clust$Chlorophyll_Fluoresence<- as.double(meta_rhd_clust$Chlorophyll_Fluoresence)
#chlorophyll
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), `Chla_ug/l`) %>%
ggplot(aes(x=`Chla_ug/l`, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 

#Chlorophyll_Fluoresence
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Chlorophyll_Fluoresence) %>%
ggplot(aes(x=Chlorophyll_Fluoresence, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 
#not that high correlation for both proxies
cor(meta_rhd_clust$`Chla_ug/l` ,meta_rhd_clust$Chlorophyll_Fluoresence, use="pairwise.complete.obs")
#Nitrate - nutritiousness of water
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Nitrate_mg.L) %>%
ggplot(aes(x=Nitrate_mg.L, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 
#silicate - no idea ... diatoms as proxi for water nutritiousness 
meta_rhd_clust %>% 
  #filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(sampleid, Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts), Silicate_mg.L) %>%
ggplot(aes(x=Silicate_mg.L, y=counts)) + geom_point()  + geom_smooth(method = 'lm', formula= y ~ x) +
    facet_wrap(~ data_type, scales="free")  #ommit rows with zero 

```
## 1.6. Explore variance
```{r}
#explore the variance as it looks like is different across stations
var_meta_rhd_clust_station <- meta_rhd_clust %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>% 
  group_by(Station, data_type) %>% 
  summarise(var = var(counts))

#variance across stations ( the variance is lower in brackish water)
rhd_var_station <- var_meta_rhd_clust_station %>% arrange(factor(Station, levels=c("Mühlenberger Loch","Twielenfleth", "Schwarztonnensand", "Brunsbuettel","Meedem Grund")))
rhd_var_station %>% filter(data_type == "METAG")
rhd_var_station %>% filter(data_type == "METAT")
```