---
title: "viral_genes"
author: "Julia Golebiowska"
date: "2024-05-17"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
The Analysis of axuliary genes carried by Viruses in the Elbe Estuary. Metagenomic data from 2021-22.   

```{r setup}
#lets load the viral data 
library(dplyr)
library(tidyverse)
library(ggplot2)
setwd("~/IGB_phd/virus/viral_genes")
path = "C:/Users/jgolebiowska/Documents/IGB_phd/workshop_gene_catalog"
```

```{r loading}
##Loading files - commented out - big files used for joining data frames  
#gene_cluster_table
#long to load - used one from the saved object
#clstr <- read.csv(paste0(path,"/GROS22.clstr"), sep="\t", skip=0, header = F)
#colnames(clstr) <- c("gene_cluster", "mem", "gene", "length", "pos")
#gene abundance (in fact gene copy number per cell and transcription per cell)
#big file - avoid loading multiple times
#geneabund <- read.csv(paste0(path,"/BICEST_cellabund.tsv"), sep="\t", skip=1, header = TRUE, row.names = 1)
#viral genes annotated
#big file avoid loading multiple times
#vibrant_annot <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/BICEST.phages.annot",  sep = "\t", header = T)
#colnames(vibrant_annot) <- c("genome", "range", "KO", "DESCRIPTION")
#vibrant_annot[,"genome"] = sub(">", "", vibrant_annot[,"genome"])
#metadata
metadata_sim <- read.csv(paste0(path,"/SAMEAID_SampleID_simplified.csv"), header=TRUE, sep=";") 
#select the needed columns to avoid doubling computations
metadata_sim <-metadata_sim[,c("BioSample", "ProjectID")]
#read simplified metadata, as it has a ProjectID 
metadata <- read.csv(paste0(path,"/SAMEAID_SampleID.csv"), header=TRUE, sep=",") %>% 
  left_join(metadata_sim, by = "BioSample")  %>% 
  mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAT.genecount.profile")) %>%
  mutate(data_type="METAT")

metadata2 <- read.csv(paste0(path, "/SAMEAID_SampleID.csv"), header=TRUE, sep=",") %>% 
  left_join(metadata_sim, by = "BioSample")  %>% 
  mutate(sampleid=paste0(ProjectID,"_",BioSample,"_METAG.genecount.profile")) %>%
  mutate(data_type="METAG")

metadata <- rbind(metadata, metadata2)
#remove not consequent date style (remove -)  
metadata <-metadata %>% 
  mutate(sample=str_remove(sampleid, ".genecount.profile")) %>%
  mutate(sample=str_replace(sample, "[.]", "-"))
#unify date names (remove '-')
metadata <-metadata %>% mutate(Sample_date = str_replace(Sample_date, "-"," "))
#unify station names
metadata$Station[metadata$Station=="Meedem Grund"] <- "Medemgrund"

#chlorophyll data 
Sys.setlocale("LC_ALL", "en_US.UTF-8")

#add chlorophyll (and gas values)  to metadata from another file
metadata_more <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_genes/Stations_Daten_RTG_250923.csv", sep=",", skip=2, header = F)
metadata_more[1,] <- paste(metadata_more[1,], metadata_more[2,], sep = "_")
#as header
names(metadata_more) <- metadata_more[1,]
#remove the first line - header  manually
metadata_more <- metadata_more[-1,]
#update enumeration
rownames(metadata_more) <- 1:nrow(metadata_more)
metadata_more <-metadata_more[metadata_more!="" & !is.na(metadata_more[,1]),]
metadata_more$Cruise_[metadata_more$Cruise_=="LP210308"]<-"Mar 21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP210503"]<-"Mai 21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP210725"]<-"Jul 21"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220228"]<-"Feb 22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220522"]<-"Mai 22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP220613"]<-"Jun 22"
metadata_more$Cruise_[metadata_more$Cruise_=="LP221107"]<-"Nov 22"
#put the same name as in metadata
colnames(metadata_more)[1] <- "Sample_date"
colnames(metadata_more)[5] <- "Station"
#unify station names
metadata_more$Station[metadata_more$Station=="Medem Grund"] <- "Medemgrund"
#remove another meassurement from the same station and date
metadata_more <- metadata_more[-5,]
#update enumeration
rownames(metadata_more) <- 1:nrow(metadata_more)
#remove 'micro' sign
colnames(metadata_more)[which(colnames(metadata_more)=="Chla_µg/l")] <- "Chla_ug/l"
colnames(metadata_more)[which(colnames(metadata_more)=="dCO2_µM")] <- "dCO2_uM"
#left_join with the CO2 CH4 Chlorophyll values from  metadata_more by Sample date
metadata <- metadata %>% merge(metadata_more[, c("Sample_date", "Station", "dCH4_nM", "Chla_ug/l","dCO2_uM")], by = c("Sample_date", "Station"), all.x = T)
#change problematic column names
colnames(metadata)[which(colnames(metadata) == "Chla_ug/l")] <- "Chlorophyll"
metadata$Chlorophyll <- as.numeric(metadata$Chlorophyll)
metadata$dCH4_nM <- as.numeric(metadata$dCH4_nM)
metadata$dCO2_uM <- as.numeric(metadata$dCO2_uM)
write.csv(metadata, paste0(path,"/metadata_all.csv"), row.names = FALSE)
```

Prepare data for prokaryotic genes exploration
```{r}
#generate dataframe to work with
geneabund <- geneabund %>% 
  rownames_to_column("gene_cluster")
annot <- read.csv(paste0(path,"/GROS22.kegg-annotations.tsv"), sep="\t", skip=0, header = TRUE, quote="")
psbA_prok <- annot %>%  
  filter(KO %in% c("K02703")) %>% 
  left_join(geneabund,  join_by(QUERY == gene_cluster))
saveRDS(psbA_prok, file = "C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/psbA_prok.RDS")

```
Prepare metadata with Chlorophyll 




```{r pressure, echo=FALSE}

vir_genecluster_annot <- vibrant_annot %>% 
  left_join(clstr,  join_by(genome == gene)) %>% left_join(geneabund)
#save it 
saveRDS(vir_genecluster_annot, file = "vir_genecluster_annot.RDS")

#clean workspace
rm(vibrant_annot, geneabund, clstr)
```

```{r genes search}
#Check if they are present  the data:
genome_quality_Combined_proviruses <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/Elbe_viruses_distribution/VIBRANT_results_Combined_proviruses/VIBRANT_genome_quality_Combined_proviruses.tsv", sep="\t")
genome_quality_Combined_viruses <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/virus/Elbe_viruses_distribution/VIBRANT_results_Combined_viruses/VIBRANT_genome_quality_Combined_viruses.tsv", sep="\t")
genome_quality_Combined_all_viruses <- rbind(genome_quality_Combined_proviruses, genome_quality_Combined_viruses)
genome_quality_Combined_all_viruses[, "scaffold"] <- sub("_[0-9] [0-9]+[-][0-9]+/[0-9]+$","",genome_quality_Combined_all_viruses[, "scaffold"])
vir_genecluster_annot[, "genome"] <- sub("_g[0-9]+_i[0-9]_[0-9]$","",vir_genecluster_annot[, "genome"])
vir_genecluster_annot[, "genome"] <- sub("_fragment_[0-9]*.", "", vir_genecluster_annot[, "genome"])
vir_genecluster_annot[, "genome"] <- sub("_[0-9]+$", "", vir_genecluster_annot[, "genome"])

#psbA_plot_num <- merge(vir_genecluster_annot, genome_quality_Combined_all_viruses, by.x="genome", by.y="scaffold")
# add quality data
vir_genecluster_annot <- vir_genecluster_annot %>% 
   left_join(genome_quality_Combined_all_viruses,  join_by(genome == scaffold))
saveRDS(vir_genecluster_annot, file = "vir_genecluster_annot.RDS")
#rhodopsins
rhd_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00909","K04641","K04642","K04643")) #not found
#psbA photosystem
psbA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02703")) #found
#for merging tables we need only gene cluster values - lets- filter out redundant data 
#if genome needed generate another table
#remove columns where different values occures
psbA_vir <- psbA_vir[,-c(1,2)]
#remove duplicated rows
psbA_vir <- psbA_vir[!duplicated(psbA_vir), ]
#remove lines with "mem" in mem column - as there are followed by NA count values 
psbA_vir <- psbA_vir[which(psbA_vir$mem!="mem"), ]
psbA_vir_gene_clust <- psbA_vir$gene_cluster # save gene clusters assigned as viral - we will nedd them to differenciate between hosts and viral psbA genes
#find psaA genes photosystem 
psaA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02689")) #not found

#Photosynthetic reaction center 
#(petC - K02636, petE - K02638, and petH - K02641) and the cyanobacteria-specific antenna proteins
petC_vir <- vir_genecluster_annot %>%  filter(KO %in% c("K02636")) #not_found

petE_vir <- vir_genecluster_annot %>%  filter(KO %in% c("K02638")) #found
 
petH_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02641")) #not_found
 
#(apcA - K02092, apcF - K02097, cpcA - K02284, cpeA - K05376, and cpeT - K05383)
# the cyanobacteria-specific antenna proteins

apcA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02092")) 
 
apcF_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02097"))
 
cpcA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02284"))
 
cpeA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K05376"))

cpeT_vir <- vir_genecluster_annot %>% filter(KO %in% c("K05383"))
 
#nitrification pathways
#nitrogen fixation
nifH_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02588"))
 
nifD_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02586"))
 
#denitrification:
napA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K02567"))
 
nirS_vir <- vir_genecluster_annot %>% filter(KO %in% c("K15864"))
 
norB_vir <- vir_genecluster_annot %>% filter(KO %in% c("K04561"))  
 
nosZ_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00376")) #1 found
#Calvin cycle
#rbcL
 #K01601
rbcL_vir <- vir_genecluster_annot %>% filter(KO %in% c("K01601")) #90!
#rbcS
#K01602
rbcS_vir <- vir_genecluster_annot %>% filter(KO %in% c("K01602")) #no
#phosphoribulokinase - prkB
#	K00855   
prkB_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00855")) #2
#glyceraldehyde-3-phosphate dehydrogenase gapN
#K00131
gapN_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00131")) #0
#gltA citrate synthase
#K01647
gltA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K01647")) #0
#isocitrate dehydrogenase IDH3
#K00030  
idh3_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00030")) #2
#malate dehydrogenase
#K00024       
md_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00024")) #388
#methyl-coenzyme M reductase alpha subunit
#methane pathway 
#K00399
mcrA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00399")) #0
#K10944
pmoA_vir <- vir_genecluster_annot %>% filter(KO %in% c("K10944")) #0
#Shemin pathway
#K00643
AlaS_vir <- vir_genecluster_annot %>% filter(KO %in% c("K00643")) #61!!!!!!
AlaS_vir <- AlaS_vir[,-c(1,2)]
#remove duplicated rows
AlaS_vir <- AlaS_vir[!duplicated(psbA_vir), ]
#remove lines with "mem" in mem column - as there are followed by NA count values 
AlaS_vir <- AlaS_vir[which(psbA_vir$mem!="mem"), ]
AlaS_vir_gene_clust <- AlaS_vir$gene_cluster 
#phycobilisome rod-core linker protein
#K02290
prc <- vir_genecluster_annot %>% filter(KO %in% c("K02290")) #0
#K08929 - pufM photosynthetic reaction center M subunit
pufM <- vir_genecluster_annot %>% filter(KO %in% c("K08929")) #0
#lets clean environment by removing all genes tables with 0 results
rm(apcA_vir, apcA_vir, cpcA_vir, cpeA_vir, cpeT_vir, gapN_vir, gltA_vir, mcrA_vir, md_vir, napA_vir, nifD_vir, nifH_vir, nirS_vir, norB_vir, petC_vir, petH_vir, pmoA_vir, psaA_vir, pufM, rbcS_vir, rhd_vir)
rm(apcF_vir)
#we found vAlaS  - Shemin pathway, the cyanobacteria-specific antenna proteins: apcF, petE; photosyhnthesis essential protins: rbcL psbA; idh3 - isocitrate dehydrogenase IDH3 - mitochondria, nosZ_vir - denitrification; phosphoribulokinase - prkB

```
```{r analysis}
#prepare psbA for analysis
psbA_vir_plot <- psbA_vir %>% 
  pivot_longer(!colnames(psbA_vir)[c(1:8, length(psbA_vir[1,])-1, length(psbA_vir[1,]))], names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% left_join(metadata, by="sampleid")
```

```{r}
#plot it! against elbe kilometer
psbA_vir_plot$Stromkilometer <- sub(",", ".", psbA_vir_plot$Stromkilometer)
psbA_vir_plot$Stromkilometer <- as.double(psbA_vir_plot$Stromkilometer)
                                          
psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  filter(Quality != "low quality draft") %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + ggtitle("viral psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 

```
In metag mostly PsbA viral genes are increasing with kilometer value, when in metat in Jun and Mai the trend has peak in brackish waters of the middle of estuary (BB or TW)  and in Nov is decreasing with increased stormkilometer. 
```{r}
#check turbidity factor
psbA_vir_plot$Turbidity_TBDHereon <- sub(",", ".", psbA_vir_plot$Turbidity_TBDHereon)
psbA_vir_plot$Turbidity_TBDHereon <- as.double(psbA_vir_plot$Turbidity_TBDHereon)

psbA_vir_plot %>%  
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  filter(Quality != "low quality draft") %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, Turbidity_TBDHereon, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Turbidity_TBDHereon, y=counts)) + ggtitle("viral psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 

```

Too less turbidity values - do not divide into sampling dates
```{r}
psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  filter(Quality != "low quality draft") %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, Turbidity_TBDHereon, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Turbidity_TBDHereon, y=counts)) + ggtitle("viral psbA counts across different Turbidity values across different data sets") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type, scale = "free_y") + geom_smooth() 

```
Trend still hard to define 

```{r Chlorophyll}
psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
  filter(Quality != "low quality draft") %>%
    group_by(sampleid, Chlorophyll, data_type, Sample_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Chlorophyll, y=counts)) + ggtitle("viral psbA counts across different chlorophyll values across different data sets") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_type, scale = "free_y") + geom_smooth() 

```
Split values by FL and PA
```{r CO2}
psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(Quality != "low quality draft") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, dCO2_uM, data_type, Sample_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=dCO2_uM, y=counts)) + ggtitle("viral psbA counts across different CO2 values for different data sets") + geom_point() + scale_y_log10() + geom_tile() + facet_wrap(~data_type *Sample_type, scale = "free_y") + geom_smooth() 
dev.copy(jpeg, filename="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/viral_psbA_CO2.jpg")
dev.off ()

```
```{r CO2}
psbA_vir_plot$O2_TBDHereon <- sub(",", ".", psbA_vir_plot$O2_TBDHereon)
psbA_vir_plot$O2_TBDHereon <- as.double(psbA_vir_plot$O2_TBDHereon)
psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
 # filter(Quality != "low quality draft") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, dCO2_uM, O2_TBDHereon, data_type, Sample_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=log(counts), y=dCO2_uM)) + ggtitle("viral psbA counts across different CO2 values for different data sets")+ geom_smooth()  + geom_point(aes(x=log(counts), y=O2_TBDHereon))  + geom_smooth(aes(x=log(counts), y=O2_TBDHereon), colour="green")+ scale_y_log10() + geom_tile() + facet_wrap(~data_type, scale = "free_y") 
dev.copy(jpeg, filename="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/viral_psbA_CO2.jpg")
dev.off ()

```

```{r CO2}
psbA_prok_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, dCO2_uM, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=dCO2_uM, y=counts)) + ggtitle("Prokaryotic psbA counts across different CO2 values for different data sets") + geom_point()  + scale_y_log10() + geom_tile() + facet_wrap(~data_type, scale = "free_y") + geom_smooth(method = "lm") 
dev.copy(jpeg, filename="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/viral_psbA_CH4.jpg")
dev.off ()

```
```{r CO2}
psbA_prok_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, dCO2_uM, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=counts, y=dCO2_uM)) + ggtitle("Prokaryotic psbA counts across different CO2 values for different data sets") + geom_point()  + scale_y_log10() + geom_tile() + facet_wrap(~data_type, scale = "free_y") + geom_smooth(method = "lm") 
dev.copy(jpeg, filename="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/viral_psbA_CH4.jpg")
dev.off ()

```
```{r CH4}
psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, dCH4_nM, data_type, Sample_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=dCH4_nM, y=counts)) + ggtitle("viral psbA counts across different CH4 values for different data sets") + geom_point()  + scale_y_log10() + geom_tile() + facet_wrap(~data_type *Sample_type, scale = "free_y") + geom_smooth(method = "lm") 
dev.copy(jpeg, filename="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/viral_psbA_CH4.jpg")
dev.off ()

```


```{r CH4}
psbA_prok_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, dCH4_nM, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=dCH4_nM, y=counts)) + ggtitle("Prokaryotic psbA counts across different CH2 values for different data sets") + geom_point()  + scale_y_log10() + geom_tile() + facet_wrap(~data_type, scale = "free_y") + geom_smooth(method = "lm") 
dev.copy(jpeg, filename="C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/viral_psbA_CH4.jpg")
dev.off ()

```

No trend found 

```{r}

psbA_vir_plot$Salinity_TBDHereon <- sub(",", ".", psbA_vir_plot$Salinity_TBDHereon)
psbA_vir_plot$Salinity_TBDHereon <- as.double(psbA_vir_plot$Salinity_TBDHereon)

psbA_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, Salinity_TBDHereon, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_TBDHereon, y=counts)) + ggtitle("viral psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 

```
No trend really found
```{r}
petE_vir_plot <- petE_vir %>% 
  pivot_longer(!colnames(petE_vir)[c(1:8, length(petE_vir)-1, length(petE_vir))], names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% left_join(metadata, by="sampleid")
petE_vir_plot$Stromkilometer <- sub(",", ".", petE_vir_plot$Stromkilometer)
petE_vir_plot$Stromkilometer <- as.double(petE_vir_plot$Stromkilometer)

petE_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid,Stromkilometer, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Stromkilometer, y=counts)) + ggtitle("viral petE counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 

```

PetE genes counts has very different trends across different seasons in METAG samples, but METAT samples share very similar trend - decreasing with a peak in TW or SH. Very similar trends to PsbA

```{r}
petE_vir_plot$Turbidity_TBDHereon <- sub(",", ".", petE_vir_plot$Turbidity_TBDHereon)
petE_vir_plot$Turbidity_TBDHereon <- as.double(petE_vir_plot$Turbidity_TBDHereon)

petE_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid,Turbidity_TBDHereon, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Turbidity_TBDHereon, y=counts)) + ggtitle("viral petE counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 

```

```{r}
petE_vir_plot$Salinity_TBDHereon <- sub(",", ".", petE_vir_plot$Salinity_TBDHereon)
petE_vir_plot$Salinity_TBDHereon <- as.double(petE_vir_plot$Salinity_TBDHereon)

petE_vir_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid,Salinity_TBDHereon, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Salinity_TBDHereon, y=counts)) + ggtitle("viral petE counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + geom_smooth() 

```
Both turbidyity and saline factors are hard to analyse. 

Lets try to plot viral PsbA genes and hosts.
```{r vPsbA vs PsbA}
#load needed objects and prepare for analysis
psbA_prok <- readRDS(file = "C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/psbA_prok.RDS")
#vir_genecluster_annot <- readRDS("C:/Users/jgolebiowska/Documents/IGB_phd/virus/viral_abundances/vir_genecluster_annot.RDS")
#select psbA genes for this analysis
#long format viral data
psbA_vir_plot <- psbA_vir %>% 
  pivot_longer(!colnames(psbA_vir)[c(1:8, length(psbA_vir[1,]) - 1, length(psbA_vir[1,]))], names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>%
  left_join(metadata, by="sampleid")
#long format prokaryotic data
psbA_prok_plot <- psbA_prok %>% 
  pivot_longer(!colnames(psbA_prok)[1:19], names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% left_join(metadata, by="sampleid") %>% select(-colnames(psbA_prok)[3:19])
#add origin label
psbA_prok_plot["origin"] <- rep("host", times=length(psbA_prok_plot[1]))
psbA_vir_plot["origin"] <- rep("virus", times=length(psbA_vir[1]))
```

```{r}
colnames(psbA_prok_plot)[colnames(psbA_prok_plot) == 'QUERY'] <- "gene_cluster"
#remove viral gene clusters from prokaryotic data using earlier prepare vector of viral data 
psbA_prok_plot <- psbA_prok_plot[which(!(psbA_prok_plot$gene_cluster %in% psbA_vir_gene_clust)),]
psbA_all_plot <- full_join(psbA_prok_plot[, c("gene_cluster","AccessionNumber_TBDSven", "Station","Sample_date", "data_type", "Stromkilometer", "counts", "Sample_type", "Chlorophyll", "sampleid")], psbA_vir_plot[, c("gene_cluster", "AccessionNumber_TBDSven", "Station","Sample_date", "data_type", "Stromkilometer", "Sample_type", "Chlorophyll","counts", "sampleid")], by = c("gene_cluster", "AccessionNumber_TBDSven", "Station","Sample_date", "data_type", "Stromkilometer", "Sample_type", "Chlorophyll",  "sampleid"))
# #plot viral against bacterial
colnames(psbA_all_plot)[which(colnames(psbA_all_plot)=="counts.x")] <- "prokaryotic_PsbA" 
colnames(psbA_all_plot)[which(colnames(psbA_all_plot)=="counts.y")] <- "viral_PsbA" 

psbA_all_plot_origin <- rbind(psbA_prok_plot[, c("gene_cluster", "AccessionNumber_TBDSven", "Station","Sample_date", "Stromkilometer", "counts","data_type","Sample_type", "origin", "Chlorophyll")], psbA_vir_plot[,  c("gene_cluster", "AccessionNumber_TBDSven", "Station","Sample_date", "Stromkilometer", "counts","data_type","Sample_type", "origin",  "Chlorophyll")])


psbA_all_plot_origin %>%
  filter(Station != 'BunthausSpitze') %>%
  filter(Sample_date != "Nov 21") %>%
  group_by(Stromkilometer, Sample_date, Sample_type, data_type, origin) %>%
  summarise(counts=sum(as.numeric(counts), na.rm = TRUE)) %>%
  ggplot(aes(x=as.numeric(Stromkilometer), y=counts, fill=as.factor(origin))) + geom_point(aes(colour=as.factor(origin))) + geom_smooth(aes(colour=as.factor(origin))) +
  ggtitle("viral and prokaryotic psbA counts across different seasons and river sites") + scale_y_log10() + xlab("Kilometer") + geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") + labs(fill="Origin")

```

```{r}
psbA_all_plot_origin %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  group_by(Stromkilometer, Sample_date, Sample_type, data_type, origin, Chlorophyll) %>% 
  summarise(counts=sum(as.numeric(counts), na.rm = TRUE)) %>%
  ggplot(aes(x=as.numeric(Stromkilometer), y=counts, fill=as.factor(origin)))  + geom_point() + geom_smooth() + geom_point(aes(x=as.numeric(Stromkilometer), y=Chlorophyll), fill="green", colour="green") + geom_smooth(aes(x=as.numeric(Stromkilometer), y=Chlorophyll), colour="green") + ggtitle("viral and prokaryotic psbA counts across different seasons and river sites with Chlorophyll values")  +
  scale_y_log10() + 
  geom_tile() + facet_wrap(~data_type *Sample_date, scale = "free_y") 
```
Too less chlorophyll values
```{r}
psbA_all_plot_origin %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  #remove all counts assigned as viral from hosts
    group_by(Stromkilometer, Sample_date, Sample_type, data_type, origin) %>% 
  summarise(counts=sum(as.numeric(counts), na.rm = TRUE)) %>%
  ggplot(aes(x=as.numeric(Stromkilometer), y=counts, fill=as.factor(origin))) + geom_point(fill=as.factor(origin)) + geom_smooth() + ggtitle("viral and prokaryotic psbA counts across different fractions")  + labs(x="Elbe km", fill="origin") +
  scale_y_log10() + 
  geom_tile() + facet_wrap(~data_type *Sample_type, scale = "free_y") 
```


Plot viral and hosts against each other by data type

```{r}
psbA_all_plot[which(!is.na(prokaryotic_PsbA) & is.na(Quality)), prokaryotic_PsbA] <- NA
#edit psbA_all_plot
psbA_all_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
    filter(case_when(origin == "host" ~ is.na(Quality), origin == "virus" ~ !is.na(Quality))) %>%
  filter(Sample_date != "Nov 21") %>%   group_by(data_type, Sample_type, Stromkilometer, Sample_date) %>% 
    summarise(viral=sum(viral_PsbA, na.rm = TRUE), prokaryotic=sum(prokaryotic_PsbA,na.rm = TRUE)) %>%
  ggplot(aes(x=prokaryotic, y=viral)) + ggtitle("viral psbA against prokaryotic psbA counts") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type , scale = "free") + geom_smooth() 
```

```{r}
psbA_all_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%   group_by(data_type, Sample_type, Stromkilometer, Sample_date) %>% 
    summarise(viral=sum(viral_PsbA, na.rm = TRUE), prokaryotic=sum(prokaryotic_PsbA,na.rm = TRUE)) %>%
  ggplot(aes(x=log(prokaryotic), y=log(viral))) + ggtitle("viral psbA against prokaryotic psbA counts") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type *Sample_type , scale = "free") + geom_smooth() + xlab("prokaryotic") + ylab("viral")
```

Quite similar pattern with peak 
```{r}
psbA_all_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(data_type == "METAG") %>%
    group_by(Stromkilometer, Sample_type, Sample_date) %>% 
  summarise(viral=sum(viral_PsbA, na.rm = TRUE), prokaryotic=sum(prokaryotic_PsbA, na.rm = TRUE)) %>%
  ggplot(aes(x=prokaryotic, y=viral)) + ggtitle("viral and prokaryotic psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
           geom_tile() + facet_wrap(~Sample_type *Sample_date, scale = "free") + geom_smooth() 
```
ones again  peaks
```{r}
psbA_all_plot %>%  
  filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(data_type == "METAT") %>%
    group_by(Stromkilometer,data_type, Sample_type, Sample_date) %>% 
  summarise(viral=sum(viral_PsbA, na.rm = TRUE), prokaryotic=sum(prokaryotic_PsbA, na.rm = TRUE)) %>%
  ggplot(aes(x=prokaryotic, y=viral)) + ggtitle("viral and prokaryotic psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~Sample_type *Sample_date, scale = "free") + geom_smooth() 
```
Not enough data points 
Other interesting genes
## vAlaS
```{r}
#check turbidity factor for AlaS
AlaS_vir_plot <- AlaS_vir %>% 
  pivot_longer(!colnames(AlaS_vir)[c(1:8, length(AlaS_vir[1,])-1, length(AlaS_vir[1,]))], names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>%
  left_join(metadata, by="sampleid")
AlaS_vir_plot$Turbidity_TBDHereon <- sub(",", ".", AlaS_vir_plot$Turbidity_TBDHereon)
AlaS_vir_plot$Turbidity_TBDHereon <- as.double(AlaS_vir_plot$Turbidity_TBDHereon)

AlaS_vir_plot %>%  
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  #filter(Quality != "low quality draft") %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, Turbidity_TBDHereon, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Turbidity_TBDHereon, y=counts)) + ggtitle("viral psbA counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type, scale = "free_y") + geom_smooth() 

```
```{r}
#check chlorophyl factor for AlaS

AlaS_vir_plot %>%  
 # filter(Station != 'BunthausSpitze') %>% 
  filter(Sample_date != "Nov 21") %>%
  filter(!is.na(counts)) %>%
  filter(Quality != "low quality draft") %>%
  mutate(counts=counts/1000) %>%
    group_by(sampleid, Chlorophyll, Sample_date, data_type) %>% 
  summarise(counts=sum(counts)) %>%
  ggplot(aes(x=Chlorophyll, y=counts)) + ggtitle("viral AlaS counts across different seasons and river sites") + geom_point()  + scale_y_log10() +
  geom_tile() + facet_wrap(~data_type, scale = "free_y") + geom_smooth() 

```
#check if similar pattern to psbA genes but first needed to generate AlaS 
We can observe slightly decrease of ALaS values with increasing Chlorophyll
```{r}
unique(vibrant_annot$hypothetical.protein)
```

Data transformation
"log-transformation, the expression profiles were computed as the difference between the log2-transformed metatranscriptomic profile and the log2-transformed metagenomic profile" 

Gene Expression Changes and Community Turnover
Differentially Shape the Global Ocean Metatranscriptome

Normalise transcription by genes copy numbers 

```{r normalise}
#group by gene_cluster to optimise calculations
#chose only columns which we need
#select columns with metat 
metat_cols <- colnames(vir_genecluster_annot)[which(grepl("METAT", colnames(vir_genecluster_annot)))]
#find paired metag
paired_METAG <- sort(gsub("METAT", "METAG", metat_cols)) 
#check and save those which are existing
paired_METAG <- sort(paired_METAG[which(paired_METAG %in% colnames(vir_genecluster_annot))])
#save respected METAT columns 
paired_METAT <- sort(gsub("METAG", "METAT", paired_METAG))
#check if the same order
sum(gsub("METAG", "METAT", colnames(paired_METAG)) == colnames(paired_METAT)) == length(colnames(paired_METAT))
#select only needed columns and summurise by gene cluster 
vir_genecluster_annot_sum <- vir_genecluster_annot[, c("gene_cluster", paired_METAG, paired_METAT)] %>% 
  group_by(gene_cluster) %>% summarise_at(c(paired_METAG, paired_METAT), sum)
vir_metag <- vir_genecluster_annot_sum %>% select(contains("METAG"))
vir_metat <- vir_genecluster_annot_sum %>% select(contains("METAT"))
#find respective metagenomes for metatranscriptomes vir_metat
paired_METAG <- sort(gsub("METAT", "METAG", colnames(vir_metat))) 
colnames_vir_metag <- sort(colnames(vir_metag))
paired_vir_metag <- vir_metag %>% select(contains(paired_METAG)) #strange some METAT do not have 
#METAG
#then select metat which has paired METAG
paired_METAT <- sort(gsub("METAG", "METAT", colnames(paired_vir_metag)))
paired_vir_metat <- vir_metat[,paired_METAT]
#check the order 
#sort it
#== sub("METAT", "METAG", colnames(vir_metat))
#log transform both
#add small pseudocount avoid problems wth NaN values coming from zeros
log_paired_vir_metag <- log(paired_vir_metag + 0.000001)
log_paired_vir_metat <- log(paired_vir_metat + 0.000001)
#normalization interpreted as gene expression
norm_expression_paired_vir <- log_paired_vir_metat - log_paired_vir_metag
#transform to long format and combine with metatranscriptome
#add dummy column as is needed for reformating  
#norm_expression_paired_vir$dummy <- rep (c(1), times=length(norm_expression_paired_vir[,1])) 
norm_expression_paired_vir_long <- norm_expression_paired_vir %>%
  pivot_longer(colnames(norm_expression_paired_vir),  names_to = "sampleid", values_to = "counts") %>% left_join(metadata, by = "sampleid")
```

