---
title: "Viruses_basic_analysis"
author: "Julia Golebiowska"
date: "2024-08-06"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Viruses distribution analysis

In this Rmarkdown we are going to explore viral length, and metabolism and pathways distribution according to they niches and occurence.

```{r setup and loading}
library(dplyr)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)
library(vegan)

setwd("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances")
#load("~/IGB_phd/BICEST/virus/viral_abundances/viruses_basic_analysis.RData")
#should be in vir_genecluster_annot
##lets load the viral data 
vir_genecluster_annot <- readRDS(file = "vir_genecluster_annot.RDS")
colnames(vir_genecluster_annot)[1] <- "gene"
#dereplication data
drep_clusters <- read.csv("Cdb.csv", header = TRUE, sep = ",")
drep_clusters[,"genome"] = sub(".fna", "", drep_clusters[,"genome"])
#hosts assignment  data - iphop
host_prediction_genome = read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Host_prediction_to_genome_m90.csv")
host_prediction_genus = read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Host_prediction_to_genus_m90.csv")
host_prediction_genome <- host_prediction_genome %>% 
  separate(Host.taxonomy, c("domain", "phylum", "class", "order", "family",  "genus", "species"), ";")
path = "C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/metadata/"
metadata <- read.csv(paste0(path,"PhysicochemicalParameters_mod3.csv"), header=TRUE, sep=",", row.names=1) 
#$ Tidal freshwater (0-0.5)/oligohaline (0.5–5.5 psu), (B) mesohaline (5.5–15 psu), and (C) polyhaline (15–25 psu).
#table with cluster salinity niche characterisation
drep_clusters_fin_taxonomy_unified <- readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CCA_marker_gene/drep_clusters_fin_taxonomy_unified.RDS")

geneabund_drep_marker_taxonomy_long <- readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CCA_marker_gene/geneabund_drep_marker_taxonomy_long.RDS")

Combined_quality_summary <- read.csv(file="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Combined_quality_summary.tsv", sep="\t")

Combined_lytic_summary <- read.csv(file="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CombinedSamples_qualityinfo.tsv", sep="\t")

```

## 1. Length distribution analysis 

You can also embed plots, for example:

```{r statistic, echo=FALSE}

#METAG only 
Combined_quality_summary$contig_length <- as.numeric(Combined_quality_summary$contig_length)
Combined_quality_summary$completeness <- as.numeric(Combined_quality_summary$completeness)
Combined_quality_summary_METAG <- Combined_quality_summary %>% filter(grepl("METAG" , contig_id)) %>% 
  mutate(estimated_contig_length=(contig_length/completeness)*100)

#how many contigs
contigs_counts_METAG <- Combined_quality_summary_METAG %>% select("contig_id") %>% unique() %>% nrow()
#960924
contigs_counts_METAG_filtered <- Combined_quality_summary_METAG %>% filter(checkv_quality!="Low-quality" & checkv_quality!="Not-determined") %>% unique() %>% nrow()
#11496 

drep_clusters <-  Combined_quality_summary_METAG[,c("contig_id", "contig_length", "estimated_contig_length", "checkv_quality")] %>%  merge(drep_clusters, by.y="genome", by.x="contig_id") #%>% filter(checkv_quality!="Low-quality" & checkv_quality!="Not-determined")

drep_clusters <-  drep_clusters %>% mutate(AccessionNumber_TBDSven = str_extract(contig_id, "SAMEA[0-9]+"))
#add salinity categories
metadata <- metadata %>% mutate(Salinity_level =  case_when(Salinity_PSU <= .5  ~ 'freshwater',  Salinity_PSU > .5 & Salinity_PSU <= 5.5  ~ 'oligohaline', Salinity_PSU > 5.5 & Salinity_PSU <= 18  ~ 'mesohaline', Salinity_PSU > 18  ~ 'polyhaline'))
drep_clusters <- merge(drep_clusters, metadata[, c("AccessionNumber_TBDSven", "Sample_type", "Salinity_level")], all.x = T, all.y = F, by="AccessionNumber_TBDSven") 
drep_clusters$data_type <- rep(c("METAG"), each=nrow(drep_clusters))
drep_clusters[grep("METAT", drep_clusters$genome),"data_type"] <-"METAT"
drep_clusters$estimated_contig_length <- drep_clusters$estimated_contig_length/1000
tiff("hist_viral_contigs_length_METAG.tiff")
hist(drep_clusters$estimated_contig_length[which(drep_clusters$data_type=="METAG")], breaks=100, ylim=c(0,1500), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
# tiff("hist_viral_contigs_length_METAT.tiff")
# hist(drep_clusters$estimated_contig_length[which(drep_clusters$data_type=="METAT")], breaks=100, ylim=c(0,100), xlim=c(0,50), main = "Good quality transcripts length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
# dev.off()
#make histograms for salinity categories
tiff("hist_viral_contigs_length_METAG_oligo.tiff")
hist(drep_clusters$estimated_contig_length[which(drep_clusters$data_type=="METAG" & drep_clusters$Salinity_level=="oligohaline")], breaks=100, ylim=c(0,2500), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
tiff("hist_viral_contigs_length_METAG_meso.tiff")
hist(drep_clusters$estimated_contig_length[which(drep_clusters$data_type=="METAG" & drep_clusters$Salinity_level=="mesohaline")], breaks=100, ylim=c(0,1000), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
tiff("hist_viral_contigs_length_METAG_poly.tiff")
hist(drep_clusters$estimated_contig_length[which(drep_clusters$data_type=="METAG" & drep_clusters$Salinity_level=="polyhaline")], breaks=100, ylim=c(0,500), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
#lets visualise together 
drep_clusters$Salinity_level <- factor(drep_clusters$Salinity_level, level = c( "freshwater", "oligohaline",  "mesohaline", "polyhaline", NA))
capitalize <- function(string) {
  substr(string, 1, 1) <- toupper(substr(string, 1, 1))
  string
}
salinity_hist <- na.omit(drep_clusters) %>% ggplot()  + geom_histogram(aes(x=estimated_contig_length, y = ..density.., fill = Salinity_level), position = 'identity', bins = 100, na.rm = T, alpha=1) +
scale_fill_manual(values=c(brewer.pal(n = 4, name = "Blues")), breaks =  c("freshwater", "oligohaline",  "mesohaline", "polyhaline")) +  scale_colour_manual(values=c(brewer.pal(n = 4, name = "Blues")), breaks =  c("freshwater", "oligohaline",  "mesohaline", "polyhaline")) + scale_y_continuous(name="Relative Frequency", labels = scales::percent) + 
  scale_x_continuous(name="Length [kb]", breaks = seq(0, 100, 20), limits= c(0, 100)) + 
  # geom_vline(aes(xintercept = 38), color = "red", linetype = "dashed", size = 1) +
  # geom_vline(aes(xintercept = 20), color = "red", linetype = "dashed", size = 1) +
  # geom_vline(aes(xintercept = 29), color = "red", linetype = "dashed", size = 1) + 
  # geom_vline(aes(xintercept = 62), color = "red", linetype = "dashed", size = 1) + theme_minimal(base_size = 18) +
  geom_step(aes(x=estimated_contig_length, y = ..density.., fill = Salinity_level),
    stat = "bin", bins = 100,
    direction = "mid"
  ) + facet_grid(vars(Salinity_level), labeller = labeller(Salinity_level = capitalize)) + guides(fill="none") + theme_bw(base_size = 20)

tiff("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/length_salinity_level.tiff")
salinity_hist
dev.off()
drep_clusters$Sample_type <- factor(drep_clusters$Sample_type, level = c("Light_fraction", "Heavy_fraction", "Free_living"))
fraction_hist <- na.omit(drep_clusters) %>% ggplot(aes(estimated_contig_length, fill = Sample_type, na.rm = TRUE)) + geom_histogram(position = 'identity', bins = 100, na.rm = T)
tiff("length_fraction_level.tiff")
fraction_hist
dev.off()
#to find a maximum manualy
# stuff <- ggplot_build(salinity_hist)
# stuff$data[[1]]$density
```

```{r}
#size distribution METAT
Combined_quality_summary_METAT <- Combined_quality_summary %>% filter(grepl("METAT" , contig_id)) %>% 
  mutate(estimated_contig_length=(contig_length/completeness)*100)

Combined_quality_summary_METAT$estimated_contig_length <- as.numeric(Combined_quality_summary_METAT$estimated_contig_length)
hist(Combined_quality_summary_METAT$estimated_contig_length, breaks=100)

```

## 2. Metabolism and pathways analysis
```{r plot metabolism}
#Prepare data
pathways <- read.csv("Combined_VIBRANT_AMG_pathways_GROS22_SAMEA.scaffolds.min0.tsv", sep='\t')
#put all different Glycosphingolipids biosynthesis into one category 
pathways[which(grepl("Glycosphingolipid", pathways$Pathway)), "Pathway"] <- "Glycosphingolipid biosynthesis"
#separate KO so we can use them as a key for merging to data frames remove redndant 
pathways <-  pathways %>% separate_rows(Present.AMG.KOs) %>% unique() 

geneabund <- read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/workshop_gene_catalog/BICEST_cellabund.tsv", sep="\t", skip=1, header = TRUE, row.names = 1)

# geneabund$gene_cluster <- rownames(geneabund)
colnames(vir_genecluster_annot)[1] <- "KO"
#merge with pathways to include metabolism information related to found axuliary genes by vibrant - selecting axuliary genes
vir_genecluster_annot <- vir_genecluster_annot[, c("genome", "KO", "DESCRIPTION", "gene_cluster")] %>% merge(pathways, by.x="KO", by.y="Present.AMG.KOs") # find more pathways 
#add drep clusters 
#but first unify genome names
vir_genecluster_annot <- vir_genecluster_annot %>% 
  mutate(genome=str_remove(genome, "_[0-9]+$"))
vibrant_genecluster_drep = merge(drep_clusters, vir_genecluster_annot, by.x="contig_id",  by.y="genome", all.y = T) %>% select(contig_id, gene_cluster, KO, primary_cluster, secondary_cluster, DESCRIPTION, Metabolism, Pathway) %>% unique()
# add gene abundance for plotting
# KOs can have multiple clusters therefore we can merge counts using  by gene cluster as a key
# we can sum them to get KO abundances
#unique(vibrant_genecluster_drep[which(vibrant_genecluster_drep$KO=="K14266"), "gene_cluster"])
vibrant_genecluster_drep_abund <- merge(vibrant_genecluster_drep, geneabund, by.x="gene_cluster", by.y=0, all.x = T)
#saveRDS(vibrant_genecluster_drep_abund, file ="vibrant_genecluster_drep_abund.RDS")
#change to long format for convenient analysis with metadata
#merge metadata so we are not analysis genomes without metadata
vibrant_genecluster_drep_abund <- vibrant_genecluster_drep_abund %>% pivot_longer(!colnames(vibrant_genecluster_drep_abund)[1:9], names_to = "sampleid", values_to = "counts", values_drop_na = 0) %>% merge(metadata, by="sampleid")
vibrant_genecluster_drep_abund$Sample_date <- factor(vibrant_genecluster_drep_abund$Sample_date, levels= c("May-21", "Jul-21", "Feb-22", "Jun-22",  "May-22", "Nov-22"))
saveRDS(vibrant_genecluster_drep_abund, file ="vibrant_genecluster_drep_abund.RDS")
save.image("viruses_basic_analysis.RData")
```

```{r}
#visualisations
#fractions
pathways_fractions_plot <- vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_type)) %>%  ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=reorder(sampleid, StationNumber), y=counts, fill = Pathway)) + facet_wrap(~data_type *Sample_type, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_x_discrete(labels = NULL, breaks = NULL) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) +  theme_bw(base_size=20) + xlab("")
tiff("Pathways_fractions_plot.tiff", unit="px", width = 1000, height = 1000)
pathways_fractions_plot
dev.off()

#fractions metabolism
new_labels <- c("Light_fraction" = "Suspended particles", "Heavy_fraction" = "Sinking particles", "Free_living" = "Free-living", "METAT" = "METAT", "METAG" = "METAG")

metabolism_fractions_plot <- vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_type)) %>%  ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=reorder(sampleid, StationNumber), y=counts, fill = Metabolism)) + facet_wrap(~data_type *Sample_type, scales= "free_x",labeller = as_labeller(new_labels)) + scale_y_continuous(labels=scales::percent_format()) + scale_x_discrete(labels = NULL, breaks = NULL) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) +xlab("") +  theme_bw(base_size=20)
#hard to interpret - big variability - probably fraction is not the most important factor
#differences between fl and PA for pathways for metabolism in metag quite similar but for both inmetat quite the difference even between heavy and light fraction
tiff("Metabolism_fractions_plot.tiff", unit="px", width = 1000, height = 1000)
metabolism_fractions_plot
dev.off()
#sampling date

pathways_date_plot <- vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_date)) %>%  ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=sampleid, y=counts, fill = Pathway)) + facet_wrap(~data_type *Sample_date, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_x_discrete(labels = NULL, breaks = NULL) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), legend.text = element_text(size=10)) + theme_bw(base_size=20)
tiff("Pathways_date_plot.tiff", unit="px", width = 1000, height = 1000)
pathways_date_plot
dev.off()

#fractions metabolism
metabolism_date_plot <- vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_date)) %>%  ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=reorder(sampleid, StationNumber), y=counts, fill = Metabolism)) + facet_wrap(~data_type *Sample_date, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_x_discrete(labels = round(vibrant_genecluster_drep_abund$Stromkilometer,0)) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), axis.text.x = element_text(angle = 90, hjust = 1))   + theme_bw(base_size=20)   #hard to interpret - big variablity - check stations more variability in metat 
tiff("metabolism_date_plot.tiff", unit="px", width = 1000, height = 1000)
metabolism_date_plot
dev.off()

#lets check stations 
metabolism_Station_plot <- vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_date)) %>%  ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=as.factor(round(Stromkilometer,0)), y=counts, fill = Metabolism)) + facet_wrap(~data_type *Sample_date, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_y_continuous(labels=scales::percent_format()) + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) + xlab("Elbe km") + theme_bw(base_size=20)

tiff("metabolism_station_plot.tiff", unit="px", width = 1000, height = 1000)
metabolism_Station_plot
dev.off()
#nice!
palette <- c(
  "dodgerblue2", "#E31A1C", # red
  "green4",
  "#6A3D9A", # purple
  "#FF7F00", # orange
  "black", "gold1",
  "skyblue2", "#FB9A99", # lt pink
  "palegreen2",
  "#CAB2D6", # lt purple
  "#FDBF6F", # lt orange
  "gray70", "khaki2",
  "maroon", "orchid1", "deeppink1", "blue1", "steelblue4",
  "darkturquoise", "green1", "yellow4", "yellow3",
  "darkorange4", "brown", "dodgerblue2"
)
vibrant_genecluster_drep_abund_pathway <- vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_date)) %>% group_by(Stromkilometer, Sample_date, data_type) %>% mutate(Percentage = counts / sum(counts) * 100) %>% group_by(Stromkilometer, Sample_date, data_type, Pathway) %>% mutate(Percentage = sum(Percentage)) %>% mutate(Pathway =  case_when(Percentage <= 2 ~  "Other", TRUE ~ Pathway )) %>% group_by(Stromkilometer, Sample_date, data_type, Pathway) %>% unique()

found_pathways<-sort(unique(vibrant_genecluster_drep_abund_pathway$Pathway))
vibrant_genecluster_drep_abund_pathway$Pathway <- factor(vibrant_genecluster_drep_abund_pathway$Pathway, levels= c(found_pathways[which(found_pathways!="Other")], "Other"))
vibrant_genecluster_drep_abund_pathway$Sample_date <- factor(vibrant_genecluster_drep_abund_pathway$Sample_date, levels= c("May-21", "Jul-21", "Feb-22", "Jun-22",  "May-22", "Nov-22"))

pathway_Station_plot <-  vibrant_genecluster_drep_abund_pathway %>% ggplot(aes(x=as.factor(round(Stromkilometer,0)), y = Percentage, fill = Pathway)) + geom_bar(stat='identity',  position = "fill") + facet_wrap(~data_type * factor(Sample_date), scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + theme_bw(base_size=18) + theme(axis.title.y = element_text(angle=90), legend.text = element_text(size=10)) + xlab("Elbe km") + ylab("Relative abundance") + scale_fill_manual(values = rep(palette,3)) + guides(fill=guide_legend(ncol=1)) 


tiff("pathway_station_plot.tiff", unit="px", width = 1010, height = 1000)
pathway_Station_plot
dev.off()
vibrant_genecluster_drep_abund %>% filter(counts>0 & !is.na(Sample_date)) %>% group_by(Stromkilometer, Sample_date, data_type, Pathway) %>% mutate(Percentage = counts / sum(counts) * 100) %>% select(c("DESCRIPTION", "Metabolism",  "Pathway",  "length", "counts", "Percentage")) %>% head()
```

```{r}
#check if viruses of different niches have different functions - exhibit different pathways 
drep_clusters_fin_taxonomy_unified <- readRDS(file = "C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CCA_marker_gene/drep_clusters_fin_taxonomy_unified.RDS")
#keep only clusters with salinity data - some scaffolds form cruises without salinity
vibrant_genecluster_drep_abund_quality <- vibrant_genecluster_drep_abund %>% merge(Combined_quality_summary) %>% merge(drep_clusters_fin_taxonomy_unified, by="primary_cluster")
#analyse complete
vibrant_genecluster_drep_abund_quality_complete <- vibrant_genecluster_drep_abund_quality[which(vibrant_genecluster_drep_abund_quality$checkv_quality=="Complete"),]
#check the taxonomy
unique(vibrant_genecluster_drep_abund_quality_complete$level_5)
# "Caudoviricetes" only
vibrant_genecluster_drep_abund_quality_hq <- vibrant_genecluster_drep_abund_quality[which(vibrant_genecluster_drep_abund_quality$checkv_quality=="High-quality"),]
#check the taxonomy complete
unique(vibrant_genecluster_drep_abund_quality_complete$level_5)
# "Caudoviricetes" only
#check the taxonomy high quality 
unique(vibrant_genecluster_drep_abund_quality_hq$level_5)
# "Caudoviricetes" only
#pathways 
#check the taxonomy complete
unique(vibrant_genecluster_drep_abund_quality_complete$Pathway)
# "Caudoviricetes" only
#check the taxonomy high quality 
unique(vibrant_genecluster_drep_abund_quality_hq$Pathway)
#check carbon fixation pathway in potosyyntetic organisms - are those
#cyanophages?
vibrant_genecluster_drep_abund_quality_photo <- vibrant_genecluster_drep_abund_quality[which((vibrant_genecluster_drep_abund_quality$checkv_quality=="High-quality" | vibrant_genecluster_drep_abund_quality$checkv_quality=="Complete") & vibrant_genecluster_drep_abund_quality$Pathway=="Carbon fixation in photosynthetic organisms"),] %>% merge(host_prediction_genome)

# "Caudoviricetes" only
drep_clusters_fin_taxonomy_unified <- drep_clusters_fin_taxonomy_unified %>% filter(!is.na(Salinity_niche))
vibrant_genecluster_drep_abund_niche <- merge(vibrant_genecluster_drep_abund, drep_clusters_fin_taxonomy_unified, by="primary_cluster")

#add quality 
salinity_niche_pathways_plot <- vibrant_genecluster_drep_abund_niche  %>% filter(counts>0 & !is.na(Sample_date)) %>% group_by(sampleid, Stromkilometer, Salinity_niche, data_type, Pathway) %>% summarise(counts=sum(counts)) %>% ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=as.factor(round(Stromkilometer,0)), y=counts, fill = Pathway)) + facet_wrap(~data_type *Salinity_niche, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_y_continuous(labels=scales::percent_format()) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), legend.text = element_text(size=10)) + xlab("Elbe km") + scale_fill_manual(values = palette)
#weird - very different specialize functions ... many NA for salinity niche - correct it 
tiff("salinity_niche_pathways_plot.tiff", unit="px", width = 1000, height = 1000)
salinity_niche_pathways_plot
dev.off()

salinity_niche_Metabolism_plot <- vibrant_genecluster_drep_abund_niche  %>% filter(counts>0 & !is.na(Sample_date)) %>% group_by(sampleid, Stromkilometer, Salinity_niche, data_type, Metabolism) %>% summarise(counts=sum(counts)) %>% ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=as.factor(round(Stromkilometer,0)), y=counts, fill = Metabolism)) + facet_wrap(~data_type *Salinity_niche, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_y_continuous(labels=scales::percent_format()) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) + xlab("Elbe km") + scale_fill_manual(values = palette)

tiff("salinity_niche_Metabolism_plot.tiff", unit="px", width = 1000, height = 1000)
salinity_niche_Metabolism_plot
dev.off()
#fraction niche

Fraction_niche_pathways_plot <- vibrant_genecluster_drep_abund_niche  %>% filter(counts>0 & !is.na(Sample_date)) %>% group_by(sampleid, Stromkilometer, Fraction_niche, data_type, Pathway) %>% summarise(counts=sum(counts)) %>% ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=reorder(sampleid, Stromkilometer), y=counts, fill = Pathway)) + facet_wrap(~data_type *Fraction_niche, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_y_continuous(labels=scales::percent_format()) + ggtitle("Relative abundace of pathways across different fractions") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), legend.text = element_text(size=10)) + xlab("Elbe km") + scale_fill_manual(values = palette)

Fraction_niche_pathways_plot <- vibrant_genecluster_drep_abund_niche  %>% filter(counts>0 & !is.na(Sample_date)) %>% ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=Fraction_niche, y=counts, fill = Pathway)) + scale_y_continuous(labels=scales::percent_format()) + scale_y_continuous(labels=scales::percent_format()) + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), legend.text = element_text(size=10)) + xlab("Elbe km") + scale_fill_manual(values = palette)
#weird - very different specialize functions ... many NA for alinity niche - correct it 
tiff("Fraction_niche_pathways_plot.tiff", unit="px", width = 1000, height = 1000)
Fraction_niche_pathways_plot
dev.off()

Fraction_niche_Metabolism_plot <- vibrant_genecluster_drep_abund_niche  %>% filter(counts>0 & !is.na(Sample_date)) %>% group_by(sampleid, Stromkilometer, Fraction_niche, data_type, Metabolism) %>% summarise(counts=sum(counts)) %>% ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=as.factor(round(Stromkilometer,0)), y=counts, fill = Metabolism)) + facet_wrap(~data_type *Fraction_niche, scales= "free_x") + scale_y_continuous(labels=scales::percent_format()) + scale_y_continuous(labels=scales::percent_format()) + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12)) + xlab("Elbe km") + scale_fill_manual(values = palette)

tiff("Fraction_niche_Metabolism_plot.tiff", unit="px", width = 1000, height = 1000)
Fraction_niche_Metabolism_plot
dev.off()


```
#3. Hosts analysis specialist and generalist
```{r , echo=FALSE}
#check hosts 
drep_clusters <- merge(drep_clusters, host_prediction_genome, by.x ="genome", by.y = "Virus", all.x = T, all.y = F)
drep_clusters_hosts_sum <-aggregate(phylum ~ primary_cluster , data = drep_clusters, FUN = function(x) length(unique(x)))
# Sort the pirates dataframe by height in decreasing order
drep_clusters_hosts_sum <- drep_clusters_hosts_sum[order(drep_clusters_hosts_sum$phylum, decreasing = TRUE),]
sum(drep_clusters_hosts_sum$phylum > 1 ) #126 clusters with host of multiple phylum - generalists
sum(drep_clusters_hosts_sum$phylum == 1 ) #558
length(unique(drep_clusters$primary_cluster)) - nrow(drep_clusters_hosts_sum) #1688

#no associated host
drep_sec_clusters_hosts_sum <-aggregate(phylum ~ secondary_cluster , data = drep_clusters, FUN = function(x) length(unique(x)))
# Sort the pirates dataframe by height in decreasing order
drep_sec_clusters_hosts_sum <- drep_sec_clusters_hosts_sum[order(drep_sec_clusters_hosts_sum$phylum, decreasing = TRUE),]
length(unique(drep_clusters$secondary_cluster)) - nrow(drep_sec_clusters_hosts_sum) #1720
drep_sec_clusters_hosts_sum <-aggregate(phylum ~ secondary_cluster , data = drep_clusters, FUN = function(x) length(unique(x)))
#more then one host - generalists
sum(drep_sec_clusters_hosts_sum$phylum > 1 ) #126
sum(drep_sec_clusters_hosts_sum$phylum == 1 ) #570
#no associated host
length(unique(drep_clusters$secondary_cluster)) - nrow(drep_sec_clusters_hosts_sum) #1720
```

```{r lytic and lysogenic , echo=FALSE}
#lytic and lysogenic 
#check if they are consistent with in viral cluster
drep_clusters <- read.csv("Cdb.csv", header = TRUE, sep = ",")
drep_clusters[,"genome"] = sub(".fna", "", drep_clusters[,"genome"])
# drep_clusters <- merge(drep_clusters, metadata[, c("AccessionNumber_TBDSven", "Sample_type", "Salinity_level")], all.x = T, all.y = F, by="AccessionNumber_TBDSven") 
Combined_lytic_summary <- read.csv(file="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CombinedSamples_qualityinfo.tsv", sep="\t")
Combined_lytic_summary$scaffold <- gsub("\\s", "_", Combined_lytic_summary$scaffold)
Combined_lytic_summary <-distinct(Combined_lytic_summary)
drep_clusters_lytic <- left_join(drep_clusters, Combined_lytic_summary, by=c("genome"="scaffold"))
#check how many different lytic lysogenic values with in cluster
lytic_lysognic_cluster <- drep_clusters_lytic %>% group_by(primary_cluster) %>% summarise(value=n_distinct(type))
#around 100 has different values
#if 80% of them has cetain valu unify - else drop them
find_niche <-function(x, threshold){
  #calculate rel frequencies
  tab = prop.table(table(x))
  #check id=f there are values >= 0.8
  #then take this value
  if(any(tab>=threshold)){
    names(tab[which(tab>=threshold)])
  }
  #else concatenate values which are higher the 0.2
  else {
    "not identifiable"
  }}
#check again %>% group_by(primary_cluster) #mistake should not be a type here
drep_clusters_lytic <- drep_clusters_lytic %>% group_by(primary_cluster) %>% mutate(lytic_lysogenic= find_niche(type, 0.8)) %>% ungroup() %>% filter(lytic_lysogenic!="not identifiable")

geneabund_drep_marker_taxonomy_long_type <- left_join(geneabund_drep_marker_taxonomy_long, drep_clusters_lytic[,c("primary_cluster", "type")], by = "primary_cluster")
#plot against spatio-temporal
lytic_lysogenic_barplot<- geneabund_drep_marker_taxonomy_long_type %>% ggplot() + geom_bar(position ="fill", stat = "identity", aes(x=Station, y=counts, fill = lytic_lysogenic)) + scale_x_discrete(labels = NULL, breaks = NULL) +
  scale_y_continuous(labels=scales::percent_format())  + ggtitle("Relative abundace of lytic and lysogenic viruses across spatio-temporal scales") + ylab("Relative abundace") + theme(axis.title.y = element_text(angle=90, size=12), legend.key = element_rect(fill = "transparent")) + guides(fill=guide_legend(title="Class")) + facet_wrap(~data_type *Sample_date)
tiff("Lytic_lysogenic_bar.tiff", unit="px", width = 1000, height = 1000)
lytic_lysogenic_barplot
dev.off()
geneabund_drep_marker_taxonomy_wide_type <- geneabund_drep_marker_taxonomy_long_type %>% group_by(Salinity_level, Salinity_PSU, Stromkilometer, type, Station, Sample_date, AccessionNumber_TBDSven) %>% summarise(sum=sum(counts), Salinity_level, Salinity_PSU, Stromkilometer, type, Station, Sample_date, data_type, Sample_type, AccessionNumber_TBDSven) %>% distinct() %>% ungroup() %>% select(c(sum, Salinity_PSU, Stromkilometer, type, Station, Sample_date, data_type, Sample_type, Salinity_level,  AccessionNumber_TBDSven)) %>% pivot_wider(values_from = sum, names_from = type) %>% mutate(ratio=lytic/lysogenic)
#
lytic_lysogenic_scatter_fraction <- geneabund_drep_marker_taxonomy_wide_type %>% ungroup()%>% ggplot(aes(x=Stromkilometer, y=ratio)) + geom_point(aes(x=Stromkilometer, y=ratio)) + geom_smooth(aes(x=Stromkilometer, y=ratio)) + ggtitle("Lytic to lysogenic ratios across spatio-temporal scales") + ylab("Lytic to lysogenic ratio")+ xlab("Elbe km") + theme(axis.title.y = element_text(angle=90, size=12), legend.key = element_rect(fill = "transparent")) + facet_wrap( ~Sample_type *data_type)  

lytic_lysogenic_scatter_season <- geneabund_drep_marker_taxonomy_wide_type %>% ggplot(aes(x=Stromkilometer, y=ratio)) + geom_point() + ggtitle("Lytic to lysogenic ratios across spatio-temporal scales") + ylab("Lytic to lysogenic ratio")+ xlab("Elbe km") + theme(axis.title.y = element_text(angle=90, size=12), legend.key = element_rect(fill = "transparent")) + facet_wrap( ~data_type *Sample_date, scales='free') + geom_smooth()

lytic_lysogenic_scatter_salinity <- geneabund_drep_marker_taxonomy_wide_type %>% ggplot(aes(x=Salinity_PSU, y=ratio)) + geom_point() + geom_smooth(span=2) + ggtitle("Lytic to lysogenic ratios across salinity") + ylab("Lytic to lysogenic ratio")+ xlab("Salinity PSU") + theme(axis.title.y = element_text(angle=90, size=12), legend.key = element_rect(fill = "transparent")) + facet_wrap( ~data_type, scales='free')

#try to visualise lytic-to-lysogenic ratios
tiff("lytic_lysogenic_scatter_fraction.tiff")
lytic_lysogenic_scatter_fraction
dev.off()
tiff("lytic_lysogenic_scatter_season.tiff")
lytic_lysogenic_scatter_season
dev.off()
tiff("lytic_lysogenic_scatter_salinity.tiff")
lytic_lysogenic_scatter_salinity
dev.off()
```
##4. Isoelectric point
```{r}
IP <- read.csv("Isolectric_proteom.csv")
#separate header column into multiple informative columns
IP <- IP %>%  separate_wider_delim(header,"\t", names=c("genome", "range", "no", "VOG", "protein name"))
#edit genome to marge with drep cluster
#colnames(IP) <- c("x", "genome", "IP")
IP$genome <- gsub("_[1-9]$", "", IP$genome)
IP_drep <- merge(IP, drep_clusters)
IP_drep <- merge(IP_drep, drep_clusters_fin_taxonomy_unified)
IP_oligo <- IP_drep$IP[which(IP_drep$Salinity_niche=="oligohaline")]
IP_fresh <- IP_drep$IP[which(IP_drep$Salinity_niche=="freshwater")]
IP_meso <- IP_drep$IP[which(IP_drep$Salinity_niche=="mesohaline")]
IP_poly <- IP_drep$IP[which(IP_drep$Salinity_niche=="polyhaline")]
mean(IP_oligo)
mean(IP_meso)
mean(IP_poly)
mean(IP_fresh)
#select MCP
MCP_list <- c("VOG00035", "VOG00461", "VOG01150", "VOG01000", "VOG01164", "VOG01920", "VOG02155", "VOG02220", "VOG02554", "VOG03097", "VOG03553", "VOG03780", "VOG03813", "VOG33139")
#MCP_list <- c("VOG4553, VOG4773")
MCP_list <- c("PF03864.15", "PF04451.12", "PF05125.12", "VOG00633", "VOG01437", "VOG06289", "VOG06316", "VOG10260", "VOG10754")
#look for major capsid proteins these are false
IP_drep_MCP <- IP_drep %>% filter(VOG %in% MCP_list)
IP_oligo <- IP_drep_MCP$IP[which(IP_drep_MCP$Salinity_niche=="oligohaline")]
IP_fresh <- IP_drep_MCP$IP[which(IP_drep_MCP$Salinity_niche=="freshwater")]
IP_meso <- IP_drep_MCP$IP[which(IP_drep_MCP$Salinity_niche=="mesohaline")]
IP_poly <- IP_drep_MCP$IP[which(IP_drep_MCP$Salinity_niche=="polyhaline")]
mean(IP_fresh)
mean(IP_oligo)
mean(IP_meso)
mean(IP_poly)
wilcox.test(IP_fresh, IP_oligo, data=IP_drep)
wilcox.test(IP_fresh, IP_meso, data=IP_drep)
#wilcox.test(IP_fresh, IP_poly, data=IP_drep) #not enough observations
wilcox.test(IP_oligo, IP_meso, data=IP_drep)
#wilcox.test(IP_oligo, IP_poly, data=IP_drep) #not enough observations
#wilcox.test(IP_meso, IP_poly, data=IP_drep) #not enough observations
#not significant
var(IP_fresh)
var(IP_oligo)
var(IP_meso)
var(IP_poly)

aov_sali <- aov(IP ~ Salinity_niche, data=IP_drep)
summary(aov_sali)

IP_FL <- IP_drep$IP[which(IP_drep_MCP$Fraction_niche=="Free_living")]
IP_HF <- IP_drep$IP[which(IP_drep_MCP$Fraction_niche=="Heavy_fraction")]
IP_LF <- IP_drep$IP[which(IP_drep_MCP$Fraction_niche=="Light_fraction")]
mean(IP_FL)
mean(IP_HF)
mean(IP_LF)
wilcox.test(IP_FL, IP_oligo, data=IP_drep)
wilcox.test(IP_HF, IP_meso, data=IP_drep)
#wilcox.test(IP_fresh, IP_poly, data=IP_drep) #not enough observations
wilcox.test(IP_LF, IP_meso, data=IP_drep)
#not significant
var(IP_FL)
var(IP_HF)
var(IP_LF)
aov_fraction <- aov(IP ~ Fraction_niche, data=IP_drep)
anosim_salinity <- anosim(IP_drep[which(!is.na(IP_drep$Salinity_niche)), "IP" ], Salinity_niche, permutations = 999, distance = "euclidian", strata = NULL)

summary(aov_fraction)

```
```{r}

IP_hist_MCP <-IP_drep_MCP %>% filter(Salinity_niche %in% c("freshwater", "oligohaline", "mesohaline", "polyhaline")) %>% ggplot(aes(x=IP, y = ..density.., color =Salinity_niche))  + #geom_histogram(bin=30, na.rm = T,  alpha=0.5, color=NA,  aes(fill =Salinity_niche))  + 
  scale_x_continuous(name="pI [pH]")  + scale_y_continuous(name="Relative Frequency", labels = scales::percent_format(scale = 1)) + scale_color_manual(values=c(brewer.pal(n = 4, name = "Blues")), name = "Salinity niche") + scale_fill_manual(values=c(brewer.pal(n = 4, name = "Blues")))  + 
 geom_density(size =1, adjust = 0.6) + theme_dark(base_size = 18) #+ facet_wrap(~Salinity_niche)
tiff("Pi_hist_MCPs.tiff")
IP_hist_MCP
dev.off()
```

```{r}
IP_hist <-IP_drep %>% filter(Salinity_niche %in% c("freshwater", "oligohaline", "mesohaline", "polyhaline")) %>% ggplot(aes(x=IP, y = ..density.., color =Salinity_niche)) + 
  scale_x_continuous(name="Isoelectric point [pH]")  + scale_y_continuous(name="Relative Frequency", labels = scales::percent) + scale_color_manual(values=c(brewer.pal(n = 4, name = "Blues")), name = "Salinity niche") + geom_density(size =1, adjust = 0.3) + theme_dark(base_size = 18)
tiff("Pi_hist.tiff")
IP_hist
dev.off()
```

