---
title: "Viruses_basic_analysis"
author: "Julia Golebiowska"
date: "2024-08-06"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

In this Rmarkdown we are going to explore viral dataset to have better understanding of the data. 
```{r setup and loading}
library(dplyr)
library(tidyverse)
library(ggplot2)
library(RColorBrewer)

setwd("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances")
#should be in vir_genecluster_annot
##lets load the viral data 
vir_genecluster_annot <- readRDS(file = "vir_genecluster_annot.RDS")
colnames(vir_genecluster_annot)[1] <- "gene"
#dereplication data
drep_clusters <- read.csv("Cdb.csv", header = TRUE, sep = ",")
drep_clusters[,"genome"] = sub(".fna", "", drep_clusters[,"genome"])
#hosts assignment  data - iphop
host_prediction_genome = read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Host_prediction_to_genome_m90.csv")
host_prediction_genus = read.csv("C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/Host_prediction_to_genus_m90.csv")
host_prediction_genome <- host_prediction_genome %>% 
  separate(Host.taxonomy, c("domain", "phylum", "class", "order", "family",  "genus", "species"), ";")
path = "C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/metadata/"
metadata <- read.csv(paste0(path,"PhysicochemicalParameters_mod3.csv"), header=TRUE, sep=",", row.names=1) 
#$A) Tidal freshwater/oligohaline (0–5 psu), (B) mesohaline (5–15 psu), and (C) polyhaline (15–25 psu).
#table with cluster salinity niche characterisation
drep_clusters_fin_taxonomy_unified <- readRDS(file ="C:/Users/jgolebiowska/Documents/IGB_phd/BICEST/virus/viral_abundances/CCA_marker_gene/drep_clusters_fin_taxonomy_unified.RDS")
```

## Including Plots

You can also embed plots, for example:

```{r statistic, echo=FALSE}
#8188834
#number of contigs
#divide into METAG
# vibrant_annot_METAG <- vibrant_annot %>% filter(grepl("METAG" , gene))
# 
# #divide into METAT 
# vibrant_annot_METAT <- vibrant_annot %>% filter(grepl("METAT" , gene))
#for each generate contig column 
vir_genecluster_annot <- vir_genecluster_annot %>%
  mutate(contig=str_remove(gene, ">GROS.*_SAMEA.*_METAG-scaffold_")) %>%
  mutate(contig=str_remove(contig, "_length=.*_orig=NODE_.*_length_.*_cov_.*"))
#METAG
vibrant_annot_METAG <- vir_genecluster_annot %>% filter(grepl("METAG" , gene)) %>% 
  mutate(contig=str_remove(gene, ">GROS.*_SAMEA.*_METAG-scaffold_")) %>%
  mutate(contig=str_remove(contig, "_length=.*_orig=NODE_.*_length_.*_cov_.*"))

#how many contigs
contigs_counts_METAG <- vibrant_annot_METAG %>% select("contig") %>% unique() %>% nrow()
#205611
#METAT
vibrant_annot_METAT <- vir_genecluster_annot %>% filter(grepl("METAT" , gene)) %>% 
  mutate(contig=str_remove(gene, ">GROS.*_SAMEA.*_METAT-transcript_")) %>%
  mutate(contig=str_remove(contig, "_length=.*_orig=NODE_.*_length_.*_cov_.*"))
#how many contigs
contigs_counts_METAT <- vibrant_annot_METAT %>% select("contig") %>% unique() %>% nrow()
#1985
contigs_counts_METAG + contigs_counts_METAT
#207596

#sizes of contigs METAG
vibrant_annot_METAG <- vibrant_annot_METAG %>% 
  mutate(length=str_remove(gene, ">GROS.*_SAMEA.*_METAG-scaffold_.*_length=.*_orig=NODE_.*_length_")) %>%
  mutate(length=str_remove(length, "_cov_.*"))
#check how many lengths and id they number equals contigs
vibrant_annot_METAG %>% select("length") %>% unique() %>% nrow()

#sizes of contigs METAT
vibrant_annot_METAT <- vibrant_annot_METAT %>% 
  mutate(length=str_remove(gene, ">GROS22-2_SAMEA.*_METAT-transcript_.*_length=.*_orig=NODE_.*_length_")) %>%
  mutate(length=str_remove(length, "_cov_.*"))
#check how many lengths and id they number equals contigs
vibrant_annot_METAT %>% select("length") %>% unique() %>% nrow()
#not at all
#size distribution for good quality contigs
#generate contig column
drep_clusters <-  drep_clusters %>% mutate(length=str_remove(genome, "GROS.*_SAMEA.*_METAG-scaffold_.*_length=.*_orig=NODE_.*_length_")) %>% mutate(length=str_remove(length, "_cov_.*")) 
drep_clusters <-  drep_clusters %>% mutate(length=str_remove(length, "GROS.*_SAMEA.*_METAT-transcript_.*_length=.*_orig=NODE_.*_length_")) %>% mutate(length=str_remove(length, "_cov_.*")) %>% mutate(length=str_remove(length, "_g[0-9]+_i[0-9]+$"))
drep_clusters <-  drep_clusters %>% mutate(AccessionNumber_TBDSven = str_extract(genome, "SAMEA[0-9]+"))
#add salinity categories
metadata <- metadata %>% mutate(Salinity_level = case_when(Salinity_PSU <= 5  ~ 'oligohaline', Salinity_PSU > 5 & Salinity_PSU <= 18  ~ 'mesohaline', Salinity_PSU > 18  ~ 'polyhaline'))
drep_clusters <- merge(drep_clusters, metadata[, c("AccessionNumber_TBDSven", "Sample_type", "Salinity_level")], all.x = T, all.y = F, by="AccessionNumber_TBDSven") 
drep_clusters$data_type <- rep(c("METAG"), each=nrow(drep_clusters))
drep_clusters[grep("METAT", drep_clusters$genome),"data_type"] <-"METAT"
drep_clusters$length <- as.numeric(drep_clusters$length)
drep_clusters$length <- drep_clusters$length/1000
tiff("hist_viral_contigs_length_METAG.tiff")
hist(drep_clusters$length[which(drep_clusters$data_type=="METAG")], breaks=100, ylim=c(0,1500), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
tiff("hist_viral_contigs_length_METAT.tiff")
hist(drep_clusters$length[which(drep_clusters$data_type=="METAT")], breaks=100, ylim=c(0,100), xlim=c(0,50), main = "Good quality transcripts length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
#make histograms for salinity categories
tiff("hist_viral_contigs_length_METAG_oligo.tiff")
hist(drep_clusters$length[which(drep_clusters$data_type=="METAG" & drep_clusters$Salinity_level=="oligohaline")], breaks=100, ylim=c(0,2500), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
tiff("hist_viral_contigs_length_METAG_meso.tiff")
hist(drep_clusters$length[which(drep_clusters$data_type=="METAG" & drep_clusters$Salinity_level=="mesohaline")], breaks=100, ylim=c(0,1000), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
tiff("hist_viral_contigs_length_METAG_poly.tiff")
hist(drep_clusters$length[which(drep_clusters$data_type=="METAG" & drep_clusters$Salinity_level=="polyhaline")], breaks=100, ylim=c(0,500), xlim=c(0,400), main = "Good quality contigs length distribution", xlab = "Length [kb]", cex.axis=1, cex.lab=1.5, cex.main=1.5)
dev.off()
#lets visualise together 
drep_clusters$Salinity_level <- factor(drep_clusters$Salinity_level, level = c("oligohaline", "mesohaline", "polyhaline", NA))
salinity_hist <- na.omit(drep_clusters) %>% ggplot(aes(length, fill = Salinity_level, na.rm = TRUE)) + geom_histogram(position = 'identity', bins = 100, na.rm = T) + 
  geom_step(
    stat = "bin", bins = 100,
    direction = "mid"
  ) + scale_fill_manual(values=c(brewer.pal(n = 3, name = "Blues")), breaks =  c("oligohaline", "mesohaline", "polyhaline")) + scale_colour_manual(values=c(brewer.pal(n = 3, name = "Blues")), breaks =  c("oligohaline", "mesohaline", "polyhaline")) +  scale_x_continuous(name="Length [kb]", limits=c(0, 100)) + 
  geom_vline(aes(xintercept = 38), color = "red", linetype = "dashed", size = 1) +
  geom_vline(aes(xintercept = 20), color = "red", linetype = "dashed", size = 1) + 
  geom_vline(aes(xintercept = 29), color = "red", linetype = "dashed", size = 1)
tiff("length_salinity_level.tiff")
salinity_hist
dev.off()
drep_clusters$Sample_type <- factor(drep_clusters$Sample_type, level = c("Light_fraction", "Heavy_fraction", "Free_living"))
fraction_hist <- na.omit(drep_clusters) %>% ggplot(aes(length, fill = Sample_type, na.rm = TRUE)) + geom_histogram(position = 'identity', bins = 100, na.rm = T)
tiff("length_fraction_level.tiff")
fraction_hist
dev.off()
#to find a maximum manualy
# stuff <- ggplot_build(salinity_hist)
# stuff$data[[1]]$density
```

```{r}
#size distribution METAT
vibrant_annot_METAT$length <- as.numeric(vibrant_annot_METAT$length)
hist(vibrant_annot_METAT$length, breaks=100)
#how many viral genomes proteins axuliary gens
#number of genes
vibrant_annot_METAG %>% select("gene") %>% unique() %>% nrow()
#number of transcribed genes
vibrant_annot_METAT %>% select("gene") %>% unique() %>% nrow()
```

Prepare table with viruses scaffolds data <- quality, lytic/lysogenic predicted host, drep cluster (to check if similar data in clusters) 
```{r , echo=FALSE}
#check hosts 
drep_clusters <- merge(drep_clusters, host_prediction_genome, by.x ="genome", by.y = "Virus", all.x = T, all.y = F)
drep_clusters_hosts_sum <-aggregate(phylum ~ primary_cluster , data = drep_clusters, FUN = function(x) length(unique(x)))
# Sort the pirates dataframe by height in decreasing order
drep_clusters_hosts_sum <- drep_clusters_hosts_sum[order(drep_clusters_hosts_sum$phylum, decreasing = TRUE),]
sum(drep_clusters_hosts_sum$phylum > 1 ) #126 clusters with host of multiple phylum - generalists
sum(drep_clusters_hosts_sum$phylum == 1 ) #558
length(unique(drep_clusters$primary_cluster)) - nrow(drep_clusters_hosts_sum) #1688

#no assicieted host
length(unique(drep_clusters$secondary_cluster)) - nrow(drep_sec_clusters_hosts_sum) #1720
drep_sec_clusters_hosts_sum <-aggregate(phylum ~ secondary_cluster , data = drep_clusters, FUN = function(x) length(unique(x)))
#more then one host - generalists
sum(drep_sec_clusters_hosts_sum$phylum > 1 ) #126??
sum(drep_sec_clusters_hosts_sum$phylum == 1 ) #570
#no associated host
length(unique(drep_clusters$secondary_cluster)) - nrow(drep_sec_clusters_hosts_sum) #1720
```

```{r nature of the data, echo=FALSE}
#ds DNA
#classes of viruses 
#eukaryotic viuses 
#pufM gene but not here
#also not here variable genes according to single copy number 
#lytic and lysogenic 
#viruses associted with fraction
#viruses vs env param
```


